# Welcome <a id="welcome"></a>

Welcome to Bioinformatics for Pathway Enrichment Analysis - June 2025

## Meet your Faculty

### Veronique Voisin
Research Associate,<br> Donnelly Centre for Cellular and Biomolecular Research, <br>University of Toronto 

<img src="./images/faculty/veronique_voisin.jpeg" style="float:right;width:125px;padding-left:10px;" />
Veronique is currently a bioinformatician applying pathway and networks analysis to high throughput genomics data for OICR cancer stem cell program. Previously, she worked on characterizing the gene signatures of different types of leukemias using a murine model

&nbsp;
&nbsp;

### Ruth Isserlin
Research data analyst,<br>Donnelly Centre for Cellular and Bimolecular Research, <br>University of Toronto

<img src="./images/faculty/ruth_isserlin.jpeg" style="float:left;width:125px;padding-right:10px;" />
Bioinformatician and data analyst in the Bader lab applying pathway and data analysis to varied data types. Developed Enrichment Map App for Cytoscape, an app to visually translate functional enrichment results from popular enrichment tools like GSEA to networks. Further developed the Enrichment Map Pipeline including development of additional Apps to help summarize and analyze resulting Enrichment Maps, including PostAnalysis, WordCloud, and AutoAnnotate App. 



***

Thank you for attending the Bioinformatics for Pathway Enrichment Analysis
 workshop! Help us make this workshop better by filling out [our survey](https://forms.gle/D8w8qyJ1r71rFnZe9).

*** 

## Class Materials 

You can download the printed course manual [here](https://drive.google.com/a/bioinformatics.ca/file/d/1HcPuiYUJe69w3_0aNpAfhk7DipcacA6r/view?usp=sharing).  

## Workshop Schedule {#schedule}

![](./images/time_table_pic.png)

## Pre-Workshop Materials and Laptop Setup Instructions {#pre-workshop}

### Laptop Setup Instructions

A Check list to setup your laptop can be found [here](https://docs.google.com/forms/d/e/1FAIpQLSdknqfaPi-XJDeFwji5xga7rg-jdGiYsZWxW6zTCjjqbHcHsw/viewform?usp=sharing)

Install these tools on your laptop before coming to the workshop:

### Basic programs

  1. A robust text editor:
      * For Windows/PC - [notepad++](http://notepad-plus-plus.org/)  
      * For Linux - [gEdit](http://projects.gnome.org/gedit/)  
      * For Mac – [TextWrangler](http://www.barebones.com/products/textwrangler/download.html)
  
  1. A file decompression tool.
      * For Windows/PC – [7zip](http://www.7-zip.org/).  
      * For Linux – [gzip](http://www.gzip.org).   
      * For Mac – already there.
  
  1. A robust internet browser such as:
      * Firefox 
      * Safari
      * Chrome
      * Microsoft Edge
  
  1. A PDF Viewer
      * Adobe Acrobat or equivalent

### Cytoscape Installation
Please install the latest version of [Cytoscape 3.10.2](https://github.com/cytoscape/cytoscape/releases/3.10.2/) or [Cytoscape Download](https://cytoscape.org/download.html) as well as a group of Cytoscape Apps that we will be using for different parts of the course.  

  1. Install Cytoscape 3.10.2:
      * Go to: https://github.com/cytoscape/cytoscape/releases/3.10.2/ OR https://cytoscape.org/download.html
      * Choose the version corresponding to your operating system (OS, Windows or UNIX)
      * Follow instructions to install cytoscape
      * Verify that Cytoscape has been installed correctly by launching the newly installed application
      
  1. Install the following Cytoscape Apps - Apps are installed from within Cytoscape. 
      * In order to install Apps launch Cytoscape  
      * From the menu bar, select ‘Apps’, then ‘App Store’, then 'Show App Store'. ![](./images/cytoscape_app_menu.png)
      * App Store will appear in left hand Panel ![](./images/Cytoscape_app_manager.png)
      * Within search bar at the top of the panel, search for the app listed below.  Once you click on search icon a web browser will be launched with the apps that match your search.
      * Select the correct app (there might be a few that match your search term).
      * Click on "Install" ![](./images/app_store_download.png)
      * install the following:  
          * EnrichmentMap 3.4.0
          * EnrichmentMap Pipeline Collection 1.1.0 (it will install ClusterMaker2 v2.3.4, WordCloud v3.1.4 and AutoAnnotate v1.5.0)
          * GeneMANIA 3.5.3
          * IRegulon  1.3
          * ReactomeFIPlugin 8.0.6 - http://apps.cytoscape.org/apps/reactomefiplugin 
          * stringApp 2.0.3
          * scNetViz 1.7.1
          * yFiles Layout Algorithms	1.1.4
    
  1. Install the data set within GeneMANIA app. **This requires time and a good network connection to download completely (~15mins)**
      * From the menu bar, select ‘Apps’, hover over ‘GeneMANIA’, then select ‘Choose Another Data Set’.
      * From the list of available data sets, select the most recent and under ‘Include only these networks:’ select ‘all’. Click on ‘Download’.
      * An ‘Install Data’ window will pop-up. Select H.Sapiens Human (2589 MB). Click on ‘Install’.
  
### GSEA Installation  
Please install the latest version of GSEA (4.3.3)  

  1. Download GSEA
      * Go to the [GSEA page](http://www.broadinstitute.org/gsea/index.jsp)    
      * Register (using an institutional email address) 
      * Login 
      * Locate the Download page and download the version corresponding to your system
          * MAC users:  download GSEA_4.3.3.app.zip
          * Window users: download GSEA_Win_4.3.3-installer.exe
          * Unix users: download GSEA_Linux_4.3.3.zip
          * ![](./images/gsea_download_exe.png)
      * Launch GSEA to test it.
  
  1. Download GSEA for command line :  this is necessary for all platform users to run GSEA from a script (integrated workflow on day 3)
      * Download  GSEA_4.3.3.zip (and keep it for later use during the workshop)
      * ![](./images/gsea_download_command.png)

### Docker Installation
Docker is a virtualization software that allows you to run programs isolated from your current laptop set up.  It eases the burden of installing multiple software requirements and packages.

  1. Please install the latest version of Docker Desktop.
      * [Windows](https://docs.docker.com/desktop/install/windows-install/)
      * [OSX](https://docs.docker.com/desktop/install/mac-install/) - make sure to select the version specific for your computer.  Newer macs (later than 2021) will contain the Apple silicon (M1/M2/M3).  Older computers might be intel based.
      * [Linux](https://docs.docker.com/desktop/install/linux-install/)
    
  1. Pull the required images used in the course
      * Open docker desktop (If docker is already running you can find the docker icon in your task bar.  Right click on the icon and select "Go to Dashboard")
      * ![](./images/docker_dashboard_open.png)
      * Find the search bar in the docker desktop dashboard
      * ![](./images/docker_dashboard_search.png)    
      * Enter "risserlin/workshop_base_image" into the search bar at the top of the docker desktop dashboard.
      * ![](./images/docker_dashboard_imagefind.png)
      * Click on "Pull" to download the image.   
      * ![](./images/docker_dashboard_imagefind_annot.png)
      * Enter "risserlin/nest_docker_lymphnode" into the search bar at the top of the docker desktop dashboard.
      * ![](./images/docker_dashboard_imagefind_nest.png) 
      * Click on "Pull" to download the image.   
      * ![](./images/docker_dashboard_imagefind_nest_annot.png)
      
  1. You should now see both of your images listed in the docker desktop image section (in the local tab)
      * ![](./images/docker_dashboard_image_installed.png)

## Pre-workshop Tutorials

It is in your best interest to complete these before the workshop.

### Cytoscape Preparation tutorials

Go to : https://github.com/cytoscape/cytoscape-tutorials/wiki and follow :

  * [Tour of Cytoscape](https://cytoscape.org/cytoscape-tutorials/protocols/tour-of-cytoscape/#/)
  * [Basic Data Visualization](https://cytoscape.org/cytoscape-tutorials/protocols/basic-data-visualization/#/)

### R Tutorial

Use your newly installed docker workshop_base_image to try out R and go through the following tutorial - 

  * [R tutorial](https://genviz.org/module-02-r/0002/02/01/introductionToR/) - There will be instructions on how to install R and RStudio in the tutorial.  Instead of installing use the workshop_base_image docker image that you installed above as follows:
      * Open docker desktop (If docker is already running you can find the docker icon in your task bar.  Right click on the icon and select "Go to Dashboard")
      * ![](./images/docker_dashboard_open.png)
      * Click on Images --> Local --> And find the workshop_base_image.  click on the Play button
      * ![](./images/docker_launch_image.png)
      * Expand the 'optional settings'
      * ![](./images/docker_new_container.png)
      * Change - 
        * 'container name' to R_tutorial, 
        * 'Host Port' to 8787, 
        * Add environment variable PASSWORD and set value to password
        * ![](./images/docker_container_settings.png)
        
      * Click on 'Run'.  Docker will display a tab with all the information about the container you just launched
      * ![](./images/docker_container_success.png)
      * Open a web browser and navigate to localhost:8787
      * ![](./images/docker_localhost.png)
      * Username - rstudio, password - password (or whatever you entered as the PASSWORD settting when you launch the container)
      * You should now have an r studio session running in your web browser
      * ![](./images/docker_rstudio.png)
      * When you are finished doing the tutorial remember to turn off your docker container and dacker as they both use up a lot of your computer's resources.
      * ![](./images/docker_stop.png)      
  
### Pre-workshop Readings and Lectures

  1. Video Module 1 - [Introduction to Pathway and Network Analysis by Gary Bader](#intro)
  1. ***Pathway enrichment analysis and visualization of omics data using g:Profiler, GSEA, Cytoscape and EnrichmentMap*** Reimand J, Isserlin R, Voisin V, Kucera M, Tannus-Lopes C, Rostamianfar A, Wadi L, Meyer M, Wong J, Xu C, Merico D, Bader GD [Nat Protoc. 2019 Feb;14(2):482-517](https://www.nature.com/articles/s41596-018-0103-9) - [Available here as well](http://baderlab.org/Publications#EM_2019)
 
***

### Additional tutorials <a id="additionaltutorials"></a>
  
  * ***How to visually interpret biological data using networks*** Merico D, Gfeller D, Bader GD [Nature Biotechnology 2009 Oct 27, 921-924](https://www.nature.com/articles/nbt.1567)  - [Available here as well](http://baderlab.org/Publications#interpret_networks)
  
  * ***g:Profiler--a web-based toolset for functional profiling of gene lists from large-scale experiments.*** Reimand J, Kull M, Peterson H, Hansen J, Vilo J   [Nucleic Acids Res. 2007 Jul;35](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1933153/)
  
  * ***g:Profiler: a web server for functional enrichment analysis and conversions of gene lists (2019 update)*** Raudvere U, Kolberg L, Kuzmin I, Arak T, Adler P, Peterson H, Vilo J [Nucleic Acids Res. 2019 May 8](https://academic.oup.com/nar/advance-article/doi/10.1093/nar/gkz369/5486750)
  
  * ***Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles*** Subramanian A, Tamayo P, Mootha VK, Mukherjee S, Ebert BL, Gillette MA, Paulovich A, Pomeroy SL, Golub TR, Lander ES, Mesirov JP [Proc Natl Acad Sci U S A. 2005 Oct 25;102(43)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/)
  
Interacting with Cytoscape using CyRest and command lines (for advanced users):
https://github.com/cytoscape/cytoscape-automation/blob/master/for-scripters/R/advanced-cancer-networks-and-data-rcy3.Rmd


