# Module 2 lab - g:Profiler {#gprofiler-lab-new}

**This work is licensed under a [Creative Commons Attribution-ShareAlike 3.0 Unported License](http://creativecommons.org/licenses/by-sa/3.0/deed.en_US). This means that you are able to copy, share and modify the work, as long as the result is distributed under the same license.**

authors: Veronique Voisin and Ruth Isserlin
 
## Introduction

During this practical lab, we will perform pathway enrichment analysis using a defined gene list.


## Goal of the exercise 1

For this exercise, our goal is to run pathway enrichment analysis using g:Profiler and explore the results.
g:Profiler performs a gene-set enrichment analysis using a hypergeometric test (Fisher’s exact test). The [Gene Ontology](http://geneontology.org/) Biological Process, [Reactome](https://reactome.org/) and [WikiPathways](https://www.wikipathways.org/) sources are going to be used as pathway databases. 

g:Profiler can be used using the website at https://biit.cs.ut.ee/gprofiler/gost. However, for this practical lab, we will run it from the g:Profiler R package.

We will run the query and explore the table of results and visualize the results as bar and dot plots.

One of the greatest features of [g:Profiler](https://biit.cs.ut.ee/gprofiler/gost) is that it is updated on a regular basis and most of the previous versions are available online ont the [gprofiler archive](https://biit.cs.ut.ee/gprofiler/page/archives).

The [gprofielr2](https://biit.cs.ut.ee/gprofiler/page/r) -[g:Profiler](https://biit.cs.ut.ee/gprofiler/gost) R implementation is a wrapper for the web version.  You require an internet connection to get enrichment results.  


## Data

g:Profiler requires a list of genes: for this, we use genes with frequent somatic SNVs identified in TCGA exome sequencing data of 3,200 tumors of 12 types. The MuSiC cancer driver mutation detection software was used to find 127 cancer driver genes that displayed higher than expected mutation frequencies in cancer samples (Supplementary Table 1, which is derived from column B of Supplementary Table 4 in [Kandoth C. et al.](https://www.nature.com/articles/nature12634). 

## Exercise 1 - run g:Profiler {#exercise-1}
Before starting this exercise, download the required files:

```{block, type="rmd-datadownload"}
Right click on link below and select "Save Link As...".

We recommend saving all these files in a personal project data folder before starting. We also recommend creating an additional result data folder to save the files generated while performing the protocol.

```


* [Pancancer_genelist.txt](./Module2/gprofiler/data/Pancancer_genelist.txt)

```{block, type="rmd-datadownload"}
Open RStudio and follow script instructions below.
```


### Step 1 - Install and Load libraries

```{r message=FALSE, warning=FALSE}
# CRAN and Bioconductor setup
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
# List of CRAN packages
cran_packages <- c(
  "tidyverse",
  "knitr",
  "kableExtra",
  "glue",
  "RCurl",
  "webshot2",
  "ggridges",
  "igraph",
  "stringr",
  "data.table"
)
# List of Bioconductor packages
bioc_packages <- c(
  "gprofiler2",
  "GSA",
  "fgsea",
  "clusterProfiler",
  "enrichplot"
)
# Install CRAN packages if not already installed
for (pkg in cran_packages) {
  if (!requireNamespace(pkg, quietly = TRUE)) {
    install.packages(pkg)
  }
}
# Install Bioconductor packages if not already installed
for (pkg in bioc_packages) {
  if (!requireNamespace(pkg, quietly = TRUE)) {
    BiocManager::install(pkg)
  }
}
# Load the libraries
library(tidyverse)
library(knitr)
library(kableExtra)
library(glue)
library(RCurl)
library(webshot2)
library(ggridges)
library(igraph)
library(stringr)
library(data.table)
# Bioconductor packages
library(gprofiler2)
library(GSA)
library(fgsea)
library(clusterProfiler)
library(enrichplot)
```


### Step 2 - Input query

Save the gene list ([Pancancer_genelist.txt](./Module2/gprofiler/data/Pancancer_genelist.txt)) on your local computer. Remember where you stored it as you need to know the path to the file to start the practical lab.

We recommend that you save it in your home directory in a folder called Module2 (./Module2) as it will be easier for you to follow the notebook.

```{block, type="rmd-tip"}
***Find path on Windows:***
Using File Explorer
Navigate to the file in File Explorer.

Right-click the file and select "Properties".

In the General tab, look for:

Location: This shows the folder path.

File name: Combine this with the location to get the full path.

Example:

Location: `C:\\Users\\Veronique\\Documents`

File name: example.txt

Full path: `C:\\Users\\Veronique\\Documents\\example.txt`
```


```{block, type="rmd-tip"}
***Find path on a Mac:***
Using Finder with Right-Click
Open Finder and locate the file.

Right-click (or Control-click) the file.

Hold the Option key — the menu item “Copy [filename] as Pathname” will appear.

Click it to copy the full file path.

Paste it anywhere (e.g., in a Terminal or text editor) : `/Users/veronique/Documents/example.txt`

```



```{r}

##The first step is to read the file containing the gene list. 
# For this purpose, we need to know the path to the file = where we stored our file on our computer 
# Please see my example below:
# myfile = "/Users/veronique/Documents/CO_summer_school/Pancancer_genelist.txt" ##!!REPLACE BY YOUR PATH!!##

myfile = "./Module2/gprofiler/data/Pancancer_genelist.txt"

##YOUR TURN: create the myfile object using the path to your file / name of the file ##

#This function will read the genelist file and import it as a data frame object in our R environment
current_genelist <- read.table(  file = myfile,
                                 header = FALSE,
                                 sep = "\t", quote = "",
                                 stringsAsFactors = FALSE)
#head(current_genelist)
#class(current_genelist)

#We take the gene list which is stored as the first column of our dataframe and convert it to a character vector for future use
query_set <- current_genelist$V1

#class(query_set)

```


### Step 3 - Run g:Profiler
The next step is to run pathway enrichment analysis using [gost g:Profiler](https://biit.cs.ut.ee/gprofiler/gost) using the g:Profiler2 R package.
For detailed descriptions of all the parameters that can be specified for the gost g:profiler function see the R package information - [here](https://cran.r-project.org/web/packages/gprofiler2/index.html) and [here](https://rdrr.io/cran/gprofiler2/man/gost.html).

For this query we are specifying - 

  * query - the set of genes of interest, as loaded in from the gene list file (`query_set`).
  * significant - set to FALSE because we want g:Profiler to return all the results not just the ones that it deems significant by its predetermined threshold. We will filter the results for significance in a next step.
  * ordered_query - set to FALSE (because we are not taking into account the order of the list)
  * exclude_iea - set to TRUE. We are removing the electronic inferred annotations from the GO database
  * correction_method - set to fdr.  by default g:Profiler uses g:Scs
  * organism - set to "hsapiens" for homo sapiens.  Organism names are constructed by concatenating the first letter of the name and the family name (according to gprofiler2 documentation)
  * source - the geneset source databases to use for the analysis.  We recommend using GO biological process (GO:BP), WikiPathways (WP) and Reactome (Reac) but there are additional sources you can add (GO molecular function or cellular component(GO:MF, GO:CC), KEGG, transcription factors (TF), microRNA targets (MIRNA), corum complexes (CORUM), Human protein atlas (HPA),Human phenotype ontology (HP) ) 

```{r run gprofiler supplied gs}
gprofiler_results <- gost(query = query_set ,
                          significant=FALSE,
                          ordered_query = FALSE,
                          exclude_iea=TRUE,
                          correction_method = "fdr",
                          organism = "hsapiens",
                          source = c("REAC","WP","GO:BP"))
```

```{r}
#get the gprofiler results table
enrichment_results <- gprofiler_results$result

#display the names of all columns
colnames(enrichment_results)

#display the top rows
head(enrichment_results)

```

The enrichment_results dataframe contains different columns. Let's rearrange the table to display the most important information. We will arrange the table so it contains: the database origin, the name of each of the pathway, term size , query size, intersection size and adjusted pvalue. - 

  * Term size: number of genes in the pathway as it is in the database
  * Query size: number of genes in our gene list
  * Intersection size: number of genes overlapping between our gene list and the tested pathway
  * pvalue: adjusted pvalue (corrected for multiple hypothesis testing using the Benjamini-Hochberg method)
  
The results are ordered by significance, from the most significant pathway (lowest p-value) to the least significant.

```{r}

enrichment_results$p_value <- formatC(enrichment_results$p_value, format = "e", digits = 2)

enrichment_results = enrichment_results[ , c("source", "term_name", "term_size", "query_size", "intersection_size", "p_value")]

kable(head(enrichment_results, n=20), caption = "Enrichment Result") %>%
kable_styling(bootstrap_options = c("striped", "condensed"), font_size = 10)
```



### Step 4 - Explore top table of results


For this analysis, we chose to use 3 databases: GO:BP, Reactome and Wikipathways. 
If we look at the top table only, we are only going to see the results of GO:BP but it is interesting to look at the top results for each of the 3 databunique(enrichment_results$source)

```{r}
##Get the name of the databases
unique(enrichment_results$source) 

##Filter the dataframe to retrieve only the Reacome results
enrichment_results_Reac <- enrichment_results %>%
  filter(source == "REAC")

##Now display the top 20 pathways
kable(head(enrichment_results_Reac, n=20), caption = "Enrichment Result") %>%
kable_styling(bootstrap_options = c("striped", "condensed"), font_size = 10)
                                                  
```
```{r}
##YOUR TURN##
##DISPLAY THE TOP 20 PATHWAYS FOR THE WIKIPATHWAY DATABASE ("WP")
```



### Step 5 - Filter results by geneset size

Restrict the results to just the ones that have at maximum gene-set size of 1000 and a minimum gene-set size of 10 and with an adjusted pvalue of 0.05.- 

  * maximum gene-set size of 1000: will remove very generic pathway terms that are less informative like "regulation of gene expression"
  * minimum gene-set size of 10: will remove small pathways where the pvalue is significant athough the overlap size is very small
  * p_thres: aim is to retrieve only pathways that are significant under this adjusted pvalue threshold of 0.05 (5% false positive)


```{r echo=TRUE}

min_gs_size = 3
max_gs_size = 1000
p_thres = 0.05
# filer by params defined above
# by default we have set the max and min gs size to 250 and 3, respectively.


##the pvalue was stored as a character vector, we have to transform back into a numerical vector to be able to use it to filter values
enrichment_results$p_value <- as.numeric(enrichment_results$p_value) 

enrichment_results_mxgssize_1000_min_10_adjp_0_05 <- 
                        subset(enrichment_results,
                                   term_size >= min_gs_size & 
                                   term_size <= max_gs_size & 
                                   p_value <= p_thres )

#
myrows = nrow(enrichment_results_mxgssize_1000_min_10_adjp_0_05)


print(glue("It results in {myrows} selected pathways with maximum gene-set size of {max_gs_size} and minimum gene-set of {min_gs_size} with an adjusted p-value of {p_thres} . "))



```


In a second step, try different thresholds of maximum gene-set size and adjusted pvalue threshold: 500 and 250 for  * max_gs_size and 0.01 or 0.001 for pvalue- 

   * Option 1 
     *  max_gs_size = 500
     * p_thres = 0.01
  
or

   * Option 2 
     *  max_gs_size = 250
     *  p_thres = 0.001

```{r}
##YOUR TURN##

## How many pathways do you get for each option (nrow()): copy your answer in the chat
```


### Step 6 - Run g:profiler with your own genesets (example using BaderLab genesets)

With regards to pathway sets there are two options when using [g:Profiler](https://biit.cs.ut.ee/gprofiler/gost) - 

  * Use the genesets that are supplied by [g:Profiler](https://biit.cs.ut.ee/gprofiler/gost) as we just did.
  * Upload your own genesets. 
  
The most common reasons for supplying your own genesets is the ability to use up to date annotations or in-house annotations that might not be available in the public sphere yet.  It can be used to test the enrichment in a particular pathway of interest that might not be in the general pathway database but that you obtained for example by extracting the data from a published paper.
You need to format this pathway in a [.gmt file](https://www.gsea-msigdb.org/gsea/doc/GSEAUserGuideFrame.html).

In this example, we will upload the Baderlab gene set file that contains multiple database sources. We will filter the gene-set prior to uploading the gene-set file to g:Profiler. 

### Step 7 - Download and load Bader lab geneset file

Download the latest [Bader lab genesets](https://download.baderlab.org/EM_Genesets/current_release/Human/)
The file will be downloaded on your local computer in the workin directory (working_dir) of your choice.


```{r download baderlab gmt file, message=FALSE, warning=FALSE}

dest_gmt_file = ""

#flag to trigger download of latest gmt file - change to TRUE 
# if you want to check the website and download the latest
# version instead of using the specified May 01, 2025 version.
downlaod_latest_version = FALSE

#working_dir = "/Users/veronique/Documents/CO_summer_school"
working_dir = file.path(getwd(),"Module2")
##  YOUR TURN: replace working directory by your own path to your working directory ##


if((dest_gmt_file == "") && downlaod_latest_version){

  gmt_url = "http://download.baderlab.org/EM_Genesets/current_release/Human/symbol/"

  #list all the files on the server
  filenames = RCurl::getURL(gmt_url)
  tc = textConnection(filenames)
  contents = readLines(tc)
  close(tc)
  
  #get the gmt that has all the pathways and does not include 
  # terms inferred from electronic annotations(IEA)
  #start with gmt file that has pathways only
  rx = gregexpr("(?<=<a href=\")(.*.GOBP_AllPathways_noPFOCR_no_GO_iea.*.)(.gmt)(?=\">)",
    contents, perl = TRUE)
  gmt_file = unlist(regmatches(contents, rx))
  
  dest_gmt_file <- file.path(working_dir,gmt_file)
  
  if(!file.exists(dest_gmt_file)){
    download.file(
      paste(gmt_url,gmt_file,sep=""),
      destfile=dest_gmt_file
    )
  }
} else if(!downlaod_latest_version){
  
  gmt_url = "http://download.baderlab.org/EM_Genesets/May_01_2025/Human/symbol/"
 gmt_file = "Human_GOBP_AllPathways_noPFOCR_no_GO_iea_May_01_2025_symbol.gmt"
  
  #download the specific file specified above
  dest_gmt_file <- file.path(working_dir,gmt_file)
  
  if(!file.exists(dest_gmt_file)){
    download.file(
      paste(gmt_url,gmt_file,sep=""),
      destfile=dest_gmt_file
    )
  }
  
}

```

Load in the GMT file

This step will take the gmt file that we just downloaded on our computers and upload them into our R environment in the GSA.genesets object called "genesets_baderlab_genesets".
The capture.output() function is only there to redirect the printed messages for a clearer notebook.
```{r echo=TRUE}

capt_output <- capture.output(genesets_baderlab_genesets <- 
                                GSA.read.gmt(filename = dest_gmt_file))

names(genesets_baderlab_genesets$genesets) <- 
                                genesets_baderlab_genesets$geneset.names
```

### Step 8 -  Filter Bader lab geneset file

The g:Profiler interface only allows for filtering genesets by size only after the analysis is complete.  After the analysis is complete means the filtering is happening after Multiple hypothesis testing.  Filtering prior to the analysis will generate more robust results because we exclude the uninformative large genesets prior to testing changing the sets that multiple hypothesis filtering will get rid of.  

Create multiple gmt files with different filtering thresholds - remove 
  * genesets greater than 250 genes
  * geneset greater than 1000 genes
  * geneset greater than 10000 genes

```{r}
# Filter geneset GSA object by specified gs size threshold 
#
# Given - 
# genesets - in GSA object
# gs_sizes - list of all the sizes of the genesets found in the genesets
# filter_threshold - value to filter the geneset by.  
# 

#This function will return a filtered GSA object. We first create the function. 
#  The function returns a filtered genesets in a GSA object
filter_genesets <- function(genesets, gs_sizes, filter_threshold) {
  
  filtered_genesets <- genesets
  
  filtered_genesets$genesets <- genesets$genesets[
                      which(gs_sizes<filter_threshold)]
  filtered_genesets$geneset.names <- genesets$geneset.names[
                      which(gs_sizes<filter_threshold)]
  filtered_genesets$geneset.descriptions <- genesets$geneset.descriptions[
                      which(gs_sizes<filter_threshold)]

  return(filtered_genesets)
}

# You can not simply write a list of lists to a file in R.  In order
# to output the new geneset file you need to convert it ot a data.frame
# To do this convert the list of genes to a tab delmiated list in one column
# of the dataframe.
# format to write out to a file. 
#
# Given - 
# genesets - in GSA object
 
# This function returns the formatted genesets as data frame
  
format_genesets <- function(genesets) {
    
  collapsed_genesets <- data.frame(name=genesets$geneset.names, 
                            description= genesets$geneset.description)
  collapsed_genesets$genes <- unlist(lapply(genesets$genesets,
                                             FUN=function(x){
                                              paste(x,collapse = "\t")
                                            }))
  
  return(collapsed_genesets)
}

```



The format of the GMT file is described [https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29](here) and consists of rows with the following

  * Name
  * Description
  * tab delimited list of genes a part of this geneset
  
Write out the gmt file with genenames

```{r write output file genenames}

#get the geneset sizes
gs_sizes_baderlab_sets <- lapply(genesets_baderlab_genesets$genesets,
                                 FUN = function(x){
                                               length(x)
                                                  })

# max 10,000
genesets_baderlab_genesets_max10000 <- filter_genesets(genesets_baderlab_genesets,
                                                gs_sizes_baderlab_sets, 
                                                10000)

genesets_baderlab_genesets_max10000_filename <- gsub(x =dest_gmt_file, 
                                                  pattern = "symbol" ,
                                                  replacement = "symbol_max10000"
                                                     )

if(!file.exists(genesets_baderlab_genesets_max10000_filename)){

  write.table(x = format_genesets(genesets_baderlab_genesets_max10000),
            file = genesets_baderlab_genesets_max10000_filename,
            quote = FALSE,sep = "\t",row.names = FALSE,
            col.names=TRUE)
}

#max gs size of 1,000
genesets_baderlab_genesets_max1000 <- filter_genesets(genesets_baderlab_genesets,
                                                      gs_sizes_baderlab_sets, 
                                                      1000)
genesets_baderlab_genesets_max1000_filename <- gsub(x =dest_gmt_file, 
                                                pattern = "symbol" ,
                                                replacement = "symbol_max1000"
                                                     )
if(!file.exists(genesets_baderlab_genesets_max1000_filename)){

  write.table(x = format_genesets(genesets_baderlab_genesets_max1000),
            file = genesets_baderlab_genesets_max1000_filename,
            quote = FALSE,sep = "\t",row.names = FALSE,
            col.names=TRUE)
}

#max gs size of 250
genesets_baderlab_genesets_max250 <- filter_genesets(genesets_baderlab_genesets,
                                                      gs_sizes_baderlab_sets, 
                                                      250)


genesets_baderlab_genesets_max250_filename <- gsub(x =dest_gmt_file, 
                                                  pattern = "symbol" ,
                                                  replacement = "symbol_max250"
                                                     )
if(!file.exists(genesets_baderlab_genesets_max250_filename)){
  write.table(x = format_genesets(genesets_baderlab_genesets_max250),
            file = genesets_baderlab_genesets_max250_filename,
            quote = FALSE,sep = "\t",row.names = FALSE,
            col.names=TRUE)
}

```

### Step 9 - Upload the gmt files to gprofiler and rerun g:Profiler

In order to use your own genesets with g:Profiler you need to upload the the file to g:Profiler server first.  The function will return an ID that you need to specify in the organism parameter of the g:Profiler gost function call. 
It is done by using the upload_GMT_file() function which is included in the gprofiler2 package.
Note: it took about 2 minutes to run this chunck of code.
```{r}
custom_gmt_max250 <- upload_GMT_file(
                        gmtfile=genesets_baderlab_genesets_max250_filename)
custom_gmt_max1000 <- upload_GMT_file(
                        gmtfile=genesets_baderlab_genesets_max1000_filename)
custom_gmt_max10000 <- upload_GMT_file(
                        gmtfile=genesets_baderlab_genesets_max10000_filename)
```

For this query we are specifying - 

  * query - the set of genes of interest, as loaded in from the gene set file.
  * significant - set to FALSE because we want g:Profiler to return all the results not just the ones that it deems significant by its perdetermined threshold.
  * ordered_query - set to FALSE (but you can try setting it to true as well because for this set of genes they are ordered in order of their significance)
  * correction_method - set to fdr.  by default g:Profiler uses g:Scs
  * organism - set to the custom_gmt ID ( for this run it is - custom_gmt_max250) that we received when we uploaded our genetset file.


```{r run gprofiler upload gs}
gprofiler_results_custom_max250 <- gost(query = query_set ,
                                     significant=TRUE,
                                 ordered_query = FALSE,
                                  exclude_iea=TRUE,
                                     correction_method = "fdr",
                                 organism = custom_gmt_max250
                                     )

gprofiler_results_custom_max1000 <- gost(query = query_set ,
                                     significant=TRUE,
                                      ordered_query = FALSE,
                                      exclude_iea=TRUE,
                                     correction_method = "fdr",
                                 organism = custom_gmt_max1000
                                     )

gprofiler_results_custom_max10000 <- gost(query = query_set ,
                                     significant=TRUE,
                                      ordered_query = FALSE,
                                      exclude_iea=TRUE,
                                     correction_method = "fdr",
                                 organism = custom_gmt_max10000
                                     )
```


```{r}

 #get the gprofiler results table
enrichment_results_customgmt_max250 <- gprofiler_results_custom_max250$result
enrichment_results_customgmt_max1000 <- gprofiler_results_custom_max1000$result
enrichment_results_customgmt_max10000 <- gprofiler_results_custom_max10000$result

    
enrichment_results_customgmt_max250[1:5,]

```

Filter the table to include just the columns that are required for the generic enrichment map file results [GEM](https://enrichmentmap.readthedocs.io/en/latest/FileFormats.html#generic-results-files). Restrict the results to just the ones that have at least min_gs_size and less than max_gs_size terms and  min_intersection size include only the term_id, term_name, p_value (and p_value again because the p_value is actually the corrected p-value.  The output file does not contain the nominal p_value.  For down stream analysis though it is expected to have both a p-value and a q-value so just duplicate the q-value as both p-value and q-value)

```{r}
# filer by params defined above
enrichment_results_customgmt_max250$p_value <- formatC(enrichment_results_customgmt_max250$p_value, format = "e", digits = 2)



enrichment_results_customgmt_max250 = enrichment_results_customgmt_max250[ order(enrichment_results_customgmt_max250$p_value), c("term_id", "term_name", "term_size", "query_size", "intersection_size", "p_value")]

kable(head(enrichment_results_customgmt_max250, n=20), caption = "Enrichment Result") %>%
kable_styling(bootstrap_options = c("striped", "condensed"), font_size = 10)

#kable(tail(enrichment_results_customgmt_max250, n=20), caption = "Enrichment Result") %>%
#kable_styling(bootstrap_options = c("striped", "condensed"), font_size = 10)



```
```{r}
##YOUR TURN##
## Display the top tables for the gprofiler_results_custom_max1000 and enrichment_results_customgmt_max10000 results
```

## Exercise 2 - visualization
We will use clusterProfiler style visualization.
### Step 1 - Bar plot
```{r}
##Calculate a score
enrichment_results_customgmt_max250$score = -log10(as.numeric(enrichment_results_customgmt_max250$p_value))

enrichmentTidy <- enrichment_results_customgmt_max250 %>%
  as_tibble() %>%
  arrange(desc(score)) %>%
  slice_head(n = 20)

p = ggplot(enrichmentTidy, aes(reorder(term_name, score), score)) +
  geom_col(aes(fill = score), width=0.8) +
  scale_fill_gradient(low = "#C1F6F8", high = "#00BFC4")+
  coord_flip() +
  labs(x="", y="Normalized Enrichment Score",
       title="Pathway enrichment analysis ") + 
  theme_minimal()

p


```

### Step 2 - Dot plot with segment
```{r}

enrichmentTop20 <- enrichmentTidy %>%
  arrange(desc(score)) %>%
  slice_head(n = 20)

p = ggplot(enrichmentTop20, aes(
  y = fct_reorder(term_name, score),
  x = score,
  color = score,
  size = score
)) +
  geom_point() +
  scale_color_gradientn(
    colours = c("#56A443", "#8AA443", "#f7ca64"),
    trans = "log10",
    guide = guide_colorbar(reverse = FALSE, order = 1)
  ) +
  scale_size_continuous(range = c(4, 8)) +
  theme_bw(base_size = 12) +
  xlab("score -log10(adj pvalue)") +
  ylab(NULL) +
  ggtitle("Pathway Enrichment Analysis") +
  theme(
    axis.text.y = element_text(size = 10)
  )
p
```

### Step 3 - Ridge plot
```{r}




enrichmentTop20 <- enrichmentTidy %>%
  arrange(desc(score)) %>%
  slice_head(n = 20)

# Expand each term to have multiple x values
ridge_data <- enrichmentTop20 %>%
  rowwise() %>%
  mutate(x_vals = list(seq(-1, 1, length.out = 100))) %>%
  unnest(x_vals)

p = ggplot(ridge_data, aes(
  x = x_vals,
  y = fct_reorder(term_name, score),
  height = score,
  group = term_name,
  fill = score
)) +
  geom_density_ridges(
    stat = "identity",
    scale = 0.9,
    rel_min_height = 0.01,
    color = "white",
    alpha = 0.9
  ) +
  scale_fill_gradientn(
    colours = c("#E6194B", "#3CB44B", "#4363D8"),
    trans = "log10",
    guide = guide_colorbar(title = "-log10(p-value)")
  ) +
  theme_ridges(font_size = 12, grid = TRUE) +
  xlim(-1.2, 1.2) +
  xlab("") +
  ylab(NULL) +
  ggtitle("Pathway enrichment analysis")

p = p + theme_minimal(base_size = 12)

p
```

### Step 4 - Network visualization - EnrichmentMap
#### Get names of genes in the overlap for each pathway
```{r echo=TRUE, message=FALSE, warning=FALSE}
gprofiler_results_custom_max250 <- gost(query = query_set ,
                                     significant=TRUE,
                                 ordered_query = FALSE,
                                  exclude_iea=TRUE,
                                     correction_method = "fdr",
                                 organism = custom_gmt_max250
                                     )


subset_genesets <- genesets_baderlab_genesets$genesets[which(genesets_baderlab_genesets$geneset.names %in% gprofiler_results_custom_max250$result$term_id)]
      

genes <- lapply(subset_genesets,FUN=function(x){intersect(x,query_set)})
  
  # For each of the genes collapse to the comma separate text
genes_collapsed <- unlist(lapply(genes,FUN=function(x){paste(x,collapse = ",")}))
  
genes_collapsed_df <- data.frame(
                            term_id = names(genes), 
                            genes = genes_collapsed,stringsAsFactors = FALSE)
  


formatted_results <- merge(gprofiler_results_custom_max250$result,genes_collapsed_df,by.x="term_id" , by.y="term_id" )

formatted_results <- formatted_results[order(formatted_results$p_value),]
colnames(formatted_results)     

formatted_results_sel = formatted_results[ , c("term_name",  "genes") ]





```

#### Create the enrichment map
```{r message=FALSE, warning=FALSE}
# Create your data frame

df = formatted_results_sel[1:10,]
# Step 2: Convert gene strings into lists
df$gene_list <- strsplit(df$genes, ",")

# Create edges based on shared genes
edges <- data.frame(from = character(), to = character(), weight = integer(), stringsAsFactors = FALSE)

for (i in 1:(nrow(df) - 1)) {
  for (j in (i + 1):nrow(df)) {
    genes_i <- df$gene_list[[i]]
    genes_j <- df$gene_list[[j]]
    shared_genes <- intersect(genes_i, genes_j)
    if (length(shared_genes) > 0) {
      edges <- rbind(edges, data.frame(
        from = df$term_name[i],
        to = df$term_name[j],
        weight = length(shared_genes),
        stringsAsFactors = FALSE
      ))
    }
  }
}

# Create igraph object
g <- graph_from_data_frame(d = edges, vertices = df$term_name, directed = FALSE)

# Plot the graph
p = plot(
  g,
  vertex.label.cex = 0.8,
  vertex.label.color = "black",
  vertex.size = 30,
  edge.width = E(g)$weight,
  edge.color = "gray50",
  vertex.color = "skyblue",
  layout = layout_with_fr(g)
)


```







