<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 2 lab - fGSEA | Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )</title>
  <meta name="description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 2 lab - fGSEA | Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 2 lab - fGSEA | Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )" />
  
  <meta name="twitter:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  

<meta name="author" content="Ruth Isserlin, Veronique Voisin" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gprofiler-lab-new.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://training.computeontario.ca/index.php"><img src="./images/compute_ontario_logo.png" width="280"></a><a href="./">Compute Ontario Summer school</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Compute Ontario Summer School</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html"><i class="fa fa-check"></i>Welcome <a id="welcome"></a></a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#meet-your-faculty"><i class="fa fa-check"></i>Meet your Faculty</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#veronique-voisin"><i class="fa fa-check"></i>Veronique Voisin</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#ruth-isserlin"><i class="fa fa-check"></i>Ruth Isserlin</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop"><i class="fa fa-check"></i>Pre-Workshop Materials and Laptop Setup Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#laptop-setup-instructions"><i class="fa fa-check"></i>Laptop Setup Instructions</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#basic-programs"><i class="fa fa-check"></i>Basic programs</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-installation"><i class="fa fa-check"></i>Cytoscape Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#rstudio-installation"><i class="fa fa-check"></i>RStudio Installation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-tutorials"><i class="fa fa-check"></i>Pre-workshop Tutorials</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-preparation-tutorials"><i class="fa fa-check"></i>Cytoscape Preparation tutorials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-readings-and-lectures"><i class="fa fa-check"></i>Pre-workshop Readings and Lectures</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#additional-tutorials"><i class="fa fa-check"></i>Additional tutorials <a id="additionaltutorials"></a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i>Module 1 - Introduction to Pathway and Network Analysis (Gary Bader)</a></li>
<li class="chapter" data-level="" data-path="module-2-finding-over-represented-pathways-veronique-voisin.html"><a href="module-2-finding-over-represented-pathways-veronique-voisin.html"><i class="fa fa-check"></i>Module 2: Finding Over-represented Pathways (Veronique Voisin)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html"><i class="fa fa-check"></i>Module 2 lab - g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#goal-of-the-exercise-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-1---install-and-load-libraries"><i class="fa fa-check"></i>Step 1 - Install and Load libraries</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-2---input-query"><i class="fa fa-check"></i>Step 2 - Input query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-3---run-gprofiler"><i class="fa fa-check"></i>Step 3 - Run g:Profiler</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-4---explore-top-table-of-results"><i class="fa fa-check"></i>Step 4 - Explore top table of results</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-5---filter-results-by-geneset-size"><i class="fa fa-check"></i>Step 5 - Filter results by geneset size</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-6---run-gprofiler-with-your-own-genesets-example-using-baderlab-genesets"><i class="fa fa-check"></i>Step 6 - Run g:profiler with your own genesets (example using BaderLab genesets)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-7---download-and-load-bader-lab-geneset-file"><i class="fa fa-check"></i>Step 7 - Download and load Bader lab geneset file</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-8---filter-bader-lab-geneset-file"><i class="fa fa-check"></i>Step 8 - Filter Bader lab geneset file</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-9---upload-the-gmt-files-to-gprofiler-and-rerun-gprofiler"><i class="fa fa-check"></i>Step 9 - Upload the gmt files to gprofiler and rerun g:Profiler</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#exercise-2---visualization"><i class="fa fa-check"></i>Exercise 2 - visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-2---dot-plot-with-segment"><i class="fa fa-check"></i>Step 2 - Dot plot with segment</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-3---ridge-plot"><i class="fa fa-check"></i>Step 3 - Ridge plot</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-4---network-visualization---enrichmentmap"><i class="fa fa-check"></i>Step 4 - Network visualization - EnrichmentMap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html"><i class="fa fa-check"></i>Module 2 lab - fGSEA</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#goal-of-the-exercise"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#data-1"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#background"><i class="fa fa-check"></i>Background</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#how-to-generate-a-rank-file."><i class="fa fa-check"></i>How to generate a rank file.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#start-the-exercise"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step1."><i class="fa fa-check"></i>Step1.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-2."><i class="fa fa-check"></i>Step 2.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-3."><i class="fa fa-check"></i>Step 3.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-4."><i class="fa fa-check"></i>Step 4.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-5."><i class="fa fa-check"></i>Step 5.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-6."><i class="fa fa-check"></i>Step 6.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-7"><i class="fa fa-check"></i>Step 7</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-8"><i class="fa fa-check"></i>Step 8</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#additional_information"><i class="fa fa-check"></i>Additional information</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#bonus---automation."><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gsea-lab" class="section level1 hasAnchor">
<h1>Module 2 lab - fGSEA<a href="gsea-lab.html#gsea-lab" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Presenter: Ruth Isserlin</p>
<div id="introduction-1" class="section level2 hasAnchor">
<h2>Introduction<a href="gsea-lab.html#introduction-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This practical lab contains one exercise. It uses <a href="https://github.com/alserglab/fgsea">fGSEA</a> to perform a gene-set enrichment analysis.</p>
</div>
<div id="goal-of-the-exercise" class="section level2 hasAnchor">
<h2>Goal of the exercise<a href="gsea-lab.html#goal-of-the-exercise" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Learn how to run fGSEA and explore the results.</p>
</div>
<div id="data-1" class="section level2 hasAnchor">
<h2>Data<a href="gsea-lab.html#data-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The data used in this exercise is gene expression (transcriptomics) obtained from high-throughput RNA sequencing of Pancreatic Ductal Adenocarcinoma samples (TCGA-PAAD).</p>
<p>This cohort has been previously stratified into many different set of subtypes <a href="https://pubmed.ncbi.nlm.nih.gov/36765128/">PMID:36765128</a> with the <a href="https://pubmed.ncbi.nlm.nih.gov/26343385/">Moffitt</a> Basal vs Classical subtypes compared to demonstrate the GSEA workflow.</p>
<div id="how-was-the-data-processed" class="section level4 hasAnchor">
<h4>How was the data processed?<a href="gsea-lab.html#how-was-the-data-processed" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Gene expression from the TCGA Pancreatic Ductal Adenocarcinoma RNASeq cohort was downloaded on 2024-06-06 from <a href="https://portal.gdc.cancer.gov/">Genomic Data Commons</a> using the <a href="https://bioconductor.org/packages/release/bioc/html/TCGAbiolinks.html">TCGABiolinks</a> R package.</li>
<li>Differential expression for all genes between the Basal and Classical groups was estimated using <a href="http://www.ncbi.nlm.nih.gov/pubmed/19910308">edgeR</a>.</li>
<li>The R code used to generate the data and the rank file used in GSEA is included at the bottom of the document in the <a href="gsea-lab.html#additional_information"><strong>Additional information</strong></a> section.</li>
</ul>
</div>
</div>
<div id="background" class="section level2 hasAnchor">
<h2>Background<a href="gsea-lab.html#background" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The goal of this lab is to:</p>
<ul>
<li>Upload the 2 required files into GSEA,</li>
<li>Adjust relevant parameters,</li>
<li>Run fGSEA,</li>
<li>Open and explore the gene-set enrichment results.</li>
</ul>
<p>The 2 required files are:</p>
<ol style="list-style-type: decimal">
<li>a rank file (.rnk)</li>
<li>a pathway definition file (.gmt).</li>
</ol>
<div id="rank-file" class="section level4 hasAnchor">
<h4>Rank File<a href="gsea-lab.html#rank-file" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To generate a rank file (.rnk), a score (-log10(pvalue) * sign(logFC)) was calculated from the edgeR differential expression results. A gene that is significantly differentially expressed (i.e associated with a very small pvalue, close to 0) will be assigned a high score.<br>The sign of the logFC indicates if the gene has an expression which is higher in Basel (logFC &gt; 0, the score will have a + sign) or lower in Classical (logFC &lt; 0, the score will have a - sign). It is used to rank the genes from top up-regulated to top down-regulated (<strong>all genes have to be included</strong>).</p>
<div class="rmd-caution">
<p>
The rank file is going to be provided for the lab, you don’t need to
generate it.
</p>
</div>
</div>
<div id="how-to-generate-a-rank-file." class="section level3 hasAnchor">
<h3>How to generate a rank file.<a href="gsea-lab.html#how-to-generate-a-rank-file." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="calculation-of-the-score" class="section level4 hasAnchor">
<h4>Calculation of the score<a href="gsea-lab.html#calculation-of-the-score" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="Module2/gsea/images/calculate_score.png" alt="rank_score" width="75%" /></p>
<p><img src="Module2/gsea/images/GSEA_KS.png" alt="GSEA_KS" width="100%" /></p>
</div>
<div id="generation-of-the-rank-file" class="section level4 hasAnchor">
<h4>Generation of the rank file<a href="gsea-lab.html#generation-of-the-rank-file" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Select the gene names and score columns and save the file as tab delimited with the extension .rnk</p>
<p><img src="Module2/gsea/images/make_rank_file.png" alt="generate rank" width="75%" /></p>
</div>
<div id="pathway-defintion-file" class="section level4 hasAnchor">
<h4>Pathway defintion file<a href="gsea-lab.html#pathway-defintion-file" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The second file that is needed for GSEA is the pathway database, a file with the .gmt extension. The pathway database (.gmt) used for the GSEA analysis was downloaded from <a href="http://baderlab.org/GeneSets" class="uri">http://baderlab.org/GeneSets</a>. This file contains gene-sets obtained from MsigDB-c2 and Hallmarks, NCI, Biocarta, IOB, Netpath, HumanCyc, Reactome, Panther, Pathbank, WikiPathways and the Gene Ontology (GO) databases.</p>
<div class="rmd-caution">
<p>
You don’t need to perform this step for the exercise, the .gmt file
will be given to you. <br>
</p>
</div>
<p>Go to:</p>
<ul>
<li><a href="http://download.baderlab.org/EM_Genesets/" class="uri">http://download.baderlab.org/EM_Genesets/</a></li>
<li>Click on May_01_2025/</li>
<li>Click on Human/</li>
<li>Click on symbol/</li>
<li>Save the Human_GOBP_AllPathways_noPFOCR_no_GO_iea…gmt file on your computer</li>
</ul>
<p><img src="Module2/gsea/images/saving_gmt.png" alt="saving_gmt" width="90%" /></p>
<p>The .gmt is a tab delimited text file which contains one gene-set per row. For each gene-set (row), the first 2 columns contain the name and the description of the gene-set and the remaining columns contain the list of genes included in the gene-set. It is possible to create a custom gene-set using Excel or R.</p>
<p><img src="Module2/gsea/images/gmt_format.png" alt="get_gmt" width="90%" /></p>
<p>GSEA performs a gene-set enrichment analysis using a modified Kolmogorov-Smirnov statistic. The output result consists of summary tables displaying enrichment statistics for each gene-set (pathway) that has been tested.</p>
</div>
</div>
<div id="start-the-exercise" class="section level3 hasAnchor">
<h3>Start the exercise<a href="gsea-lab.html#start-the-exercise" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before starting this exercise, download the 2 required files:</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select “Save Link As…”.
</p>
<p>
Place it in the corresponding module directory of your CBW work
directory.
</p>
</div>
<ul>
<li><a href="./Module2/gsea/data/Human_GOBP_AllPathways_noPFOCR_no_GO_iea_May_01_2025_symbol.gmt">Human_GOBP_AllPathways_noPFOCR_no_GO_iea_May_01_2025_symbol.gmt</a></li>
<li><a href="./Module2/gsea/data//TCGA-PAAD_GDC_Subtype_Moffitt_BasalvsClassical_ranks.rnk">TCGA-PAAD_GDC_Subtype_Moffitt_BasalvsClassical_ranks.rnk</a></li>
</ul>
</div>
<div id="step1." class="section level3 hasAnchor">
<h3>Step1.<a href="gsea-lab.html#step1." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Load required libraries.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="gsea-lab.html#cb44-1" tabindex="-1"></a><span class="co"># List of required packages</span></span>
<span id="cb44-2"><a href="gsea-lab.html#cb44-2" tabindex="-1"></a>required_packages <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb44-3"><a href="gsea-lab.html#cb44-3" tabindex="-1"></a>  <span class="st">&quot;tidyverse&quot;</span>,     <span class="co"># Data manipulation and visualization</span></span>
<span id="cb44-4"><a href="gsea-lab.html#cb44-4" tabindex="-1"></a>  <span class="st">&quot;GSA&quot;</span>,           <span class="co"># Gene set analysis</span></span>
<span id="cb44-5"><a href="gsea-lab.html#cb44-5" tabindex="-1"></a>  <span class="st">&quot;RCurl&quot;</span>,         <span class="co"># HTTP requests</span></span>
<span id="cb44-6"><a href="gsea-lab.html#cb44-6" tabindex="-1"></a>  <span class="st">&quot;fgsea&quot;</span>,         <span class="co"># fast GSEA</span></span>
<span id="cb44-7"><a href="gsea-lab.html#cb44-7" tabindex="-1"></a>  <span class="st">&quot;data.table&quot;</span>,    <span class="co"># for formatting the results</span></span>
<span id="cb44-8"><a href="gsea-lab.html#cb44-8" tabindex="-1"></a>  <span class="st">&quot;ggridges&quot;</span>,      <span class="co"># Ridgeline plots</span></span>
<span id="cb44-9"><a href="gsea-lab.html#cb44-9" tabindex="-1"></a>  <span class="st">&quot;clusterProfiler&quot;</span>, <span class="co"># for the gsea plots</span></span>
<span id="cb44-10"><a href="gsea-lab.html#cb44-10" tabindex="-1"></a>  <span class="st">&quot;enrichplot&quot;</span><span class="co"># for the gsea plots</span></span>
<span id="cb44-11"><a href="gsea-lab.html#cb44-11" tabindex="-1"></a>)</span>
<span id="cb44-12"><a href="gsea-lab.html#cb44-12" tabindex="-1"></a></span>
<span id="cb44-13"><a href="gsea-lab.html#cb44-13" tabindex="-1"></a><span class="co"># Install any missing packages</span></span>
<span id="cb44-14"><a href="gsea-lab.html#cb44-14" tabindex="-1"></a>installed <span class="ot">&lt;-</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>())</span>
<span id="cb44-15"><a href="gsea-lab.html#cb44-15" tabindex="-1"></a><span class="cf">for</span> (pkg <span class="cf">in</span> required_packages) {</span>
<span id="cb44-16"><a href="gsea-lab.html#cb44-16" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>(pkg <span class="sc">%in%</span> installed)) {</span>
<span id="cb44-17"><a href="gsea-lab.html#cb44-17" tabindex="-1"></a>    <span class="fu">install.packages</span>(pkg)</span>
<span id="cb44-18"><a href="gsea-lab.html#cb44-18" tabindex="-1"></a>  }</span>
<span id="cb44-19"><a href="gsea-lab.html#cb44-19" tabindex="-1"></a>}</span>
<span id="cb44-20"><a href="gsea-lab.html#cb44-20" tabindex="-1"></a></span>
<span id="cb44-21"><a href="gsea-lab.html#cb44-21" tabindex="-1"></a><span class="co"># Load all packages</span></span>
<span id="cb44-22"><a href="gsea-lab.html#cb44-22" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">lapply</span>(required_packages, library, <span class="at">character.only =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:lubridate&#39;:
## 
##     hour, isoweek, mday, minute, month, quarter, second, wday, week,
##     yday, year</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     between, first, last</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     transpose</code></pre>
<pre><code>## </code></pre>
<pre><code>## clusterProfiler v4.12.6 Learn more at https://yulab-smu.top/contribution-knowledge-mining/
## 
## Please cite:
## 
## T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,
## X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal
## enrichment tool for interpreting omics data. The Innovation. 2021,
## 2(3):100141</code></pre>
<pre><code>## 
## Attaching package: &#39;clusterProfiler&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:igraph&#39;:
## 
##     simplify</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     simplify</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     filter</code></pre>
</div>
<div id="step-2." class="section level3 hasAnchor">
<h3>Step 2.<a href="gsea-lab.html#step-2." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Set up parameters need for the analysis.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="gsea-lab.html#cb55-1" tabindex="-1"></a><span class="co"># defined in the paramters at top of notebook</span></span>
<span id="cb55-2"><a href="gsea-lab.html#cb55-2" tabindex="-1"></a></span>
<span id="cb55-3"><a href="gsea-lab.html#cb55-3" tabindex="-1"></a><span class="co">#directory where all the data files are found.  </span></span>
<span id="cb55-4"><a href="gsea-lab.html#cb55-4" tabindex="-1"></a>working_dir <span class="ot">&lt;-</span> <span class="st">&quot;./data/&quot;</span></span>
<span id="cb55-5"><a href="gsea-lab.html#cb55-5" tabindex="-1"></a></span>
<span id="cb55-6"><a href="gsea-lab.html#cb55-6" tabindex="-1"></a><span class="co">#directory where all the data files are found.  </span></span>
<span id="cb55-7"><a href="gsea-lab.html#cb55-7" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> <span class="st">&quot;./generated_data/&quot;</span></span>
<span id="cb55-8"><a href="gsea-lab.html#cb55-8" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">exists</span>(output_dir)){</span>
<span id="cb55-9"><a href="gsea-lab.html#cb55-9" tabindex="-1"></a>  <span class="fu">dir.create</span>(output_dir)</span>
<span id="cb55-10"><a href="gsea-lab.html#cb55-10" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## Warning in dir.create(output_dir): &#39;./generated_data&#39; already exists</code></pre>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="gsea-lab.html#cb57-1" tabindex="-1"></a><span class="co">#The name to give the analysis in GSEA - for example</span></span>
<span id="cb57-2"><a href="gsea-lab.html#cb57-2" tabindex="-1"></a>analysis_name <span class="ot">&lt;-</span> <span class="st">&quot;Basal_vs_Classical&quot;</span></span>
<span id="cb57-3"><a href="gsea-lab.html#cb57-3" tabindex="-1"></a></span>
<span id="cb57-4"><a href="gsea-lab.html#cb57-4" tabindex="-1"></a><span class="co">#rank file to use in GSEA analysis.  </span></span>
<span id="cb57-5"><a href="gsea-lab.html#cb57-5" tabindex="-1"></a>rnk_file <span class="ot">&lt;-</span> <span class="st">&quot;TCGA-PAAD_GDC_Subtype_Moffitt_BasalvsClassical_ranks.rnk&quot;</span></span>
<span id="cb57-6"><a href="gsea-lab.html#cb57-6" tabindex="-1"></a></span>
<span id="cb57-7"><a href="gsea-lab.html#cb57-7" tabindex="-1"></a><span class="co">#set the gmt file you want to use if you don&#39;t want to use the latest gmt file.</span></span>
<span id="cb57-8"><a href="gsea-lab.html#cb57-8" tabindex="-1"></a><span class="co"># For example, if you set dest_gmt_file ==&quot;&quot; the below script will automatically</span></span>
<span id="cb57-9"><a href="gsea-lab.html#cb57-9" tabindex="-1"></a><span class="co"># download the latest gmt file from baderlab webstie.  If it is set then it</span></span>
<span id="cb57-10"><a href="gsea-lab.html#cb57-10" tabindex="-1"></a><span class="co"># will use the file specified.  </span></span>
<span id="cb57-11"><a href="gsea-lab.html#cb57-11" tabindex="-1"></a>dest_gmt_file <span class="ot">=</span> <span class="st">&quot;Human_GOBP_AllPathways_noPFOCR_no_GO_iea_May_01_2025_symbol.gmt&quot;</span></span></code></pre></div>
</div>
<div id="step-3." class="section level3 hasAnchor">
<h3>Step 3.<a href="gsea-lab.html#step-3." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Download the latest pathway definition file</p>
<p>Only Human, Mouse, Rat, and Woodchuck gene set files are currently available on the baderlab downloads site. If you are working with a species other than human (and it is either rat,mouse or woodchuck) change the gmt_url below to the correct species. Check here to see all available species.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="gsea-lab.html#cb58-1" tabindex="-1"></a><span class="co">#flag to trigger download of latest gmt file - change to TRUE </span></span>
<span id="cb58-2"><a href="gsea-lab.html#cb58-2" tabindex="-1"></a><span class="co"># if you want to check the website and download the latest</span></span>
<span id="cb58-3"><a href="gsea-lab.html#cb58-3" tabindex="-1"></a><span class="co"># version instead of using the specified May 01, 2025 version. </span></span>
<span id="cb58-4"><a href="gsea-lab.html#cb58-4" tabindex="-1"></a>downlaod_latest_version <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb58-5"><a href="gsea-lab.html#cb58-5" tabindex="-1"></a></span>
<span id="cb58-6"><a href="gsea-lab.html#cb58-6" tabindex="-1"></a><span class="cf">if</span>((dest_gmt_file <span class="sc">==</span> <span class="st">&quot;&quot;</span>) <span class="sc">&amp;&amp;</span> download_latest_version){</span>
<span id="cb58-7"><a href="gsea-lab.html#cb58-7" tabindex="-1"></a>  gmt_url <span class="ot">=</span> <span class="st">&quot;http://download.baderlab.org/EM_Genesets/current_release/Human/symbol/&quot;</span></span>
<span id="cb58-8"><a href="gsea-lab.html#cb58-8" tabindex="-1"></a>  </span>
<span id="cb58-9"><a href="gsea-lab.html#cb58-9" tabindex="-1"></a>  <span class="co">#list all the files on the server</span></span>
<span id="cb58-10"><a href="gsea-lab.html#cb58-10" tabindex="-1"></a>  filenames <span class="ot">=</span> <span class="fu">getURL</span>(gmt_url)</span>
<span id="cb58-11"><a href="gsea-lab.html#cb58-11" tabindex="-1"></a>  tc <span class="ot">=</span> <span class="fu">textConnection</span>(filenames)</span>
<span id="cb58-12"><a href="gsea-lab.html#cb58-12" tabindex="-1"></a>  contents <span class="ot">=</span> <span class="fu">readLines</span>(tc)</span>
<span id="cb58-13"><a href="gsea-lab.html#cb58-13" tabindex="-1"></a>  <span class="fu">close</span>(tc)</span>
<span id="cb58-14"><a href="gsea-lab.html#cb58-14" tabindex="-1"></a>  </span>
<span id="cb58-15"><a href="gsea-lab.html#cb58-15" tabindex="-1"></a>  <span class="co">#get the gmt that has all the pathways and does not include terms </span></span>
<span id="cb58-16"><a href="gsea-lab.html#cb58-16" tabindex="-1"></a>  <span class="co"># inferred from electronic annotations(IEA)</span></span>
<span id="cb58-17"><a href="gsea-lab.html#cb58-17" tabindex="-1"></a>  <span class="co">#start with gmt file that has pathways only and GO Biological Process only.</span></span>
<span id="cb58-18"><a href="gsea-lab.html#cb58-18" tabindex="-1"></a>  rx <span class="ot">=</span> <span class="fu">gregexpr</span>(<span class="st">&quot;(?&lt;=&lt;a href=</span><span class="sc">\&quot;</span><span class="st">)(.*.GOBP_AllPathways_noPFOCR_no_GO_iea.*.)(.gmt)(?=</span><span class="sc">\&quot;</span><span class="st">&gt;)&quot;</span>,</span>
<span id="cb58-19"><a href="gsea-lab.html#cb58-19" tabindex="-1"></a>    contents, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb58-20"><a href="gsea-lab.html#cb58-20" tabindex="-1"></a>  gmt_file <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">regmatches</span>(contents, rx))</span>
<span id="cb58-21"><a href="gsea-lab.html#cb58-21" tabindex="-1"></a>  </span>
<span id="cb58-22"><a href="gsea-lab.html#cb58-22" tabindex="-1"></a>  dest_gmt_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir,gmt_file )</span>
<span id="cb58-23"><a href="gsea-lab.html#cb58-23" tabindex="-1"></a>  </span>
<span id="cb58-24"><a href="gsea-lab.html#cb58-24" tabindex="-1"></a>  <span class="co">#check if this gmt file already exists</span></span>
<span id="cb58-25"><a href="gsea-lab.html#cb58-25" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(dest_gmt_file)){</span>
<span id="cb58-26"><a href="gsea-lab.html#cb58-26" tabindex="-1"></a>    <span class="fu">download.file</span>(</span>
<span id="cb58-27"><a href="gsea-lab.html#cb58-27" tabindex="-1"></a>      <span class="fu">paste</span>(gmt_url,gmt_file,<span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb58-28"><a href="gsea-lab.html#cb58-28" tabindex="-1"></a>      <span class="at">destfile=</span>dest_gmt_file</span>
<span id="cb58-29"><a href="gsea-lab.html#cb58-29" tabindex="-1"></a>    )</span>
<span id="cb58-30"><a href="gsea-lab.html#cb58-30" tabindex="-1"></a>  }</span>
<span id="cb58-31"><a href="gsea-lab.html#cb58-31" tabindex="-1"></a>}<span class="cf">else</span> <span class="cf">if</span>(<span class="sc">!</span>downlaod_latest_version){</span>
<span id="cb58-32"><a href="gsea-lab.html#cb58-32" tabindex="-1"></a>  </span>
<span id="cb58-33"><a href="gsea-lab.html#cb58-33" tabindex="-1"></a>  gmt_url <span class="ot">=</span> <span class="st">&quot;http://download.baderlab.org/EM_Genesets/May_01_2025/Human/symbol/&quot;</span></span>
<span id="cb58-34"><a href="gsea-lab.html#cb58-34" tabindex="-1"></a> gmt_file <span class="ot">=</span> <span class="st">&quot;Human_GOBP_AllPathways_noPFOCR_no_GO_iea_May_01_2025_symbol.gmt&quot;</span></span>
<span id="cb58-35"><a href="gsea-lab.html#cb58-35" tabindex="-1"></a>  </span>
<span id="cb58-36"><a href="gsea-lab.html#cb58-36" tabindex="-1"></a>  <span class="co">#download the specific file specified above</span></span>
<span id="cb58-37"><a href="gsea-lab.html#cb58-37" tabindex="-1"></a>  dest_gmt_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(working_dir,gmt_file)</span>
<span id="cb58-38"><a href="gsea-lab.html#cb58-38" tabindex="-1"></a>  </span>
<span id="cb58-39"><a href="gsea-lab.html#cb58-39" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(dest_gmt_file)){</span>
<span id="cb58-40"><a href="gsea-lab.html#cb58-40" tabindex="-1"></a>    <span class="fu">download.file</span>(</span>
<span id="cb58-41"><a href="gsea-lab.html#cb58-41" tabindex="-1"></a>      <span class="fu">paste</span>(gmt_url,gmt_file,<span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb58-42"><a href="gsea-lab.html#cb58-42" tabindex="-1"></a>      <span class="at">destfile=</span>dest_gmt_file</span>
<span id="cb58-43"><a href="gsea-lab.html#cb58-43" tabindex="-1"></a>    )</span>
<span id="cb58-44"><a href="gsea-lab.html#cb58-44" tabindex="-1"></a>  } </span>
<span id="cb58-45"><a href="gsea-lab.html#cb58-45" tabindex="-1"></a>  <span class="fu">file.copy</span>(dest_gmt_file,<span class="at">to =</span> output_dir)</span>
<span id="cb58-46"><a href="gsea-lab.html#cb58-46" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p>Load your gmt file</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="gsea-lab.html#cb60-1" tabindex="-1"></a><span class="fu">capture.output</span>(all_gs <span class="ot">&lt;-</span> <span class="fu">GSA.read.gmt</span>(dest_gmt_file) ,<span class="at">file=</span><span class="st">&quot;gsa_load.out&quot;</span>)</span>
<span id="cb60-2"><a href="gsea-lab.html#cb60-2" tabindex="-1"></a><span class="fu">names</span>(all_gs<span class="sc">$</span>genesets) <span class="ot">&lt;-</span> all_gs<span class="sc">$</span>geneset.names</span></code></pre></div>
</div>
<div id="step-4." class="section level3 hasAnchor">
<h3>Step 4.<a href="gsea-lab.html#step-4." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Load your Rank file</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="gsea-lab.html#cb61-1" tabindex="-1"></a>current_ranks <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">file.path</span>(working_dir,rnk_file),<span class="at">header=</span><span class="cn">TRUE</span>,<span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb61-2"><a href="gsea-lab.html#cb61-2" tabindex="-1"></a>    fgsea_ranks <span class="ot">&lt;-</span> current_ranks[,<span class="dv">2</span>]</span>
<span id="cb61-3"><a href="gsea-lab.html#cb61-3" tabindex="-1"></a>    <span class="fu">names</span>(fgsea_ranks) <span class="ot">&lt;-</span> current_ranks[,<span class="dv">1</span>]</span>
<span id="cb61-4"><a href="gsea-lab.html#cb61-4" tabindex="-1"></a> </span>
<span id="cb61-5"><a href="gsea-lab.html#cb61-5" tabindex="-1"></a>current_ranks <span class="ot">&lt;-</span> fgsea_ranks</span>
<span id="cb61-6"><a href="gsea-lab.html#cb61-6" tabindex="-1"></a> </span>
<span id="cb61-7"><a href="gsea-lab.html#cb61-7" tabindex="-1"></a><span class="co">#remove duplicated genes</span></span>
<span id="cb61-8"><a href="gsea-lab.html#cb61-8" tabindex="-1"></a>duplicated_gene_names <span class="ot">&lt;-</span>   </span>
<span id="cb61-9"><a href="gsea-lab.html#cb61-9" tabindex="-1"></a>      <span class="fu">names</span>(current_ranks)[<span class="fu">which</span>(<span class="fu">duplicated</span>(<span class="fu">names</span>(current_ranks)))]</span>
<span id="cb61-10"><a href="gsea-lab.html#cb61-10" tabindex="-1"></a>    current_ranks <span class="ot">&lt;-</span> current_ranks[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">names</span>(current_ranks) <span class="sc">%in%</span> </span>
<span id="cb61-11"><a href="gsea-lab.html#cb61-11" tabindex="-1"></a>                                           duplicated_gene_names)]</span>
<span id="cb61-12"><a href="gsea-lab.html#cb61-12" tabindex="-1"></a></span>
<span id="cb61-13"><a href="gsea-lab.html#cb61-13" tabindex="-1"></a>current_ranks <span class="ot">&lt;-</span> <span class="fu">sort</span>(current_ranks,<span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="step-5." class="section level3 hasAnchor">
<h3>Step 5.<a href="gsea-lab.html#step-5." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Run fGSEA - Adjust parameters</p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/fgsea.htm">fGSEA</a> is an R package that runs a fast Gene Set Enrichment Analysis.</p>
<p>In the below command the following options have been specified:</p>
<ul>
<li>pathways - list of genesets to use for the calculation</li>
<li>stats - genes and their associated statistic, sorted</li>
<li>max_size - maximum size for individual gene sets. In GSEA interface this is set to 500 but we prefer to use a more stringent setting of 200.</li>
<li>min_size - minimum size for individual gene sets</li>
<li>gseaParam - GSEA parameter value</li>
</ul>
<div class="rmd-caution">
<p>
gseaParam - GSEA parameter value. All ranks will be raised to this
power when calculating the ES scores. Equivalent to the weight in the
original GSEA algorithm.
</p>
</div>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="gsea-lab.html#cb62-1" tabindex="-1"></a>current_fgsea_results <span class="ot">&lt;-</span> fgsea<span class="sc">::</span><span class="fu">fgsea</span>(all_gs<span class="sc">$</span>genesets, </span>
<span id="cb62-2"><a href="gsea-lab.html#cb62-2" tabindex="-1"></a>                                      current_ranks,</span>
<span id="cb62-3"><a href="gsea-lab.html#cb62-3" tabindex="-1"></a>                                      <span class="at">minSize=</span><span class="dv">15</span>,</span>
<span id="cb62-4"><a href="gsea-lab.html#cb62-4" tabindex="-1"></a>                                      <span class="at">maxSize =</span> <span class="dv">500</span></span>
<span id="cb62-5"><a href="gsea-lab.html#cb62-5" tabindex="-1"></a>                                      )</span></code></pre></div>
<div class="rmd-tip">
<p>
Set <strong>gseaParam</strong> to 2 if you want to add more weight on
the most top up-regulated and top down-regulated. <br>
<strong>2</strong> is a more stringent parameter and it will result in
less gene-sets significant under FDR &lt;0.05.
</p>
</div>
</div>
<div id="step-6." class="section level3 hasAnchor">
<h3>Step 6.<a href="gsea-lab.html#step-6." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Examining the results</p>
<p>Get the top results and visualize as table</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="gsea-lab.html#cb63-1" tabindex="-1"></a>topPathwaysUp <span class="ot">&lt;-</span> current_fgsea_results[ES <span class="sc">&gt;</span> <span class="dv">0</span>][<span class="fu">head</span>(<span class="fu">order</span>(pval), <span class="at">n=</span><span class="dv">10</span>), pathway]</span>
<span id="cb63-2"><a href="gsea-lab.html#cb63-2" tabindex="-1"></a>topPathwaysDown <span class="ot">&lt;-</span> current_fgsea_results[ES <span class="sc">&lt;</span> <span class="dv">0</span>][<span class="fu">head</span>(<span class="fu">order</span>(pval), <span class="at">n=</span><span class="dv">10</span>), pathway]</span>
<span id="cb63-3"><a href="gsea-lab.html#cb63-3" tabindex="-1"></a>topPathways <span class="ot">&lt;-</span> <span class="fu">c</span>(topPathwaysUp, <span class="fu">rev</span>(topPathwaysDown))</span>
<span id="cb63-4"><a href="gsea-lab.html#cb63-4" tabindex="-1"></a><span class="fu">plotGseaTable</span>(all_gs<span class="sc">$</span>genesets[topPathways], current_ranks, current_fgsea_results, </span>
<span id="cb63-5"><a href="gsea-lab.html#cb63-5" tabindex="-1"></a>              <span class="at">gseaParam=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="Compute-Ontario---Bioinformatics-for-Pathway-Enrichment-Analysis_files/figure-html/top_pathways-1.png" width="672" /></p>
<p>When examining the results there are a few things to look for:</p>
<p>Check the number of gene-sets that have been used for the analysis.</p>
<div class="rmd-tip">
<p>
A small number (a few hundred genesets if using baderlab genesets)
could indicate an issue with identifier mapping.
</p>
</div>
<p>Check the number of sets that have FDR less than 0.25 – in order to determine what thresholds to start with when creating the enrichment map. It is not uncommon to see a thousand gene sets pass the threshold of FDR less than 0.25. FDR less than 0.25 is a very lax threshold and for robust data we can set thresholds of FDR less than 0.05 or lower.</p>
<p>Number of pathways with corrected pvalue &lt; 0.05</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="gsea-lab.html#cb64-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span></code></pre></div>
<pre><code>## [1] 482</code></pre>
<p>Number of pathways with corrected pvalue &lt; 0.01</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="gsea-lab.html#cb66-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.01</span>))</span></code></pre></div>
<pre><code>## [1] 216</code></pre>
<p>Number of Basal (up -regulated) pathways with corrected pvalue &lt; 0.05</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="gsea-lab.html#cb68-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> current_fgsea_results<span class="sc">$</span>ES <span class="sc">&gt;</span> <span class="dv">0</span>))</span></code></pre></div>
<pre><code>## [1] 385</code></pre>
<p>Number of Classical (down -regulated) pathways with corrected pvalue &lt; 0.05</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="gsea-lab.html#cb70-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> current_fgsea_results<span class="sc">$</span>ES <span class="sc">&lt;</span> <span class="dv">0</span>))</span></code></pre></div>
<pre><code>## [1] 97</code></pre>
<p>5e. Explore the tabular format of the results.</p>
<div id="basal" class="section level4 hasAnchor">
<h4>Basal<a href="gsea-lab.html#basal" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="gsea-lab.html#cb72-1" tabindex="-1"></a>topPathwaysUp <span class="ot">&lt;-</span> current_fgsea_results[ES <span class="sc">&gt;</span> <span class="dv">0</span>][<span class="fu">head</span>(<span class="fu">order</span>(pval), <span class="at">n=</span><span class="dv">5</span>), pathway]</span>
<span id="cb72-2"><a href="gsea-lab.html#cb72-2" tabindex="-1"></a></span>
<span id="cb72-3"><a href="gsea-lab.html#cb72-3" tabindex="-1"></a>top_basal_hits <span class="ot">&lt;-</span> current_fgsea_results[<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>pathway <span class="sc">%in%</span> topPathwaysUp),]</span>
<span id="cb72-4"><a href="gsea-lab.html#cb72-4" tabindex="-1"></a></span>
<span id="cb72-5"><a href="gsea-lab.html#cb72-5" tabindex="-1"></a><span class="co">#format the pathway name colum so that it is easier to see the whole table</span></span>
<span id="cb72-6"><a href="gsea-lab.html#cb72-6" tabindex="-1"></a>top_basal_hits<span class="sc">$</span>pathway <span class="ot">&lt;-</span> <span class="fu">substr</span>(top_basal_hits<span class="sc">$</span>pathway, <span class="at">start =</span> <span class="dv">1</span>, <span class="at">stop=</span><span class="dv">25</span>)</span>
<span id="cb72-7"><a href="gsea-lab.html#cb72-7" tabindex="-1"></a></span>
<span id="cb72-8"><a href="gsea-lab.html#cb72-8" tabindex="-1"></a>top_basal_hits</span></code></pre></div>
<pre><code>##                      pathway         pval         padj   log2err        ES
##                       &lt;char&gt;        &lt;num&gt;        &lt;num&gt;     &lt;num&gt;     &lt;num&gt;
## 1: CELL CYCLE CHECKPOINTS%RE 3.832152e-10 3.992464e-07 0.8140358 0.5400523
## 2: CELL CYCLE, MITOTIC%REACT 1.056162e-16 4.951468e-13 1.0574636 0.5211659
## 3: HALLMARK_EPITHELIAL_MESEN 1.584216e-16 4.951468e-13 1.0476265 0.6962643
## 4: KERATINIZATION%REACTOME%R 6.764172e-10 5.285355e-07 0.8012156 0.8141973
## 5: SKIN DEVELOPMENT%GOBP%GO: 2.840735e-10 3.551487e-07 0.8140358 0.7078168
##         NES  size  leadingEdge
##       &lt;num&gt; &lt;int&gt;       &lt;list&gt;
## 1: 2.014014   243 PSMD2, M....
## 2: 2.056739   469 PSMD2, M....
## 3: 2.476533   156 SNAI2, T....
## 4: 2.378305    47 KRT6A, K....
## 5: 2.285129    87 KRT6A, A....</code></pre>
</div>
<div id="classical" class="section level4 hasAnchor">
<h4>Classical<a href="gsea-lab.html#classical" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="gsea-lab.html#cb74-1" tabindex="-1"></a>topPathwaysDown <span class="ot">&lt;-</span> current_fgsea_results[ES <span class="sc">&lt;</span> <span class="dv">0</span>][<span class="fu">head</span>(<span class="fu">order</span>(pval), <span class="at">n=</span><span class="dv">5</span>), pathway]</span>
<span id="cb74-2"><a href="gsea-lab.html#cb74-2" tabindex="-1"></a></span>
<span id="cb74-3"><a href="gsea-lab.html#cb74-3" tabindex="-1"></a>top_classical_hits <span class="ot">&lt;-</span> current_fgsea_results[<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>pathway <span class="sc">%in%</span> topPathwaysDown),]</span>
<span id="cb74-4"><a href="gsea-lab.html#cb74-4" tabindex="-1"></a></span>
<span id="cb74-5"><a href="gsea-lab.html#cb74-5" tabindex="-1"></a><span class="co">#format the pathway name colum so that it is easier to see the whole table</span></span>
<span id="cb74-6"><a href="gsea-lab.html#cb74-6" tabindex="-1"></a>top_classical_hits<span class="sc">$</span>pathway <span class="ot">&lt;-</span> <span class="fu">substr</span>(top_classical_hits<span class="sc">$</span>pathway, <span class="at">start =</span> <span class="dv">1</span>, <span class="at">stop=</span><span class="dv">25</span>)</span>
<span id="cb74-7"><a href="gsea-lab.html#cb74-7" tabindex="-1"></a></span>
<span id="cb74-8"><a href="gsea-lab.html#cb74-8" tabindex="-1"></a>top_classical_hits</span></code></pre></div>
<pre><code>##                      pathway         pval         padj   log2err         ES
##                       &lt;char&gt;        &lt;num&gt;        &lt;num&gt;     &lt;num&gt;      &lt;num&gt;
## 1: BIOLOGICAL OXIDATIONS%REA 5.538639e-10 4.946005e-07 0.8012156 -0.6362964
## 2: CARBOXYLIC ACID METABOLIC 1.375623e-08 5.374387e-06 0.7477397 -0.4563033
## 3: CELLULAR RESPONSE TO XENO 1.067081e-10 1.667580e-07 0.8390889 -0.7080315
## 4: OXIDATIVE PHOSPHORYLATION 4.664829e-08 1.380360e-05 0.7195128 -0.6498294
## 5: XENOBIOTIC METABOLIC PROC 4.506163e-11 9.389341e-08 0.8513391 -0.7533267
##          NES  size  leadingEdge
##        &lt;num&gt; &lt;int&gt;       &lt;list&gt;
## 1: -2.094249   148 CYP2S1, ....
## 2: -1.682257   490 ALDOB, C....
## 3: -2.189054    97 FMO5, HN....
## 4: -2.002848    95 COA6, AT....
## 5: -2.257171    78 FMO5, HN....</code></pre>
</div>
</div>
<div id="step-7" class="section level3 hasAnchor">
<h3>Step 7<a href="gsea-lab.html#step-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Visualize the top results as different sorts of plots, as demonstrated in the first part of the lab.</p>
<div id="gsea-style-enrichment-plots" class="section level4 hasAnchor">
<h4>GSEA style Enrichment Plots<a href="gsea-lab.html#gsea-style-enrichment-plots" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In order to create GSEA Enrichment plots we need to use the package:
* <a href="https://bioconductor.org/packages/devel/bioc/html/clusterProfiler.html">ClusterProfiler</a> and
* <a href="https://bioconductor.org/packages/devel/bioc/html/enrichplot.html">enrichplot</a></p>
<p>Clusterprofiler runs with its own set of pathways that it parses from GO or KEGG or MSigDB or other sources that are coded into the package. It does not take as input the standard GMT file. If you want to use your own pathway definitions you need to translate it into a dataframe where for every gene in a given pathway there is a row with the pathway name and the gene.</p>
<p>In order to generate an enrichment plot with enrichplot we need the fgsea results in the datastructure returned by Clusterprofiler. To run fGSEA in this way we need:
* the ranks file, as before
* the pathways as a data frame.</p>
<p>Below we show how to run fGSEA using the ClusterProfiler package and user supplied pathway definitions.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="gsea-lab.html#cb76-1" tabindex="-1"></a><span class="co">#translate our all_gs gmt file into a dataframe.</span></span>
<span id="cb76-2"><a href="gsea-lab.html#cb76-2" tabindex="-1"></a>TERMS <span class="ot">=</span> <span class="fu">cbind.data.frame</span>(</span>
<span id="cb76-3"><a href="gsea-lab.html#cb76-3" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">rep</span>( <span class="fu">names</span>(all_gs<span class="sc">$</span>genesets), <span class="fu">unlist</span>(<span class="fu">lapply</span>(all_gs<span class="sc">$</span>genesets, <span class="at">FUN=</span><span class="cf">function</span>(x){<span class="fu">length</span>(x)})) ),</span>
<span id="cb76-4"><a href="gsea-lab.html#cb76-4" tabindex="-1"></a>    <span class="at">gene =</span> <span class="fu">unlist</span>(all_gs<span class="sc">$</span>genesets)</span>
<span id="cb76-5"><a href="gsea-lab.html#cb76-5" tabindex="-1"></a>  )</span>
<span id="cb76-6"><a href="gsea-lab.html#cb76-6" tabindex="-1"></a></span>
<span id="cb76-7"><a href="gsea-lab.html#cb76-7" tabindex="-1"></a><span class="co"># specify the gene ranks, and make sure it is in descending order. </span></span>
<span id="cb76-8"><a href="gsea-lab.html#cb76-8" tabindex="-1"></a>  gsea_scores <span class="ot">&lt;-</span> current_ranks</span>
<span id="cb76-9"><a href="gsea-lab.html#cb76-9" tabindex="-1"></a>  <span class="co">#names(gsea_scores) &lt;-  ranks</span></span>
<span id="cb76-10"><a href="gsea-lab.html#cb76-10" tabindex="-1"></a>  gsea_scores <span class="ot">&lt;-</span> <span class="fu">sort</span>(gsea_scores, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb76-11"><a href="gsea-lab.html#cb76-11" tabindex="-1"></a>  gsea_scores <span class="ot">&lt;-</span> gsea_scores[<span class="sc">!</span><span class="fu">is.na</span>(gsea_scores)]</span>
<span id="cb76-12"><a href="gsea-lab.html#cb76-12" tabindex="-1"></a>  <span class="co"># Set a seed for reproducibility (optional)</span></span>
<span id="cb76-13"><a href="gsea-lab.html#cb76-13" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb76-14"><a href="gsea-lab.html#cb76-14" tabindex="-1"></a></span>
<span id="cb76-15"><a href="gsea-lab.html#cb76-15" tabindex="-1"></a>  <span class="co">#Run fgsea using clusterProfiler. </span></span>
<span id="cb76-16"><a href="gsea-lab.html#cb76-16" tabindex="-1"></a>  selected_gsea <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">GSEA</span>(</span>
<span id="cb76-17"><a href="gsea-lab.html#cb76-17" tabindex="-1"></a>    <span class="at">geneList =</span> gsea_scores,</span>
<span id="cb76-18"><a href="gsea-lab.html#cb76-18" tabindex="-1"></a>    <span class="at">TERM2GENE =</span> TERMS,</span>
<span id="cb76-19"><a href="gsea-lab.html#cb76-19" tabindex="-1"></a>    <span class="at">minGSSize =</span> <span class="dv">15</span>,</span>
<span id="cb76-20"><a href="gsea-lab.html#cb76-20" tabindex="-1"></a>    <span class="at">maxGSSize =</span> <span class="dv">500</span>,</span>
<span id="cb76-21"><a href="gsea-lab.html#cb76-21" tabindex="-1"></a>    <span class="at">pvalueCutoff =</span> <span class="dv">10</span>,</span>
<span id="cb76-22"><a href="gsea-lab.html#cb76-22" tabindex="-1"></a>    <span class="at">verbose =</span> T,</span>
<span id="cb76-23"><a href="gsea-lab.html#cb76-23" tabindex="-1"></a>    <span class="at">pAdjustMethod =</span> <span class="st">&quot;fdr&quot;</span>,</span>
<span id="cb76-24"><a href="gsea-lab.html#cb76-24" tabindex="-1"></a>    <span class="at">scoreType =</span> <span class="st">&#39;std&#39;</span>,</span>
<span id="cb76-25"><a href="gsea-lab.html#cb76-25" tabindex="-1"></a>    <span class="at">eps =</span> <span class="dv">0</span>,</span>
<span id="cb76-26"><a href="gsea-lab.html#cb76-26" tabindex="-1"></a>    <span class="co">#seed = myseed</span></span>
<span id="cb76-27"><a href="gsea-lab.html#cb76-27" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>## using &#39;fgsea&#39; for GSEA analysis, please cite Korotkevich et al (2019).</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<p>Print out a few of the top enrichment plots</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="gsea-lab.html#cb82-1" tabindex="-1"></a>all_figures <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb82-2"><a href="gsea-lab.html#cb82-2" tabindex="-1"></a></span>
<span id="cb82-3"><a href="gsea-lab.html#cb82-3" tabindex="-1"></a><span class="co">#gerenrate GSEA plots for a few of the Top Up genesets that we have already seen in the initial analysis</span></span>
<span id="cb82-4"><a href="gsea-lab.html#cb82-4" tabindex="-1"></a>rows_to_use <span class="ot">&lt;-</span> <span class="fu">which</span>(selected_gsea<span class="sc">@</span>result<span class="sc">$</span>ID <span class="sc">%in%</span> topPathwaysDown)</span>
<span id="cb82-5"><a href="gsea-lab.html#cb82-5" tabindex="-1"></a></span>
<span id="cb82-6"><a href="gsea-lab.html#cb82-6" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(rows_to_use)){</span>
<span id="cb82-7"><a href="gsea-lab.html#cb82-7" tabindex="-1"></a>  row_number <span class="ot">=</span> rows_to_use[i]</span>
<span id="cb82-8"><a href="gsea-lab.html#cb82-8" tabindex="-1"></a>  </span>
<span id="cb82-9"><a href="gsea-lab.html#cb82-9" tabindex="-1"></a>  <span class="co">#format the title for the enrichment plot</span></span>
<span id="cb82-10"><a href="gsea-lab.html#cb82-10" tabindex="-1"></a>  pw_des <span class="ot">&lt;-</span> selected_gsea<span class="sc">@</span>result<span class="sc">$</span>Description[row_number]</span>
<span id="cb82-11"><a href="gsea-lab.html#cb82-11" tabindex="-1"></a>  pw_pval <span class="ot">&lt;-</span> selected_gsea<span class="sc">@</span>result<span class="sc">$</span>pvalue[row_number]</span>
<span id="cb82-12"><a href="gsea-lab.html#cb82-12" tabindex="-1"></a>  pw_pval <span class="ot">&lt;-</span> <span class="fu">format</span>(pw_pval, <span class="at">scientific =</span> F, <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb82-13"><a href="gsea-lab.html#cb82-13" tabindex="-1"></a>  pw_qval <span class="ot">&lt;-</span> selected_gsea<span class="sc">@</span>result<span class="sc">$</span>p.adjust[row_number]</span>
<span id="cb82-14"><a href="gsea-lab.html#cb82-14" tabindex="-1"></a>  pw_qval <span class="ot">&lt;-</span> <span class="fu">format</span>(pw_qval, <span class="at">scientific =</span> F, <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb82-15"><a href="gsea-lab.html#cb82-15" tabindex="-1"></a>  pw_nes <span class="ot">&lt;-</span> selected_gsea<span class="sc">@</span>result<span class="sc">$</span>NES[row_number]</span>
<span id="cb82-16"><a href="gsea-lab.html#cb82-16" tabindex="-1"></a>  pw_nes <span class="ot">&lt;-</span> <span class="fu">format</span>(pw_nes, <span class="at">scientific =</span> F, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb82-17"><a href="gsea-lab.html#cb82-17" tabindex="-1"></a>  pw_es <span class="ot">&lt;-</span> selected_gsea<span class="sc">@</span>result<span class="sc">$</span>enrichmentScore[row_number]</span>
<span id="cb82-18"><a href="gsea-lab.html#cb82-18" tabindex="-1"></a>  pw_es <span class="ot">&lt;-</span> <span class="fu">format</span>(pw_es, <span class="at">scientific =</span> F, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb82-19"><a href="gsea-lab.html#cb82-19" tabindex="-1"></a>  mytitle <span class="ot">&lt;-</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">&#39;{pw_des}</span><span class="sc">\n</span><span class="st">(pval: {pw_pval}, qval: {pw_qval}, NES: {pw_nes}, ES: {pw_es})&#39;</span>)</span>
<span id="cb82-20"><a href="gsea-lab.html#cb82-20" tabindex="-1"></a>  </span>
<span id="cb82-21"><a href="gsea-lab.html#cb82-21" tabindex="-1"></a></span>
<span id="cb82-22"><a href="gsea-lab.html#cb82-22" tabindex="-1"></a>  figure <span class="ot">&lt;-</span> enrichplot<span class="sc">::</span><span class="fu">gseaplot2</span>(</span>
<span id="cb82-23"><a href="gsea-lab.html#cb82-23" tabindex="-1"></a>    selected_gsea,</span>
<span id="cb82-24"><a href="gsea-lab.html#cb82-24" tabindex="-1"></a>    <span class="at">geneSetID =</span>  <span class="dv">1</span>,</span>
<span id="cb82-25"><a href="gsea-lab.html#cb82-25" tabindex="-1"></a>    <span class="at">title =</span> mytitle,</span>
<span id="cb82-26"><a href="gsea-lab.html#cb82-26" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;green2&quot;</span>,</span>
<span id="cb82-27"><a href="gsea-lab.html#cb82-27" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">12</span>,</span>
<span id="cb82-28"><a href="gsea-lab.html#cb82-28" tabindex="-1"></a>    <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="fl">0.5</span>, <span class="dv">1</span>),</span>
<span id="cb82-29"><a href="gsea-lab.html#cb82-29" tabindex="-1"></a>    <span class="at">subplots =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,</span>
<span id="cb82-30"><a href="gsea-lab.html#cb82-30" tabindex="-1"></a>    <span class="at">pvalue_table =</span> F,</span>
<span id="cb82-31"><a href="gsea-lab.html#cb82-31" tabindex="-1"></a>    <span class="at">ES_geom =</span> <span class="st">&quot;line&quot;</span></span>
<span id="cb82-32"><a href="gsea-lab.html#cb82-32" tabindex="-1"></a>  )</span>
<span id="cb82-33"><a href="gsea-lab.html#cb82-33" tabindex="-1"></a>  </span>
<span id="cb82-34"><a href="gsea-lab.html#cb82-34" tabindex="-1"></a>  all_figures[[i]] <span class="ot">&lt;-</span> figure </span>
<span id="cb82-35"><a href="gsea-lab.html#cb82-35" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="gsea-lab.html#cb83-1" tabindex="-1"></a>  <span class="fu">capture.output</span>(<span class="fu">print</span>(all_figures[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]),<span class="at">file =</span> <span class="st">&quot;figure_outputs.txt&quot;</span>)</span></code></pre></div>
<p><img src="Compute-Ontario---Bioinformatics-for-Pathway-Enrichment-Analysis_files/figure-html/unnamed-chunk-36-1.png" width="672" /><img src="Compute-Ontario---Bioinformatics-for-Pathway-Enrichment-Analysis_files/figure-html/unnamed-chunk-36-2.png" width="672" /><img src="Compute-Ontario---Bioinformatics-for-Pathway-Enrichment-Analysis_files/figure-html/unnamed-chunk-36-3.png" width="672" /></p>
</div>
<div id="bar-plots" class="section level4 hasAnchor">
<h4>Bar plots<a href="gsea-lab.html#bar-plots" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>top pathways associated with classical</li>
</ul>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="gsea-lab.html#cb84-1" tabindex="-1"></a>topPathwaysDown <span class="ot">&lt;-</span> current_fgsea_results[ES <span class="sc">&lt;</span> <span class="dv">0</span>][<span class="fu">head</span>(<span class="fu">order</span>(pval), <span class="at">n=</span><span class="dv">20</span>), pathway]</span>
<span id="cb84-2"><a href="gsea-lab.html#cb84-2" tabindex="-1"></a></span>
<span id="cb84-3"><a href="gsea-lab.html#cb84-3" tabindex="-1"></a>enrichment_results_topdown <span class="ot">&lt;-</span>current_fgsea_results[<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>pathway <span class="sc">%in%</span> topPathwaysDown),]</span>
<span id="cb84-4"><a href="gsea-lab.html#cb84-4" tabindex="-1"></a></span>
<span id="cb84-5"><a href="gsea-lab.html#cb84-5" tabindex="-1"></a><span class="co">#change the name to be just the first part of the name</span></span>
<span id="cb84-6"><a href="gsea-lab.html#cb84-6" tabindex="-1"></a>enrichment_results_topdown<span class="sc">$</span>pathway <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(enrichment_results_topdown<span class="sc">$</span>pathway, <span class="at">FUN=</span><span class="cf">function</span>(x){<span class="fu">unlist</span>(<span class="fu">strsplit</span>(x,<span class="at">split =</span> <span class="st">&quot;%&quot;</span>))[<span class="dv">1</span>]}))</span>
<span id="cb84-7"><a href="gsea-lab.html#cb84-7" tabindex="-1"></a></span>
<span id="cb84-8"><a href="gsea-lab.html#cb84-8" tabindex="-1"></a><span class="do">##Calculate a score</span></span>
<span id="cb84-9"><a href="gsea-lab.html#cb84-9" tabindex="-1"></a>enrichment_results_topdown<span class="sc">$</span>score <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">as.numeric</span>(enrichment_results_topdown<span class="sc">$</span>pval))</span>
<span id="cb84-10"><a href="gsea-lab.html#cb84-10" tabindex="-1"></a></span>
<span id="cb84-11"><a href="gsea-lab.html#cb84-11" tabindex="-1"></a>enrichmentTidy <span class="ot">&lt;-</span> enrichment_results_topdown <span class="sc">%&gt;%</span></span>
<span id="cb84-12"><a href="gsea-lab.html#cb84-12" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb84-13"><a href="gsea-lab.html#cb84-13" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(score)) <span class="sc">%&gt;%</span></span>
<span id="cb84-14"><a href="gsea-lab.html#cb84-14" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb84-15"><a href="gsea-lab.html#cb84-15" tabindex="-1"></a></span>
<span id="cb84-16"><a href="gsea-lab.html#cb84-16" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggplot</span>(enrichmentTidy, <span class="fu">aes</span>(<span class="fu">reorder</span>(pathway, score), score)) <span class="sc">+</span></span>
<span id="cb84-17"><a href="gsea-lab.html#cb84-17" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">fill =</span> score), <span class="at">width=</span><span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb84-18"><a href="gsea-lab.html#cb84-18" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#C1F6F8&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#00BFC4&quot;</span>)<span class="sc">+</span></span>
<span id="cb84-19"><a href="gsea-lab.html#cb84-19" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb84-20"><a href="gsea-lab.html#cb84-20" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;&quot;</span>, <span class="at">y=</span><span class="st">&quot;Normalized Enrichment Score&quot;</span>,</span>
<span id="cb84-21"><a href="gsea-lab.html#cb84-21" tabindex="-1"></a>       <span class="at">title=</span><span class="st">&quot;Pathway enrichment analysis - </span><span class="sc">\n</span><span class="st">Classical PDAC&quot;</span>) <span class="sc">+</span> </span>
<span id="cb84-22"><a href="gsea-lab.html#cb84-22" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb84-23"><a href="gsea-lab.html#cb84-23" tabindex="-1"></a></span>
<span id="cb84-24"><a href="gsea-lab.html#cb84-24" tabindex="-1"></a>p</span></code></pre></div>
<p><img src="Compute-Ontario---Bioinformatics-for-Pathway-Enrichment-Analysis_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="gsea-lab.html#cb85-1" tabindex="-1"></a><span class="do">##YOUR TURN##</span></span>
<span id="cb85-2"><a href="gsea-lab.html#cb85-2" tabindex="-1"></a></span>
<span id="cb85-3"><a href="gsea-lab.html#cb85-3" tabindex="-1"></a><span class="do">## Generate the same bar plot for the top 20 Basal results.  </span></span>
<span id="cb85-4"><a href="gsea-lab.html#cb85-4" tabindex="-1"></a><span class="do">## Change the colour of the bars to a different colour as </span></span>
<span id="cb85-5"><a href="gsea-lab.html#cb85-5" tabindex="-1"></a><span class="do">## they are a different phenotype than the Classical set.</span></span></code></pre></div>
</div>
<div id="dot-plot" class="section level4 hasAnchor">
<h4>Dot plot<a href="gsea-lab.html#dot-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>top pathways associated with classical</li>
</ul>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="gsea-lab.html#cb86-1" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggplot</span>(enrichment_results_topdown, <span class="fu">aes</span>(</span>
<span id="cb86-2"><a href="gsea-lab.html#cb86-2" tabindex="-1"></a>  <span class="at">y =</span> <span class="fu">fct_reorder</span>(pathway, score),</span>
<span id="cb86-3"><a href="gsea-lab.html#cb86-3" tabindex="-1"></a>  <span class="at">x =</span> score,</span>
<span id="cb86-4"><a href="gsea-lab.html#cb86-4" tabindex="-1"></a>  <span class="at">color =</span> score,</span>
<span id="cb86-5"><a href="gsea-lab.html#cb86-5" tabindex="-1"></a>  <span class="at">size =</span> score</span>
<span id="cb86-6"><a href="gsea-lab.html#cb86-6" tabindex="-1"></a>)) <span class="sc">+</span></span>
<span id="cb86-7"><a href="gsea-lab.html#cb86-7" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb86-8"><a href="gsea-lab.html#cb86-8" tabindex="-1"></a>  <span class="fu">scale_color_gradientn</span>(</span>
<span id="cb86-9"><a href="gsea-lab.html#cb86-9" tabindex="-1"></a>    <span class="at">colours =</span> <span class="fu">c</span>(<span class="st">&quot;lightpink&quot;</span>, <span class="st">&quot;lightblue&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb86-10"><a href="gsea-lab.html#cb86-10" tabindex="-1"></a>    <span class="at">trans =</span> <span class="st">&quot;log10&quot;</span>,</span>
<span id="cb86-11"><a href="gsea-lab.html#cb86-11" tabindex="-1"></a>    <span class="at">guide =</span> <span class="fu">guide_colorbar</span>(<span class="at">reverse =</span> <span class="cn">FALSE</span>, <span class="at">order =</span> <span class="dv">1</span>)</span>
<span id="cb86-12"><a href="gsea-lab.html#cb86-12" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb86-13"><a href="gsea-lab.html#cb86-13" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">8</span>)) <span class="sc">+</span></span>
<span id="cb86-14"><a href="gsea-lab.html#cb86-14" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb86-15"><a href="gsea-lab.html#cb86-15" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;score -log10(adj pvalue)&quot;</span>) <span class="sc">+</span></span>
<span id="cb86-16"><a href="gsea-lab.html#cb86-16" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb86-17"><a href="gsea-lab.html#cb86-17" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Pathway Enrichment Analysis - </span><span class="sc">\n</span><span class="st"> Classical PDAC&quot;</span>) <span class="sc">+</span></span>
<span id="cb86-18"><a href="gsea-lab.html#cb86-18" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb86-19"><a href="gsea-lab.html#cb86-19" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>)</span>
<span id="cb86-20"><a href="gsea-lab.html#cb86-20" tabindex="-1"></a>  )</span>
<span id="cb86-21"><a href="gsea-lab.html#cb86-21" tabindex="-1"></a>p</span></code></pre></div>
<p><img src="Compute-Ontario---Bioinformatics-for-Pathway-Enrichment-Analysis_files/figure-html/unnamed-chunk-39-1.png" width="672" /></p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="gsea-lab.html#cb87-1" tabindex="-1"></a><span class="do">##YOUR TURN##</span></span>
<span id="cb87-2"><a href="gsea-lab.html#cb87-2" tabindex="-1"></a></span>
<span id="cb87-3"><a href="gsea-lab.html#cb87-3" tabindex="-1"></a><span class="do">## Generate the same dot plot for the top 20 Basal results.  </span></span>
<span id="cb87-4"><a href="gsea-lab.html#cb87-4" tabindex="-1"></a><span class="do">## Change the colour of the bars to a different colour  range as </span></span>
<span id="cb87-5"><a href="gsea-lab.html#cb87-5" tabindex="-1"></a><span class="do">## they are a different phenotype than the Classical set.</span></span></code></pre></div>
</div>
<div id="ridge-plots" class="section level4 hasAnchor">
<h4>Ridge plots<a href="gsea-lab.html#ridge-plots" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>top pathways associated with either class</li>
</ul>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="gsea-lab.html#cb88-1" tabindex="-1"></a><span class="co"># Filter top pathways, e.g., by adjusted p-value</span></span>
<span id="cb88-2"><a href="gsea-lab.html#cb88-2" tabindex="-1"></a>topPathways <span class="ot">&lt;-</span> current_fgsea_results[padj <span class="sc">&lt;</span> <span class="fl">0.05</span>][<span class="fu">order</span>(padj)][<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb88-3"><a href="gsea-lab.html#cb88-3" tabindex="-1"></a></span>
<span id="cb88-4"><a href="gsea-lab.html#cb88-4" tabindex="-1"></a><span class="co"># Create a data.frame for plotting with the gene ranks for each of the top pathways</span></span>
<span id="cb88-5"><a href="gsea-lab.html#cb88-5" tabindex="-1"></a>ridge_data <span class="ot">&lt;-</span> <span class="fu">rbindlist</span>(<span class="fu">lapply</span>(topPathways<span class="sc">$</span>pathway, <span class="cf">function</span>(pw) {</span>
<span id="cb88-6"><a href="gsea-lab.html#cb88-6" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> all_gs<span class="sc">$</span>genesets[[pw]]</span>
<span id="cb88-7"><a href="gsea-lab.html#cb88-7" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb88-8"><a href="gsea-lab.html#cb88-8" tabindex="-1"></a>    <span class="at">gene =</span> genes,</span>
<span id="cb88-9"><a href="gsea-lab.html#cb88-9" tabindex="-1"></a>    <span class="at">stat =</span> current_ranks[genes],</span>
<span id="cb88-10"><a href="gsea-lab.html#cb88-10" tabindex="-1"></a>    <span class="at">pathway =</span> pw</span>
<span id="cb88-11"><a href="gsea-lab.html#cb88-11" tabindex="-1"></a>  )</span>
<span id="cb88-12"><a href="gsea-lab.html#cb88-12" tabindex="-1"></a>}))</span>
<span id="cb88-13"><a href="gsea-lab.html#cb88-13" tabindex="-1"></a></span>
<span id="cb88-14"><a href="gsea-lab.html#cb88-14" tabindex="-1"></a></span>
<span id="cb88-15"><a href="gsea-lab.html#cb88-15" tabindex="-1"></a><span class="co"># get NES for each of the top pathways</span></span>
<span id="cb88-16"><a href="gsea-lab.html#cb88-16" tabindex="-1"></a>nes_map <span class="ot">&lt;-</span> <span class="fu">setNames</span>(topPathways<span class="sc">$</span>NES, topPathways<span class="sc">$</span>pathway)</span>
<span id="cb88-17"><a href="gsea-lab.html#cb88-17" tabindex="-1"></a></span>
<span id="cb88-18"><a href="gsea-lab.html#cb88-18" tabindex="-1"></a><span class="co"># Add NES values to ridge_data</span></span>
<span id="cb88-19"><a href="gsea-lab.html#cb88-19" tabindex="-1"></a>ridge_data<span class="sc">$</span>NES <span class="ot">&lt;-</span> nes_map[ridge_data<span class="sc">$</span>pathway]</span>
<span id="cb88-20"><a href="gsea-lab.html#cb88-20" tabindex="-1"></a></span>
<span id="cb88-21"><a href="gsea-lab.html#cb88-21" tabindex="-1"></a><span class="co">#change the name to be just the first part of the name (because the actual names are too long)</span></span>
<span id="cb88-22"><a href="gsea-lab.html#cb88-22" tabindex="-1"></a>ridge_data<span class="sc">$</span>pathway <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(ridge_data<span class="sc">$</span>pathway, <span class="at">FUN=</span><span class="cf">function</span>(x){<span class="fu">unlist</span>(<span class="fu">strsplit</span>(x,<span class="at">split =</span> <span class="st">&quot;%&quot;</span>))[<span class="dv">1</span>]}))</span>
<span id="cb88-23"><a href="gsea-lab.html#cb88-23" tabindex="-1"></a></span>
<span id="cb88-24"><a href="gsea-lab.html#cb88-24" tabindex="-1"></a><span class="co">#get rid of na values for genes found in the gene set but not ranked in our dataset.</span></span>
<span id="cb88-25"><a href="gsea-lab.html#cb88-25" tabindex="-1"></a>ridge_data <span class="ot">&lt;-</span> ridge_data[<span class="sc">!</span><span class="fu">is.na</span>(ridge_data<span class="sc">$</span>stat),]</span>
<span id="cb88-26"><a href="gsea-lab.html#cb88-26" tabindex="-1"></a></span>
<span id="cb88-27"><a href="gsea-lab.html#cb88-27" tabindex="-1"></a><span class="fu">ggplot</span>(ridge_data, <span class="fu">aes</span>(<span class="at">x =</span> stat, <span class="at">y =</span> <span class="fu">reorder</span>(pathway, NES), <span class="at">fill =</span> NES)) <span class="sc">+</span></span>
<span id="cb88-28"><a href="gsea-lab.html#cb88-28" tabindex="-1"></a>  <span class="fu">geom_density_ridges</span>(<span class="at">scale =</span> <span class="dv">2</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb88-29"><a href="gsea-lab.html#cb88-29" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(</span>
<span id="cb88-30"><a href="gsea-lab.html#cb88-30" tabindex="-1"></a>    <span class="at">low =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">mid =</span> <span class="st">&quot;white&quot;</span>, <span class="at">high =</span> <span class="st">&quot;red&quot;</span>, <span class="at">midpoint =</span> <span class="dv">0</span>,</span>
<span id="cb88-31"><a href="gsea-lab.html#cb88-31" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;NES&quot;</span></span>
<span id="cb88-32"><a href="gsea-lab.html#cb88-32" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb88-33"><a href="gsea-lab.html#cb88-33" tabindex="-1"></a>  <span class="fu">theme_ridges</span>() <span class="sc">+</span></span>
<span id="cb88-34"><a href="gsea-lab.html#cb88-34" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>,<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>)) <span class="sc">+</span></span>
<span id="cb88-35"><a href="gsea-lab.html#cb88-35" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb88-36"><a href="gsea-lab.html#cb88-36" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Ridgeplot of Gene Stats</span><span class="sc">\n</span><span class="st"> for Top Enriched Pathways&quot;</span>,</span>
<span id="cb88-37"><a href="gsea-lab.html#cb88-37" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;Gene-level Statistic&quot;</span>,</span>
<span id="cb88-38"><a href="gsea-lab.html#cb88-38" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Pathway&quot;</span></span>
<span id="cb88-39"><a href="gsea-lab.html#cb88-39" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="Compute-Ontario---Bioinformatics-for-Pathway-Enrichment-Analysis_files/figure-html/unnamed-chunk-41-1.png" width="672" /></p>
</div>
</div>
<div id="step-8" class="section level3 hasAnchor">
<h3>Step 8<a href="gsea-lab.html#step-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The above plots are only showing a small piece of the underlying data. Ideally we would like to visualize all the signficant results in one figure. To do that we need to use <a href="https://cytoscape.org/">Cytoscape</a>. Unfortunately that is beyond the scope of this current workshop but we will demo it quickly at the end of the workshop.</p>
<p>Output the results so we can visualize the results with programs like cytoscape. In order to do that we need to format the fGSEA output as a file that would be created by GSEA.</p>
<p>Required columns include:
* pathway name
* pathway description
* Details column - found in the GSEA results file but it is not used by Cytoscape
* size - gene set size
* ES - enrichment score
* NES - normalized enrichment score
* Pval - empirical p-value
* Padj - adjusted p-value
* FWER
* Rank at max - the rank of the gene when the ES score is at is max/min
* leading edge genes - the set of genes that contribute to the enrichment of this set.</p>
<p>Create a function to output the properly formatted fgsea result.</p>
<p>The Function requires:
* current_fgsea_results - fgsea results returned
* current_results_dir - directory to the resulting file
* current_sample - name to give the file</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="gsea-lab.html#cb89-1" tabindex="-1"></a>write_sample_fgsea_results<span class="ot">&lt;-</span> <span class="cf">function</span>(current_fgsea_results, current_results_dir, </span>
<span id="cb89-2"><a href="gsea-lab.html#cb89-2" tabindex="-1"></a>                                       current_sample){</span>
<span id="cb89-3"><a href="gsea-lab.html#cb89-3" tabindex="-1"></a>    </span>
<span id="cb89-4"><a href="gsea-lab.html#cb89-4" tabindex="-1"></a>  current_sample <span class="ot">&lt;-</span> current_sample</span>
<span id="cb89-5"><a href="gsea-lab.html#cb89-5" tabindex="-1"></a>  </span>
<span id="cb89-6"><a href="gsea-lab.html#cb89-6" tabindex="-1"></a>    current_sample_directory_fullpath <span class="ot">&lt;-</span> <span class="fu">file.path</span>(current_results_dir, current_sample)</span>
<span id="cb89-7"><a href="gsea-lab.html#cb89-7" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(current_sample_directory_fullpath)){</span>
<span id="cb89-8"><a href="gsea-lab.html#cb89-8" tabindex="-1"></a>      <span class="fu">dir.create</span>(current_sample_directory_fullpath)</span>
<span id="cb89-9"><a href="gsea-lab.html#cb89-9" tabindex="-1"></a>    }</span>
<span id="cb89-10"><a href="gsea-lab.html#cb89-10" tabindex="-1"></a>  </span>
<span id="cb89-11"><a href="gsea-lab.html#cb89-11" tabindex="-1"></a>    <span class="co">#calculate the rank at max</span></span>
<span id="cb89-12"><a href="gsea-lab.html#cb89-12" tabindex="-1"></a>    <span class="co">#fgsea returns the leading edge.  Just need to extract the highest rank from </span></span>
<span id="cb89-13"><a href="gsea-lab.html#cb89-13" tabindex="-1"></a>    <span class="co"># set to get the rank at max</span></span>
<span id="cb89-14"><a href="gsea-lab.html#cb89-14" tabindex="-1"></a>    calculated_rank_at_max <span class="ot">&lt;-</span> <span class="fu">apply</span>(current_fgsea_results,<span class="dv">1</span>,<span class="at">FUN=</span><span class="cf">function</span>(x){ <span class="fu">max</span>(<span class="fu">which</span>(<span class="fu">names</span>(current_ranks) <span class="sc">%in%</span> <span class="fu">unlist</span>(x[<span class="dv">8</span>])))})</span>
<span id="cb89-15"><a href="gsea-lab.html#cb89-15" tabindex="-1"></a>    </span>
<span id="cb89-16"><a href="gsea-lab.html#cb89-16" tabindex="-1"></a>    </span>
<span id="cb89-17"><a href="gsea-lab.html#cb89-17" tabindex="-1"></a>    fakeenr_current_sample <span class="ot">&lt;-</span> <span class="fu">cbind</span>(current_fgsea_results<span class="sc">$</span>pathway,</span>
<span id="cb89-18"><a href="gsea-lab.html#cb89-18" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>pathway,</span>
<span id="cb89-19"><a href="gsea-lab.html#cb89-19" tabindex="-1"></a>                                     <span class="st">&quot;Details&quot;</span>,</span>
<span id="cb89-20"><a href="gsea-lab.html#cb89-20" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>size,</span>
<span id="cb89-21"><a href="gsea-lab.html#cb89-21" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>ES,</span>
<span id="cb89-22"><a href="gsea-lab.html#cb89-22" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>NES,</span>
<span id="cb89-23"><a href="gsea-lab.html#cb89-23" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>pval,</span>
<span id="cb89-24"><a href="gsea-lab.html#cb89-24" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>padj,</span>
<span id="cb89-25"><a href="gsea-lab.html#cb89-25" tabindex="-1"></a>                                     <span class="dv">0</span>,</span>
<span id="cb89-26"><a href="gsea-lab.html#cb89-26" tabindex="-1"></a>                                     calculated_rank_at_max,</span>
<span id="cb89-27"><a href="gsea-lab.html#cb89-27" tabindex="-1"></a>                                     <span class="fu">apply</span>(current_fgsea_results,<span class="dv">1</span>,</span>
<span id="cb89-28"><a href="gsea-lab.html#cb89-28" tabindex="-1"></a>                                           <span class="at">FUN=</span><span class="cf">function</span>(x){<span class="fu">paste</span>(<span class="fu">unlist</span>(x[<span class="dv">8</span>]),<span class="at">collapse=</span><span class="st">&quot;,&quot;</span>)})) </span>
<span id="cb89-29"><a href="gsea-lab.html#cb89-29" tabindex="-1"></a>    </span>
<span id="cb89-30"><a href="gsea-lab.html#cb89-30" tabindex="-1"></a>    <span class="fu">colnames</span>(fakeenr_current_sample) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;name&quot;</span>,<span class="st">&quot;description&quot;</span>,<span class="st">&quot;GS details&quot;</span>,<span class="st">&quot;SIZE&quot;</span>,<span class="st">&quot;ES&quot;</span>,<span class="st">&quot;NES&quot;</span>,<span class="st">&quot;pval&quot;</span>,<span class="st">&quot;padj&quot;</span>,<span class="st">&quot;FWER&quot;</span>,<span class="st">&quot;Rank at Max&quot;</span>,<span class="st">&quot;leading edge genes&quot;</span>)</span>
<span id="cb89-31"><a href="gsea-lab.html#cb89-31" tabindex="-1"></a>    </span>
<span id="cb89-32"><a href="gsea-lab.html#cb89-32" tabindex="-1"></a>    fakeenr_filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(current_sample, <span class="st">&quot;_fgsea_enr_results.txt&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb89-33"><a href="gsea-lab.html#cb89-33" tabindex="-1"></a>    fakeenr_filename_docker <span class="ot">&lt;-</span> <span class="fu">file.path</span>(current_sample_directory_fullpath,fakeenr_filename)</span>
<span id="cb89-34"><a href="gsea-lab.html#cb89-34" tabindex="-1"></a>    </span>
<span id="cb89-35"><a href="gsea-lab.html#cb89-35" tabindex="-1"></a>    <span class="fu">write.table</span>(fakeenr_current_sample ,</span>
<span id="cb89-36"><a href="gsea-lab.html#cb89-36" tabindex="-1"></a>                fakeenr_filename_docker,</span>
<span id="cb89-37"><a href="gsea-lab.html#cb89-37" tabindex="-1"></a>                <span class="at">col.name=</span><span class="cn">TRUE</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">fileEncoding=</span><span class="st">&quot;latin1&quot;</span>)</span>
<span id="cb89-38"><a href="gsea-lab.html#cb89-38" tabindex="-1"></a>    </span>
<span id="cb89-39"><a href="gsea-lab.html#cb89-39" tabindex="-1"></a>    <span class="co"># &quot;upload&quot; the files to the host machine and replace each path with the host machine path</span></span>
<span id="cb89-40"><a href="gsea-lab.html#cb89-40" tabindex="-1"></a>    </span>
<span id="cb89-41"><a href="gsea-lab.html#cb89-41" tabindex="-1"></a>    <span class="co">#create a fake expression file</span></span>
<span id="cb89-42"><a href="gsea-lab.html#cb89-42" tabindex="-1"></a>    fakeexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">name =</span> <span class="fu">names</span>(current_ranks), </span>
<span id="cb89-43"><a href="gsea-lab.html#cb89-43" tabindex="-1"></a>                          <span class="at">description =</span> <span class="fu">names</span>(current_ranks),current_ranks)</span>
<span id="cb89-44"><a href="gsea-lab.html#cb89-44" tabindex="-1"></a>    fakeexp_filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(current_sample,<span class="st">&quot;fakeexpression.txt&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb89-45"><a href="gsea-lab.html#cb89-45" tabindex="-1"></a>    fakeexp_name_docker <span class="ot">&lt;-</span> <span class="fu">file.path</span>( current_sample_directory_fullpath,fakeexp_filename)</span>
<span id="cb89-46"><a href="gsea-lab.html#cb89-46" tabindex="-1"></a></span>
<span id="cb89-47"><a href="gsea-lab.html#cb89-47" tabindex="-1"></a>    <span class="fu">write.table</span>(fakeexp,</span>
<span id="cb89-48"><a href="gsea-lab.html#cb89-48" tabindex="-1"></a>                fakeexp_name_docker,</span>
<span id="cb89-49"><a href="gsea-lab.html#cb89-49" tabindex="-1"></a>                <span class="at">col.name=</span><span class="cn">TRUE</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">fileEncoding=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb89-50"><a href="gsea-lab.html#cb89-50" tabindex="-1"></a>    </span>
<span id="cb89-51"><a href="gsea-lab.html#cb89-51" tabindex="-1"></a>    </span>
<span id="cb89-52"><a href="gsea-lab.html#cb89-52" tabindex="-1"></a>    <span class="co">#create a rank expression file</span></span>
<span id="cb89-53"><a href="gsea-lab.html#cb89-53" tabindex="-1"></a>    fakernk <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">name =</span> <span class="fu">names</span>(current_ranks), </span>
<span id="cb89-54"><a href="gsea-lab.html#cb89-54" tabindex="-1"></a>                          current_ranks)</span>
<span id="cb89-55"><a href="gsea-lab.html#cb89-55" tabindex="-1"></a>    fakernk_filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(current_sample,<span class="st">&quot;fakeranks.rnk&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb89-56"><a href="gsea-lab.html#cb89-56" tabindex="-1"></a>    fakernk_name_docker <span class="ot">&lt;-</span> <span class="fu">file.path</span>( current_sample_directory_fullpath,fakernk_filename)</span>
<span id="cb89-57"><a href="gsea-lab.html#cb89-57" tabindex="-1"></a></span>
<span id="cb89-58"><a href="gsea-lab.html#cb89-58" tabindex="-1"></a>    <span class="fu">write.table</span>(fakernk,</span>
<span id="cb89-59"><a href="gsea-lab.html#cb89-59" tabindex="-1"></a>                fakernk_name_docker,</span>
<span id="cb89-60"><a href="gsea-lab.html#cb89-60" tabindex="-1"></a>                <span class="at">col.name=</span><span class="cn">TRUE</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">fileEncoding=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb89-61"><a href="gsea-lab.html#cb89-61" tabindex="-1"></a>    </span>
<span id="cb89-62"><a href="gsea-lab.html#cb89-62" tabindex="-1"></a>      </span>
<span id="cb89-63"><a href="gsea-lab.html#cb89-63" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="gsea-lab.html#cb90-1" tabindex="-1"></a><span class="co">#write out the fgsea results for this dataset</span></span>
<span id="cb90-2"><a href="gsea-lab.html#cb90-2" tabindex="-1"></a><span class="fu">write_sample_fgsea_results</span>(current_fgsea_results,output_dir,analysis_name)</span></code></pre></div>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="gsea-lab.html#cb91-1" tabindex="-1"></a><span class="do">##YOUR TURN##</span></span>
<span id="cb91-2"><a href="gsea-lab.html#cb91-2" tabindex="-1"></a></span>
<span id="cb91-3"><a href="gsea-lab.html#cb91-3" tabindex="-1"></a><span class="do">## Using the file we just created try and create an enrichment map   </span></span>
<span id="cb91-4"><a href="gsea-lab.html#cb91-4" tabindex="-1"></a><span class="do">## in Cytoscape - see the bonus work for example notebooks with the </span></span>
<span id="cb91-5"><a href="gsea-lab.html#cb91-5" tabindex="-1"></a><span class="do">## detailed instructions</span></span></code></pre></div>
<p><a href="http://www.baderlab.org/CancerStemCellProject/VeroniqueVoisin/AdditionalResources/GSEA#GSEA_enrichment_scores_and_statistics">Link to information about GSEA results</a></p>
</div>
</div>
<div id="additional_information" class="section level2 hasAnchor">
<h2>Additional information<a href="gsea-lab.html#additional_information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="http://www.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats">More on GSEA data format</a></p>
<p><a href="https://baderlab.github.io/Cytoscape_workflows/EnrichmentMapPipeline/supplemental_protocol1_rnaseq.html">More on processing the RNAseq using EdgeR and generate the .rank file</a></p>
<p><a href="http://download.baderlab.org/EM_Genesets/">More on which .gmt file to download from the Baderlab gene-set file</a>, select current release, Human, symbol, Human_GOBP_AllPathways_no_GO_iea_….gmt</p>
<p><a href="http://www.baderlab.org/CancerStemCellProject/VeroniqueVoisin/AdditionalResources/GSEA">More on GSEA : link to the Baderlab wiki page on GSEA</a></p>
</div>
<div id="bonus---automation." class="section level2 hasAnchor">
<h2>Bonus - Automation.<a href="gsea-lab.html#bonus---automation." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Run analysis directly from R for easy integration into existing pipelines.</p>
<div class="rmd-bonus">
<p>
Try out Cytoscape - following this tutorial -
<a href="https://baderlab.github.io/CBW_Pathways_2024/cytoscape_mod3.html" class="uri">https://baderlab.github.io/CBW_Pathways_2024/cytoscape_mod3.html</a>
</p>
<p>
Create an enrichment map from g:profiler or GSEA results - <em>
<a href="https://baderlab.github.io/CBW_Pathways_2024/gprofiler_mod3.html" class="uri">https://baderlab.github.io/CBW_Pathways_2024/gprofiler_mod3.html</a> </em>
<a href="https://baderlab.github.io/CBW_Pathways_2024/gsea_mod3.html" class="uri">https://baderlab.github.io/CBW_Pathways_2024/gsea_mod3.html</a>
</p>
<p>
Instead of using the GSEA application you can run it directly from R
using the GSEA java jar that can be easily used within the workshop
docker image (workshop_base_image) that you setup duing your
prework.
</p>
<p>
Follow the step by step instructions on how to run from R here -
<a href="https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/run-gsea-from-within-r.html" class="uri">https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/run-gsea-from-within-r.html</a>
</p>
<p>
First, make sure your environment is set up correctly by following
there instructions -
<a href="https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/setup.html" class="uri">https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/setup.html</a>
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gprofiler-lab-new.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": "https://github.com/risserlin/ComputeOntario_Pathways_2025/commit/master/2.2-Module2lab_gsea.Rmd",
"text": null
},
"view": {
"link": "https://github.com/risserlin/ComputeOntario_Pathways_2025/master/2.2-Module2lab_gsea.Rmd",
"text": null
},
"download": ["Compute Ontario - Bioinformatics for Pathway Enrichment Analysis.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
