<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 2 lab - fGSEA | Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )</title>
  <meta name="description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 2 lab - fGSEA | Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 2 lab - fGSEA | Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )" />
  
  <meta name="twitter:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  

<meta name="author" content="Ruth Isserlin, Veronique Voisin" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gprofiler-lab-new.html"/>
<link rel="next" href="module-3-network-visualization-and-analysis-with-cytoscape.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://training.computeontario.ca/index.php"><img src="./images/compute_ontario_logo.png" width="280"></a><a href="./">Compute Ontario Summer school</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Compute Ontario Summer School</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html"><i class="fa fa-check"></i>Welcome <a id="welcome"></a></a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#meet-your-faculty"><i class="fa fa-check"></i>Meet your Faculty</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#veronique-voisin"><i class="fa fa-check"></i>Veronique Voisin</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#ruth-isserlin"><i class="fa fa-check"></i>Ruth Isserlin</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop"><i class="fa fa-check"></i>Pre-Workshop Materials and Laptop Setup Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#laptop-setup-instructions"><i class="fa fa-check"></i>Laptop Setup Instructions</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#basic-programs"><i class="fa fa-check"></i>Basic programs</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-installation"><i class="fa fa-check"></i>Cytoscape Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#rstudio-installation"><i class="fa fa-check"></i>RStudio Installation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-tutorials"><i class="fa fa-check"></i>Pre-workshop Tutorials</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-preparation-tutorials"><i class="fa fa-check"></i>Cytoscape Preparation tutorials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-readings-and-lectures"><i class="fa fa-check"></i>Pre-workshop Readings and Lectures</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#additional-tutorials"><i class="fa fa-check"></i>Additional tutorials <a id="additionaltutorials"></a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i>Module 1 - Introduction to Pathway and Network Analysis (Gary Bader)</a></li>
<li class="chapter" data-level="" data-path="module-2-finding-over-represented-pathways-veronique-voisin.html"><a href="module-2-finding-over-represented-pathways-veronique-voisin.html"><i class="fa fa-check"></i>Module 2: Finding Over-represented Pathways (Veronique Voisin)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html"><i class="fa fa-check"></i>Module 2 lab - g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#goal-of-the-exercise-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-1---install-and-load-libraries"><i class="fa fa-check"></i>Step 1 - Install and Load libraries</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-2---input-query"><i class="fa fa-check"></i>Step 2 - Input query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-3---run-gprofiler"><i class="fa fa-check"></i>Step 3 - Run g:Profiler</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-4---explore-top-table-of-results"><i class="fa fa-check"></i>Step 4 - Explore top table of results</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-5---filter-results-by-geneset-size"><i class="fa fa-check"></i>Step 5 - Filter results by geneset size</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-6---run-gprofiler-with-your-own-genesets-example-using-baderlab-genesets"><i class="fa fa-check"></i>Step 6 - Run g:profiler with your own genesets (example using BaderLab genesets)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-7---download-and-load-bader-lab-geneset-file"><i class="fa fa-check"></i>Step 7 - Download and load Bader lab geneset file</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-8---filter-bader-lab-geneset-file"><i class="fa fa-check"></i>Step 8 - Filter Bader lab geneset file</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-9---upload-the-gmt-files-to-gprofiler-and-rerun-gprofiler"><i class="fa fa-check"></i>Step 9 - Upload the gmt files to gprofiler and rerun g:Profiler</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#exercise-2---visualization"><i class="fa fa-check"></i>Exercise 2 - visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-2---dot-plot-with-segment"><i class="fa fa-check"></i>Step 2 - Dot plot with segment</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-3---ridge-plot"><i class="fa fa-check"></i>Step 3 - Ridge plot</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab-new.html"><a href="gprofiler-lab-new.html#step-4---network-visualization---enrichmentmap"><i class="fa fa-check"></i>Step 4 - Network visualization - EnrichmentMap</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html"><i class="fa fa-check"></i>Module 2 lab - fGSEA</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#goal-of-the-exercise"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#data-1"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#background"><i class="fa fa-check"></i>Background</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#how-to-generate-a-rank-file."><i class="fa fa-check"></i>How to generate a rank file.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#start-the-exercise"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step1."><i class="fa fa-check"></i>Step1.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-2."><i class="fa fa-check"></i>Step 2.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-3."><i class="fa fa-check"></i>Step 3.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-4."><i class="fa fa-check"></i>Step 4.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-5."><i class="fa fa-check"></i>Step 5.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-6."><i class="fa fa-check"></i>Step 6.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-7"><i class="fa fa-check"></i>Step 7</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-8"><i class="fa fa-check"></i>Step 8</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#additional_information"><i class="fa fa-check"></i>Additional information</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#bonus---automation."><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-network-visualization-and-analysis-with-cytoscape.html"><a href="module-3-network-visualization-and-analysis-with-cytoscape.html"><i class="fa fa-check"></i>Module 3: Network Visualization and Analysis with Cytoscape</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html"><i class="fa fa-check"></i>Module 3 Lab Primer: Cytoscape Primer</a>
<ul>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#goal-of-the-exercise-2"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#data-2"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#start-the-exercise-1"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1a---create-network-from-table"><i class="fa fa-check"></i>Exercise 1a - Create Network from table</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1b---load-node-attributes"><i class="fa fa-check"></i>Exercise 1b - Load node attributes</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1c---map-node-attributes-to-visual-style"><i class="fa fa-check"></i>Exercise 1c - Map node attributes to Visual Style</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-2---work-with-larger-networks"><i class="fa fa-check"></i>Exercise 2 - Work with larger networks</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-3---perform-basic-enrichment-analysis-using-enrichmenttable"><i class="fa fa-check"></i>Exercise 3 - Perform basic enrichment analysis using EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#enrichmenttabl-features"><i class="fa fa-check"></i>Exercise 3B - create Enrichment Map and Enhanced graphics nodes from EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-4---load-network-from-ndex"><i class="fa fa-check"></i>Exercise 4 - Load network from NDex</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: g:profiler Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#goal-of-the-exercise-3"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#data-3"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#enrichmentmap"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#description-of-this-exercise"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#start-the-exercise-2"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1a---compare-different-gprofiler-geneset-size-results"><i class="fa fa-check"></i>Exercise 1a - compare different gprofiler geneset size results</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step3-explore-the-results"><i class="fa fa-check"></i>Step3: Explore the results:</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#explore-detailed-results"><i class="fa fa-check"></i>Explore Detailed results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1b---is-specifying-the-gmt-file-important"><i class="fa fa-check"></i>Exercise 1b - Is specifying the gmt file important?</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1c---create-em-from-results-using-baderlab-genesets"><i class="fa fa-check"></i>Exercise 1c - create EM from results using Baderlab genesets</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1d-optional---investigate-individual-pathways-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 1d (optional) - investigate individual pathways in GeneMANIA or String</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#genemania"><i class="fa fa-check"></i>GeneMANIA</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#string"><i class="fa fa-check"></i>String</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#bonus---automation.-1"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: GSEA Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#goal-of-the-exercise-4"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#data-4"><i class="fa fa-check"></i>Data</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#enrichmentmap-1"><i class="fa fa-check"></i>EnrichmentMap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-1---gsea-output-and-enrichmentmap"><i class="fa fa-check"></i>Exercise 1 - GSEA output and EnrichmentMap</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-1-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-2-1"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-3"><i class="fa fa-check"></i>Step 3</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-4"><i class="fa fa-check"></i>Step 4</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-2---post-analysis-add-drug-target-gene-sets-to-the-network"><i class="fa fa-check"></i>Exercise 2 - Post analysis (add drug target gene-sets to the network)</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-5"><i class="fa fa-check"></i>Step 5</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-3---autoannotate-the-network"><i class="fa fa-check"></i>Exercise 3 - Autoannotate the Network</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-6"><i class="fa fa-check"></i>Step 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-4-optional---explore-results-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 4 (Optional) - Explore results in GeneMANIA or STRING</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-7-1"><i class="fa fa-check"></i>Step 7</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#bonus---automation.-2"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html"><i class="fa fa-check"></i>Module 3 Lab: (Bonus) Automation</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#goal-of-the-exercise-5"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-1---install-rrstudio"><i class="fa fa-check"></i>Set Up - Option 1 - Install R/Rstudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-2---docker-image-with-rrstudio"><i class="fa fa-check"></i>Set Up - Option 2 - Docker image with R/Rstudio</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#what-is-docker"><i class="fa fa-check"></i>What is docker?</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker---basic-term-definition"><i class="fa fa-check"></i>Docker - Basic term definition</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#container"><i class="fa fa-check"></i>Container</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#image"><i class="fa fa-check"></i>Image</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker-volumes"><i class="fa fa-check"></i>Docker Volumes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#r_docker"><i class="fa fa-check"></i>Install Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#windows"><i class="fa fa-check"></i>Windows</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#macos-linux"><i class="fa fa-check"></i>MacOS / Linux</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#create-your-first-notebook-using-docker"><i class="fa fa-check"></i>Create your first notebook using Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-coding"><i class="fa fa-check"></i>Start coding!</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-using-automation"><i class="fa fa-check"></i>Start using automation</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#running-example-notebooks-in-local-rstudio"><i class="fa fa-check"></i>Running example notebooks in local RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-1---launch-rstudio"><i class="fa fa-check"></i>Step 1 - launch RStudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-2---create-a-new-project"><i class="fa fa-check"></i>Step 2 - create a new project</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-3---open-example-rnotebook"><i class="fa fa-check"></i>Step 3 - Open example RNotebook</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-4---step-through-notebook-to-run-the-analysis"><i class="fa fa-check"></i>Step 4 - Step through notebook to run the analysis</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#additional-resources"><i class="fa fa-check"></i>Additional resources</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gsea-lab" class="section level1 hasAnchor">
<h1>Module 2 lab - fGSEA<a href="gsea-lab.html#gsea-lab" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Presenter: Ruth Isserlin</p>
<div id="introduction-1" class="section level2 hasAnchor">
<h2>Introduction<a href="gsea-lab.html#introduction-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This practical lab contains one exercise. It uses <a href="https://github.com/alserglab/fgsea">fGSEA</a> to perform a gene-set enrichment analysis.</p>
</div>
<div id="goal-of-the-exercise" class="section level2 hasAnchor">
<h2>Goal of the exercise<a href="gsea-lab.html#goal-of-the-exercise" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Learn how to run fGSEA and explore the results.</p>
</div>
<div id="data-1" class="section level2 hasAnchor">
<h2>Data<a href="gsea-lab.html#data-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The data used in this exercise is gene expression (transcriptomics) obtained from high-throughput RNA sequencing of Pancreatic Ductal Adenocarcinoma samples (TCGA-PAAD).</p>
<p>This cohort has been previously stratified into many different set of subtypes <a href="https://pubmed.ncbi.nlm.nih.gov/36765128/">PMID:36765128</a> with the <a href="https://pubmed.ncbi.nlm.nih.gov/26343385/">Moffitt</a> Basal vs Classical subtypes compared to demonstrate the GSEA workflow.</p>
<div id="how-was-the-data-processed" class="section level4 hasAnchor">
<h4>How was the data processed?<a href="gsea-lab.html#how-was-the-data-processed" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Gene expression from the TCGA Pancreatic Ductal Adenocarcinoma RNASeq cohort was downloaded on 2024-06-06 from <a href="https://portal.gdc.cancer.gov/">Genomic Data Commons</a> using the <a href="https://bioconductor.org/packages/release/bioc/html/TCGAbiolinks.html">TCGABiolinks</a> R package.</li>
<li>Differential expression for all genes between the Basal and Classical groups was estimated using <a href="http://www.ncbi.nlm.nih.gov/pubmed/19910308">edgeR</a>.</li>
<li>The R code used to generate the data and the rank file used in GSEA is included at the bottom of the document in the <a href="gsea-lab.html#additional_information"><strong>Additional information</strong></a> section.</li>
</ul>
</div>
</div>
<div id="background" class="section level2 hasAnchor">
<h2>Background<a href="gsea-lab.html#background" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The goal of this lab is to:</p>
<ul>
<li>Upload the 2 required files into GSEA,</li>
<li>Adjust relevant parameters,</li>
<li>Run fGSEA,</li>
<li>Open and explore the gene-set enrichment results.</li>
</ul>
<p>The 2 required files are:</p>
<ol style="list-style-type: decimal">
<li>a rank file (.rnk)</li>
<li>a pathway definition file (.gmt).</li>
</ol>
<div id="rank-file" class="section level4 hasAnchor">
<h4>Rank File<a href="gsea-lab.html#rank-file" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To generate a rank file (.rnk), a score (-log10(pvalue) * sign(logFC)) was calculated from the edgeR differential expression results. A gene that is significantly differentially expressed (i.e associated with a very small pvalue, close to 0) will be assigned a high score.<br>The sign of the logFC indicates if the gene has an expression which is higher in Basel (logFC &gt; 0, the score will have a + sign) or lower in Classical (logFC &lt; 0, the score will have a - sign). It is used to rank the genes from top up-regulated to top down-regulated (<strong>all genes have to be included</strong>).</p>
<div class="rmd-caution">
<p>
The rank file is going to be provided for the lab, you don’t need to
generate it.
</p>
</div>
</div>
<div id="how-to-generate-a-rank-file." class="section level3 hasAnchor">
<h3>How to generate a rank file.<a href="gsea-lab.html#how-to-generate-a-rank-file." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="calculation-of-the-score" class="section level4 hasAnchor">
<h4>Calculation of the score<a href="gsea-lab.html#calculation-of-the-score" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="Module2/gsea/images/calculate_score.png" alt="rank_score" width="75%" /></p>
<p><img src="Module2/gsea/images/GSEA_KS.png" alt="GSEA_KS" width="100%" /></p>
</div>
<div id="generation-of-the-rank-file" class="section level4 hasAnchor">
<h4>Generation of the rank file<a href="gsea-lab.html#generation-of-the-rank-file" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Select the gene names and score columns and save the file as tab delimited with the extension .rnk</p>
<p><img src="Module2/gsea/images/make_rank_file.png" alt="generate rank" width="75%" /></p>
</div>
<div id="pathway-defintion-file" class="section level4 hasAnchor">
<h4>Pathway defintion file<a href="gsea-lab.html#pathway-defintion-file" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The second file that is needed for GSEA is the pathway database, a file with the .gmt extension. The pathway database (.gmt) used for the GSEA analysis was downloaded from <a href="http://baderlab.org/GeneSets" class="uri">http://baderlab.org/GeneSets</a>. This file contains gene-sets obtained from MsigDB-c2 and Hallmarks, NCI, Biocarta, IOB, Netpath, HumanCyc, Reactome, Panther, Pathbank, WikiPathways and the Gene Ontology (GO) databases.</p>
<div class="rmd-caution">
<p>
You don’t need to perform this step for the exercise, the .gmt file
will be given to you. <br>
</p>
</div>
<p>Go to:</p>
<ul>
<li><a href="http://download.baderlab.org/EM_Genesets/" class="uri">http://download.baderlab.org/EM_Genesets/</a></li>
<li>Click on June_01_2024/</li>
<li>Click on Human/</li>
<li>Click on symbol/</li>
<li>Save the Human_GOBP_AllPathways_noPFOCR_no_GO_iea…gmt file on your computer</li>
</ul>
<p><img src="Module2/gsea/images/saving_gmt.png" alt="saving_gmt" width="90%" /></p>
<p>The .gmt is a tab delimited text file which contains one gene-set per row. For each gene-set (row), the first 2 columns contain the name and the description of the gene-set and the remaining columns contain the list of genes included in the gene-set. It is possible to create a custom gene-set using Excel or R.</p>
<p><img src="Module2/gsea/images/gmt_format.png" alt="get_gmt" width="90%" /></p>
<p>GSEA performs a gene-set enrichment analysis using a modified Kolmogorov-Smirnov statistic. The output result consists of summary tables displaying enrichment statistics for each gene-set (pathway) that has been tested.</p>
</div>
</div>
<div id="start-the-exercise" class="section level3 hasAnchor">
<h3>Start the exercise<a href="gsea-lab.html#start-the-exercise" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before starting this exercise, download the 2 required files:</p>
<div class="rmd-datadownload">
<p>
Right click on link below and select “Save Link As…”.
</p>
<p>
Place it in the corresponding module directory of your CBW work
directory.
</p>
</div>
<ul>
<li><a href="./Module2/gsea/data/Human_GOBP_AllPathways_noPFOCR_no_GO_iea_June_01_2024_symbol.gmt">Human_GOBP_AllPathways_noPFOCR_no_GO_iea_June_01_2024_symbol.gmt</a></li>
<li><a href="./Module2/gsea/data//TCGA-PAAD_GDC_Subtype_Moffitt_BasalvsClassical_ranks.rnk">TCGA-PAAD_GDC_Subtype_Moffitt_BasalvsClassical_ranks.rnk</a></li>
</ul>
</div>
<div id="step1." class="section level3 hasAnchor">
<h3>Step1.<a href="gsea-lab.html#step1." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Load required libraries.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="gsea-lab.html#cb44-1" tabindex="-1"></a><span class="co"># List of required packages</span></span>
<span id="cb44-2"><a href="gsea-lab.html#cb44-2" tabindex="-1"></a>required_packages <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb44-3"><a href="gsea-lab.html#cb44-3" tabindex="-1"></a>  <span class="st">&quot;tidyverse&quot;</span>,     <span class="co"># Data manipulation and visualization</span></span>
<span id="cb44-4"><a href="gsea-lab.html#cb44-4" tabindex="-1"></a>  <span class="st">&quot;GSA&quot;</span>,           <span class="co"># Gene set analysis</span></span>
<span id="cb44-5"><a href="gsea-lab.html#cb44-5" tabindex="-1"></a>  <span class="st">&quot;RCurl&quot;</span>,         <span class="co"># HTTP requests</span></span>
<span id="cb44-6"><a href="gsea-lab.html#cb44-6" tabindex="-1"></a>  <span class="st">&quot;fgsea&quot;</span>,         <span class="co"># fast GSEA</span></span>
<span id="cb44-7"><a href="gsea-lab.html#cb44-7" tabindex="-1"></a>  <span class="st">&quot;data.table&quot;</span>,    <span class="co"># for formatting the results</span></span>
<span id="cb44-8"><a href="gsea-lab.html#cb44-8" tabindex="-1"></a>  <span class="st">&quot;ggridges&quot;</span>      <span class="co"># Ridgeline plots</span></span>
<span id="cb44-9"><a href="gsea-lab.html#cb44-9" tabindex="-1"></a>)</span>
<span id="cb44-10"><a href="gsea-lab.html#cb44-10" tabindex="-1"></a></span>
<span id="cb44-11"><a href="gsea-lab.html#cb44-11" tabindex="-1"></a><span class="co"># Install any missing packages</span></span>
<span id="cb44-12"><a href="gsea-lab.html#cb44-12" tabindex="-1"></a>installed <span class="ot">&lt;-</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>())</span>
<span id="cb44-13"><a href="gsea-lab.html#cb44-13" tabindex="-1"></a><span class="cf">for</span> (pkg <span class="cf">in</span> required_packages) {</span>
<span id="cb44-14"><a href="gsea-lab.html#cb44-14" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>(pkg <span class="sc">%in%</span> installed)) {</span>
<span id="cb44-15"><a href="gsea-lab.html#cb44-15" tabindex="-1"></a>    <span class="fu">install.packages</span>(pkg)</span>
<span id="cb44-16"><a href="gsea-lab.html#cb44-16" tabindex="-1"></a>  }</span>
<span id="cb44-17"><a href="gsea-lab.html#cb44-17" tabindex="-1"></a>}</span>
<span id="cb44-18"><a href="gsea-lab.html#cb44-18" tabindex="-1"></a></span>
<span id="cb44-19"><a href="gsea-lab.html#cb44-19" tabindex="-1"></a><span class="co"># Load all packages</span></span>
<span id="cb44-20"><a href="gsea-lab.html#cb44-20" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">lapply</span>(required_packages, library, <span class="at">character.only =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:lubridate&#39;:
## 
##     hour, isoweek, mday, minute, month, quarter, second, wday, week,
##     yday, year</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     between, first, last</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     transpose</code></pre>
</div>
<div id="step-2." class="section level3 hasAnchor">
<h3>Step 2.<a href="gsea-lab.html#step-2." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Set up parameters need for the analysis.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="gsea-lab.html#cb49-1" tabindex="-1"></a><span class="co"># defined in the paramters at top of notebook</span></span>
<span id="cb49-2"><a href="gsea-lab.html#cb49-2" tabindex="-1"></a></span>
<span id="cb49-3"><a href="gsea-lab.html#cb49-3" tabindex="-1"></a><span class="co">#directory where all the data files are found.  </span></span>
<span id="cb49-4"><a href="gsea-lab.html#cb49-4" tabindex="-1"></a>working_dir <span class="ot">&lt;-</span> <span class="st">&quot;./data/&quot;</span></span>
<span id="cb49-5"><a href="gsea-lab.html#cb49-5" tabindex="-1"></a></span>
<span id="cb49-6"><a href="gsea-lab.html#cb49-6" tabindex="-1"></a><span class="co">#directory where all the data files are found.  </span></span>
<span id="cb49-7"><a href="gsea-lab.html#cb49-7" tabindex="-1"></a>output_dir <span class="ot">&lt;-</span> <span class="st">&quot;./generated_data/&quot;</span></span>
<span id="cb49-8"><a href="gsea-lab.html#cb49-8" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">exists</span>(output_dir)){</span>
<span id="cb49-9"><a href="gsea-lab.html#cb49-9" tabindex="-1"></a>  <span class="fu">dir.create</span>(output_dir)</span>
<span id="cb49-10"><a href="gsea-lab.html#cb49-10" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## Warning in dir.create(output_dir): &#39;./generated_data&#39; already exists</code></pre>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="gsea-lab.html#cb51-1" tabindex="-1"></a><span class="co">#The name to give the analysis in GSEA - for example</span></span>
<span id="cb51-2"><a href="gsea-lab.html#cb51-2" tabindex="-1"></a>analysis_name <span class="ot">&lt;-</span> <span class="st">&quot;Basal_vs_Classical&quot;</span></span>
<span id="cb51-3"><a href="gsea-lab.html#cb51-3" tabindex="-1"></a></span>
<span id="cb51-4"><a href="gsea-lab.html#cb51-4" tabindex="-1"></a><span class="co">#rank file to use in GSEA analysis.  </span></span>
<span id="cb51-5"><a href="gsea-lab.html#cb51-5" tabindex="-1"></a>rnk_file <span class="ot">&lt;-</span> <span class="st">&quot;TCGA-PAAD_GDC_Subtype_Moffitt_BasalvsClassical_ranks.rnk&quot;</span></span>
<span id="cb51-6"><a href="gsea-lab.html#cb51-6" tabindex="-1"></a></span>
<span id="cb51-7"><a href="gsea-lab.html#cb51-7" tabindex="-1"></a><span class="co">#set the gmt file you want to use if you don&#39;t want to use the latest gmt file.</span></span>
<span id="cb51-8"><a href="gsea-lab.html#cb51-8" tabindex="-1"></a><span class="co"># For example, if you set dest_gmt_file ==&quot;&quot; the below script will automatically</span></span>
<span id="cb51-9"><a href="gsea-lab.html#cb51-9" tabindex="-1"></a><span class="co"># download the latest gmt file from baderlab webstie.  If it is set then it</span></span>
<span id="cb51-10"><a href="gsea-lab.html#cb51-10" tabindex="-1"></a><span class="co"># will use the file specified.  </span></span>
<span id="cb51-11"><a href="gsea-lab.html#cb51-11" tabindex="-1"></a>dest_gmt_file <span class="ot">=</span> <span class="st">&quot;&quot;</span></span></code></pre></div>
</div>
<div id="step-3." class="section level3 hasAnchor">
<h3>Step 3.<a href="gsea-lab.html#step-3." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Download the latest pathway definition file</p>
<p>Only Human, Mouse, Rat, and Woodchuck gene set files are currently available on the baderlab downloads site. If you are working with a species other than human (and it is either rat,mouse or woodchuck) change the gmt_url below to the correct species. Check here to see all available species.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="gsea-lab.html#cb52-1" tabindex="-1"></a><span class="cf">if</span>(dest_gmt_file <span class="sc">==</span> <span class="st">&quot;&quot;</span>){</span>
<span id="cb52-2"><a href="gsea-lab.html#cb52-2" tabindex="-1"></a>  gmt_url <span class="ot">=</span> <span class="st">&quot;http://download.baderlab.org/EM_Genesets/current_release/Human/symbol/&quot;</span></span>
<span id="cb52-3"><a href="gsea-lab.html#cb52-3" tabindex="-1"></a>  </span>
<span id="cb52-4"><a href="gsea-lab.html#cb52-4" tabindex="-1"></a>  <span class="co">#list all the files on the server</span></span>
<span id="cb52-5"><a href="gsea-lab.html#cb52-5" tabindex="-1"></a>  filenames <span class="ot">=</span> <span class="fu">getURL</span>(gmt_url)</span>
<span id="cb52-6"><a href="gsea-lab.html#cb52-6" tabindex="-1"></a>  tc <span class="ot">=</span> <span class="fu">textConnection</span>(filenames)</span>
<span id="cb52-7"><a href="gsea-lab.html#cb52-7" tabindex="-1"></a>  contents <span class="ot">=</span> <span class="fu">readLines</span>(tc)</span>
<span id="cb52-8"><a href="gsea-lab.html#cb52-8" tabindex="-1"></a>  <span class="fu">close</span>(tc)</span>
<span id="cb52-9"><a href="gsea-lab.html#cb52-9" tabindex="-1"></a>  </span>
<span id="cb52-10"><a href="gsea-lab.html#cb52-10" tabindex="-1"></a>  <span class="co">#get the gmt that has all the pathways and does not include terms </span></span>
<span id="cb52-11"><a href="gsea-lab.html#cb52-11" tabindex="-1"></a>  <span class="co"># inferred from electronic annotations(IEA)</span></span>
<span id="cb52-12"><a href="gsea-lab.html#cb52-12" tabindex="-1"></a>  <span class="co">#start with gmt file that has pathways only and GO Biological Process only.</span></span>
<span id="cb52-13"><a href="gsea-lab.html#cb52-13" tabindex="-1"></a>  rx <span class="ot">=</span> <span class="fu">gregexpr</span>(<span class="st">&quot;(?&lt;=&lt;a href=</span><span class="sc">\&quot;</span><span class="st">)(.*.GOBP_AllPathways_noPFOCR_no_GO_iea.*.)(.gmt)(?=</span><span class="sc">\&quot;</span><span class="st">&gt;)&quot;</span>,</span>
<span id="cb52-14"><a href="gsea-lab.html#cb52-14" tabindex="-1"></a>    contents, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb52-15"><a href="gsea-lab.html#cb52-15" tabindex="-1"></a>  gmt_file <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">regmatches</span>(contents, rx))</span>
<span id="cb52-16"><a href="gsea-lab.html#cb52-16" tabindex="-1"></a>  </span>
<span id="cb52-17"><a href="gsea-lab.html#cb52-17" tabindex="-1"></a>  dest_gmt_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_dir,gmt_file )</span>
<span id="cb52-18"><a href="gsea-lab.html#cb52-18" tabindex="-1"></a>  </span>
<span id="cb52-19"><a href="gsea-lab.html#cb52-19" tabindex="-1"></a>  <span class="co">#check if this gmt file already exists</span></span>
<span id="cb52-20"><a href="gsea-lab.html#cb52-20" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(dest_gmt_file)){</span>
<span id="cb52-21"><a href="gsea-lab.html#cb52-21" tabindex="-1"></a>    <span class="fu">download.file</span>(</span>
<span id="cb52-22"><a href="gsea-lab.html#cb52-22" tabindex="-1"></a>      <span class="fu">paste</span>(gmt_url,gmt_file,<span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb52-23"><a href="gsea-lab.html#cb52-23" tabindex="-1"></a>      <span class="at">destfile=</span>dest_gmt_file</span>
<span id="cb52-24"><a href="gsea-lab.html#cb52-24" tabindex="-1"></a>    )</span>
<span id="cb52-25"><a href="gsea-lab.html#cb52-25" tabindex="-1"></a>  }</span>
<span id="cb52-26"><a href="gsea-lab.html#cb52-26" tabindex="-1"></a>}<span class="cf">else</span> {</span>
<span id="cb52-27"><a href="gsea-lab.html#cb52-27" tabindex="-1"></a>  <span class="fu">file.copy</span>(dest_gmt_file,<span class="at">to =</span> output_dir)</span>
<span id="cb52-28"><a href="gsea-lab.html#cb52-28" tabindex="-1"></a>}</span></code></pre></div>
<p>Load your gmt file</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="gsea-lab.html#cb53-1" tabindex="-1"></a><span class="fu">capture.output</span>(all_gs <span class="ot">&lt;-</span> <span class="fu">GSA.read.gmt</span>(dest_gmt_file) ,<span class="at">file=</span><span class="st">&quot;gsa_load.out&quot;</span>)</span>
<span id="cb53-2"><a href="gsea-lab.html#cb53-2" tabindex="-1"></a><span class="fu">names</span>(all_gs<span class="sc">$</span>genesets) <span class="ot">&lt;-</span> all_gs<span class="sc">$</span>geneset.names</span></code></pre></div>
</div>
<div id="step-4." class="section level3 hasAnchor">
<h3>Step 4.<a href="gsea-lab.html#step-4." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Load your Rank file</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="gsea-lab.html#cb54-1" tabindex="-1"></a>current_ranks <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">file.path</span>(working_dir,rnk_file),<span class="at">header=</span><span class="cn">TRUE</span>,<span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb54-2"><a href="gsea-lab.html#cb54-2" tabindex="-1"></a>    fgsea_ranks <span class="ot">&lt;-</span> current_ranks[,<span class="dv">2</span>]</span>
<span id="cb54-3"><a href="gsea-lab.html#cb54-3" tabindex="-1"></a>    <span class="fu">names</span>(fgsea_ranks) <span class="ot">&lt;-</span> current_ranks[,<span class="dv">1</span>]</span>
<span id="cb54-4"><a href="gsea-lab.html#cb54-4" tabindex="-1"></a> </span>
<span id="cb54-5"><a href="gsea-lab.html#cb54-5" tabindex="-1"></a>current_ranks <span class="ot">&lt;-</span> fgsea_ranks</span>
<span id="cb54-6"><a href="gsea-lab.html#cb54-6" tabindex="-1"></a> </span>
<span id="cb54-7"><a href="gsea-lab.html#cb54-7" tabindex="-1"></a><span class="co">#remove duplicated genes</span></span>
<span id="cb54-8"><a href="gsea-lab.html#cb54-8" tabindex="-1"></a>duplicated_gene_names <span class="ot">&lt;-</span>   </span>
<span id="cb54-9"><a href="gsea-lab.html#cb54-9" tabindex="-1"></a>      <span class="fu">names</span>(current_ranks)[<span class="fu">which</span>(<span class="fu">duplicated</span>(<span class="fu">names</span>(current_ranks)))]</span>
<span id="cb54-10"><a href="gsea-lab.html#cb54-10" tabindex="-1"></a>    current_ranks <span class="ot">&lt;-</span> current_ranks[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">names</span>(current_ranks) <span class="sc">%in%</span> </span>
<span id="cb54-11"><a href="gsea-lab.html#cb54-11" tabindex="-1"></a>                                           duplicated_gene_names)]</span>
<span id="cb54-12"><a href="gsea-lab.html#cb54-12" tabindex="-1"></a></span>
<span id="cb54-13"><a href="gsea-lab.html#cb54-13" tabindex="-1"></a>current_ranks <span class="ot">&lt;-</span> <span class="fu">sort</span>(current_ranks,<span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="step-5." class="section level3 hasAnchor">
<h3>Step 5.<a href="gsea-lab.html#step-5." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Run fGSEA - Adjust parameters</p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/fgsea.htm">fGSEA</a>] is an R package that runs a fast Gene Set Enrichment Analysis.</p>
<p>In the below command the following options have been specified:</p>
<ul>
<li>pathways - list of genesets to use for the calculation</li>
<li>stats - genes and their associated statistic, sorted</li>
<li>max_size - maximum size for individual gene sets. In GSEA interface this is set to 500 but we prefer to use a more stringent setting of 200.</li>
<li>min_size - minimum size for individual gene sets</li>
<li>gseaParam - GSEA parameter value</li>
</ul>
<div class="rmd-caution">
<p>
gseaParam - GSEA parameter value. All ranks will be raised to this
power when calculating the ES scores. Equivalent to the weight in the
original GSEA algorithm.
</p>
</div>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="gsea-lab.html#cb55-1" tabindex="-1"></a>current_fgsea_results <span class="ot">&lt;-</span> fgsea<span class="sc">::</span><span class="fu">fgsea</span>(all_gs<span class="sc">$</span>genesets, </span>
<span id="cb55-2"><a href="gsea-lab.html#cb55-2" tabindex="-1"></a>                                      current_ranks,</span>
<span id="cb55-3"><a href="gsea-lab.html#cb55-3" tabindex="-1"></a>                                      <span class="at">minSize=</span><span class="dv">15</span>,</span>
<span id="cb55-4"><a href="gsea-lab.html#cb55-4" tabindex="-1"></a>                                      <span class="at">maxSize =</span> <span class="dv">500</span></span>
<span id="cb55-5"><a href="gsea-lab.html#cb55-5" tabindex="-1"></a>                                      )</span></code></pre></div>
<div class="rmd-tip">
<p>
Set <strong>gseaParam</strong> to 2 if you want to add more weight on
the most top up-regulated and top down-regulated. <br>
<strong>2</strong> is a more stringent parameter and it will result in
less gene-sets significant under FDR &lt;0.05.
</p>
</div>
</div>
<div id="step-6." class="section level3 hasAnchor">
<h3>Step 6.<a href="gsea-lab.html#step-6." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Examining the results</p>
<p>Get the top results and visualize as table</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="gsea-lab.html#cb56-1" tabindex="-1"></a>topPathwaysUp <span class="ot">&lt;-</span> current_fgsea_results[ES <span class="sc">&gt;</span> <span class="dv">0</span>][<span class="fu">head</span>(<span class="fu">order</span>(pval), <span class="at">n=</span><span class="dv">10</span>), pathway]</span>
<span id="cb56-2"><a href="gsea-lab.html#cb56-2" tabindex="-1"></a>topPathwaysDown <span class="ot">&lt;-</span> current_fgsea_results[ES <span class="sc">&lt;</span> <span class="dv">0</span>][<span class="fu">head</span>(<span class="fu">order</span>(pval), <span class="at">n=</span><span class="dv">10</span>), pathway]</span>
<span id="cb56-3"><a href="gsea-lab.html#cb56-3" tabindex="-1"></a>topPathways <span class="ot">&lt;-</span> <span class="fu">c</span>(topPathwaysUp, <span class="fu">rev</span>(topPathwaysDown))</span>
<span id="cb56-4"><a href="gsea-lab.html#cb56-4" tabindex="-1"></a><span class="fu">plotGseaTable</span>(all_gs<span class="sc">$</span>genesets[topPathways], current_ranks, current_fgsea_results, </span>
<span id="cb56-5"><a href="gsea-lab.html#cb56-5" tabindex="-1"></a>              <span class="at">gseaParam=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="Compute-Ontario---Bioinformatics-for-Pathway-Enrichment-Analysis_files/figure-html/top_pathways-1.png" width="672" /></p>
<p>When examining the results there are a few things to look for:</p>
<p>Check the number of gene-sets that have been used for the analysis.</p>
<div class="rmd-tip">
<p>
A small number (a few hundred genesets if using baderlab genesets)
could indicate an issue with identifier mapping.
</p>
</div>
<p>Check the number of sets that have FDR less than 0.25 – in order to determine what thresholds to start with when creating the enrichment map. It is not uncommon to see a thousand gene sets pass the threshold of FDR less than 0.25. FDR less than 0.25 is a very lax threshold and for robust data we can set thresholds of FDR less than 0.05 or lower.</p>
<p>Number of pathways with corrected pvalue &lt; 0.05</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="gsea-lab.html#cb57-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span></code></pre></div>
<pre><code>## [1] 459</code></pre>
<p>Number of pathways with corrected pvalue &lt; 0.01</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="gsea-lab.html#cb59-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.01</span>))</span></code></pre></div>
<pre><code>## [1] 213</code></pre>
<p>Number of Basal (up -regulated) pathways with corrected pvalue &lt; 0.05</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="gsea-lab.html#cb61-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> current_fgsea_results<span class="sc">$</span>ES <span class="sc">&gt;</span> <span class="dv">0</span>))</span></code></pre></div>
<pre><code>## [1] 366</code></pre>
<p>Number of Classical (down -regulated) pathways with corrected pvalue &lt; 0.05</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="gsea-lab.html#cb63-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> current_fgsea_results<span class="sc">$</span>ES <span class="sc">&lt;</span> <span class="dv">0</span>))</span></code></pre></div>
<pre><code>## [1] 93</code></pre>
<p>5e. Explore the tabular format of the results.</p>
<div id="basal" class="section level4 hasAnchor">
<h4>Basal<a href="gsea-lab.html#basal" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="gsea-lab.html#cb65-1" tabindex="-1"></a>topPathwaysUp <span class="ot">&lt;-</span> current_fgsea_results[ES <span class="sc">&gt;</span> <span class="dv">0</span>][<span class="fu">head</span>(<span class="fu">order</span>(pval), <span class="at">n=</span><span class="dv">5</span>), pathway]</span>
<span id="cb65-2"><a href="gsea-lab.html#cb65-2" tabindex="-1"></a></span>
<span id="cb65-3"><a href="gsea-lab.html#cb65-3" tabindex="-1"></a>top_basal_hits <span class="ot">&lt;-</span> current_fgsea_results[<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>pathway <span class="sc">%in%</span> topPathwaysUp),]</span>
<span id="cb65-4"><a href="gsea-lab.html#cb65-4" tabindex="-1"></a></span>
<span id="cb65-5"><a href="gsea-lab.html#cb65-5" tabindex="-1"></a><span class="co">#format the pathway name colum so that it is easier to see the whole table</span></span>
<span id="cb65-6"><a href="gsea-lab.html#cb65-6" tabindex="-1"></a>top_basal_hits<span class="sc">$</span>pathway <span class="ot">&lt;-</span> <span class="fu">substr</span>(top_basal_hits<span class="sc">$</span>pathway, <span class="at">start =</span> <span class="dv">1</span>, <span class="at">stop=</span><span class="dv">25</span>)</span>
<span id="cb65-7"><a href="gsea-lab.html#cb65-7" tabindex="-1"></a></span>
<span id="cb65-8"><a href="gsea-lab.html#cb65-8" tabindex="-1"></a>top_basal_hits</span></code></pre></div>
<pre><code>##                      pathway         pval         padj   log2err        ES
##                       &lt;char&gt;        &lt;num&gt;        &lt;num&gt;     &lt;num&gt;     &lt;num&gt;
## 1: CELL CYCLE CHECKPOINTS%RE 3.015287e-10 2.356070e-07 0.8140358 0.5400523
## 2: CELL CYCLE, MITOTIC%REACT 2.415773e-16 7.550499e-13 1.0376962 0.5211659
## 3: HALLMARK_EPITHELIAL_MESEN 1.353353e-16 7.550499e-13 1.0476265 0.6962643
## 4: KERATINIZATION%REACTOME%R 2.212670e-10 2.178571e-07 0.8266573 0.8141973
## 5: SKIN DEVELOPMENT%GOBP%GO: 2.439609e-10 2.178571e-07 0.8140358 0.7078168
##         NES  size                            leadingEdge
##       &lt;num&gt; &lt;int&gt;                                 &lt;list&gt;
## 1: 1.997942   243 PSMD2,MCM4,CENPH,KIF18A,MCM7,CHEK1,...
## 2: 2.038142   469  PSMD2,MCM4,CCND1,FOXM1,TUBB,NUP93,...
## 3: 2.459937   156  SNAI2,TGFBI,LAMC2,CDH2,NT5E,FSTL3,...
## 4: 2.389030    47  KRT6A,KRT5,KRT81,KRT14,KRT13,KRT7,...
## 5: 2.299659    87   KRT6A,ANXA1,KRT5,KRT81,TP63,KRT7,...</code></pre>
</div>
<div id="classical" class="section level4 hasAnchor">
<h4>Classical<a href="gsea-lab.html#classical" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="gsea-lab.html#cb67-1" tabindex="-1"></a>topPathwaysDown <span class="ot">&lt;-</span> current_fgsea_results[ES <span class="sc">&lt;</span> <span class="dv">0</span>][<span class="fu">head</span>(<span class="fu">order</span>(pval), <span class="at">n=</span><span class="dv">5</span>), pathway]</span>
<span id="cb67-2"><a href="gsea-lab.html#cb67-2" tabindex="-1"></a></span>
<span id="cb67-3"><a href="gsea-lab.html#cb67-3" tabindex="-1"></a>top_classical_hits <span class="ot">&lt;-</span> current_fgsea_results[<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>pathway <span class="sc">%in%</span> topPathwaysDown),]</span>
<span id="cb67-4"><a href="gsea-lab.html#cb67-4" tabindex="-1"></a></span>
<span id="cb67-5"><a href="gsea-lab.html#cb67-5" tabindex="-1"></a><span class="co">#format the pathway name colum so that it is easier to see the whole table</span></span>
<span id="cb67-6"><a href="gsea-lab.html#cb67-6" tabindex="-1"></a>top_classical_hits<span class="sc">$</span>pathway <span class="ot">&lt;-</span> <span class="fu">substr</span>(top_classical_hits<span class="sc">$</span>pathway, <span class="at">start =</span> <span class="dv">1</span>, <span class="at">stop=</span><span class="dv">25</span>)</span>
<span id="cb67-7"><a href="gsea-lab.html#cb67-7" tabindex="-1"></a></span>
<span id="cb67-8"><a href="gsea-lab.html#cb67-8" tabindex="-1"></a>top_classical_hits</span></code></pre></div>
<pre><code>##                      pathway         pval         padj   log2err         ES
##                       &lt;char&gt;        &lt;num&gt;        &lt;num&gt;     &lt;num&gt;      &lt;num&gt;
## 1: BIOLOGICAL OXIDATIONS%REA 1.970798e-10 2.178571e-07 0.8266573 -0.6362964
## 2: CARBOXYLIC ACID METABOLIC 4.268676e-08 1.201526e-05 0.7195128 -0.4563033
## 3: CELLULAR RESPIRATION%GOBP 1.444767e-08 5.879109e-06 0.7477397 -0.6108973
## 4: CELLULAR RESPONSE TO XENO 1.081606e-10 1.690279e-07 0.8390889 -0.7080315
## 5: XENOBIOTIC METABOLIC PROC 5.064558e-11 1.055285e-07 0.8513391 -0.7533267
##          NES  size                                leadingEdge
##        &lt;num&gt; &lt;int&gt;                                     &lt;list&gt;
## 1: -2.087654   148   CYP2S1,ALDH2,ACSS1,ADH1C,AOC1,AKR7A3,...
## 2: -1.690261   490  ALDOB,CYP2S1,PLA2G10,ABHD3,CPS1,LPIN1,...
## 3: -1.965509   135 BLOC1S1,COA6,PPARGC1A,FH,ATP5ME,SUCLG1,...
## 4: -2.206002    97     FMO5,HNF4A,CYP2S1,ALDH2,NR1I2,CES2,...
## 5: -2.258234    78     FMO5,HNF4A,CYP2S1,ALDH2,NR1I2,CES2,...</code></pre>
</div>
</div>
<div id="step-7" class="section level3 hasAnchor">
<h3>Step 7<a href="gsea-lab.html#step-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Visualize the top results as different sorts of plots, as demonstrated in the first part of the lab.</p>
<div id="bar-plots" class="section level4 hasAnchor">
<h4>Bar plots<a href="gsea-lab.html#bar-plots" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>top pathways associated with classical</li>
</ul>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="gsea-lab.html#cb69-1" tabindex="-1"></a>topPathwaysDown <span class="ot">&lt;-</span> current_fgsea_results[ES <span class="sc">&lt;</span> <span class="dv">0</span>][<span class="fu">head</span>(<span class="fu">order</span>(pval), <span class="at">n=</span><span class="dv">20</span>), pathway]</span>
<span id="cb69-2"><a href="gsea-lab.html#cb69-2" tabindex="-1"></a></span>
<span id="cb69-3"><a href="gsea-lab.html#cb69-3" tabindex="-1"></a>enrichment_results_topdown <span class="ot">&lt;-</span>current_fgsea_results[<span class="fu">which</span>(current_fgsea_results<span class="sc">$</span>pathway <span class="sc">%in%</span> topPathwaysDown),]</span>
<span id="cb69-4"><a href="gsea-lab.html#cb69-4" tabindex="-1"></a></span>
<span id="cb69-5"><a href="gsea-lab.html#cb69-5" tabindex="-1"></a><span class="co">#change the name to be just the first part of the name</span></span>
<span id="cb69-6"><a href="gsea-lab.html#cb69-6" tabindex="-1"></a>enrichment_results_topdown<span class="sc">$</span>pathway <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(enrichment_results_topdown<span class="sc">$</span>pathway, <span class="at">FUN=</span><span class="cf">function</span>(x){<span class="fu">unlist</span>(<span class="fu">strsplit</span>(x,<span class="at">split =</span> <span class="st">&quot;%&quot;</span>))[<span class="dv">1</span>]}))</span>
<span id="cb69-7"><a href="gsea-lab.html#cb69-7" tabindex="-1"></a></span>
<span id="cb69-8"><a href="gsea-lab.html#cb69-8" tabindex="-1"></a><span class="do">##Calculate a score</span></span>
<span id="cb69-9"><a href="gsea-lab.html#cb69-9" tabindex="-1"></a>enrichment_results_topdown<span class="sc">$</span>score <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">as.numeric</span>(enrichment_results_topdown<span class="sc">$</span>pval))</span>
<span id="cb69-10"><a href="gsea-lab.html#cb69-10" tabindex="-1"></a></span>
<span id="cb69-11"><a href="gsea-lab.html#cb69-11" tabindex="-1"></a>enrichmentTidy <span class="ot">&lt;-</span> enrichment_results_topdown <span class="sc">%&gt;%</span></span>
<span id="cb69-12"><a href="gsea-lab.html#cb69-12" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb69-13"><a href="gsea-lab.html#cb69-13" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(score)) <span class="sc">%&gt;%</span></span>
<span id="cb69-14"><a href="gsea-lab.html#cb69-14" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb69-15"><a href="gsea-lab.html#cb69-15" tabindex="-1"></a></span>
<span id="cb69-16"><a href="gsea-lab.html#cb69-16" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggplot</span>(enrichmentTidy, <span class="fu">aes</span>(<span class="fu">reorder</span>(pathway, score), score)) <span class="sc">+</span></span>
<span id="cb69-17"><a href="gsea-lab.html#cb69-17" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">fill =</span> score), <span class="at">width=</span><span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb69-18"><a href="gsea-lab.html#cb69-18" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#C1F6F8&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#00BFC4&quot;</span>)<span class="sc">+</span></span>
<span id="cb69-19"><a href="gsea-lab.html#cb69-19" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb69-20"><a href="gsea-lab.html#cb69-20" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;&quot;</span>, <span class="at">y=</span><span class="st">&quot;Normalized Enrichment Score&quot;</span>,</span>
<span id="cb69-21"><a href="gsea-lab.html#cb69-21" tabindex="-1"></a>       <span class="at">title=</span><span class="st">&quot;Pathway enrichment analysis - </span><span class="sc">\n</span><span class="st">Classical PDAC&quot;</span>) <span class="sc">+</span> </span>
<span id="cb69-22"><a href="gsea-lab.html#cb69-22" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb69-23"><a href="gsea-lab.html#cb69-23" tabindex="-1"></a></span>
<span id="cb69-24"><a href="gsea-lab.html#cb69-24" tabindex="-1"></a>p</span></code></pre></div>
<p><img src="Compute-Ontario---Bioinformatics-for-Pathway-Enrichment-Analysis_files/figure-html/unnamed-chunk-34-1.png" width="672" /></p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="gsea-lab.html#cb70-1" tabindex="-1"></a><span class="do">##YOUR TURN##</span></span>
<span id="cb70-2"><a href="gsea-lab.html#cb70-2" tabindex="-1"></a></span>
<span id="cb70-3"><a href="gsea-lab.html#cb70-3" tabindex="-1"></a><span class="do">## Generate the same bar plot for the top 20 Basal results.  </span></span>
<span id="cb70-4"><a href="gsea-lab.html#cb70-4" tabindex="-1"></a><span class="do">## Change the colour of the bars to a different colour as they are a different phecotype from the Classical set.</span></span></code></pre></div>
</div>
<div id="dot-plot" class="section level4 hasAnchor">
<h4>Dot plot<a href="gsea-lab.html#dot-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>top pathways associated with classical</li>
</ul>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="gsea-lab.html#cb71-1" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggplot</span>(enrichment_results_topdown, <span class="fu">aes</span>(</span>
<span id="cb71-2"><a href="gsea-lab.html#cb71-2" tabindex="-1"></a>  <span class="at">y =</span> <span class="fu">fct_reorder</span>(pathway, score),</span>
<span id="cb71-3"><a href="gsea-lab.html#cb71-3" tabindex="-1"></a>  <span class="at">x =</span> score,</span>
<span id="cb71-4"><a href="gsea-lab.html#cb71-4" tabindex="-1"></a>  <span class="at">color =</span> score,</span>
<span id="cb71-5"><a href="gsea-lab.html#cb71-5" tabindex="-1"></a>  <span class="at">size =</span> score</span>
<span id="cb71-6"><a href="gsea-lab.html#cb71-6" tabindex="-1"></a>)) <span class="sc">+</span></span>
<span id="cb71-7"><a href="gsea-lab.html#cb71-7" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb71-8"><a href="gsea-lab.html#cb71-8" tabindex="-1"></a>  <span class="fu">scale_color_gradientn</span>(</span>
<span id="cb71-9"><a href="gsea-lab.html#cb71-9" tabindex="-1"></a>    <span class="at">colours =</span> <span class="fu">c</span>(<span class="st">&quot;lightpink&quot;</span>, <span class="st">&quot;lightblue&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb71-10"><a href="gsea-lab.html#cb71-10" tabindex="-1"></a>    <span class="at">trans =</span> <span class="st">&quot;log10&quot;</span>,</span>
<span id="cb71-11"><a href="gsea-lab.html#cb71-11" tabindex="-1"></a>    <span class="at">guide =</span> <span class="fu">guide_colorbar</span>(<span class="at">reverse =</span> <span class="cn">FALSE</span>, <span class="at">order =</span> <span class="dv">1</span>)</span>
<span id="cb71-12"><a href="gsea-lab.html#cb71-12" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb71-13"><a href="gsea-lab.html#cb71-13" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">8</span>)) <span class="sc">+</span></span>
<span id="cb71-14"><a href="gsea-lab.html#cb71-14" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb71-15"><a href="gsea-lab.html#cb71-15" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;score -log10(adj pvalue)&quot;</span>) <span class="sc">+</span></span>
<span id="cb71-16"><a href="gsea-lab.html#cb71-16" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb71-17"><a href="gsea-lab.html#cb71-17" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Pathway Enrichment Analysis - </span><span class="sc">\n</span><span class="st"> Classical PDAC&quot;</span>) <span class="sc">+</span></span>
<span id="cb71-18"><a href="gsea-lab.html#cb71-18" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb71-19"><a href="gsea-lab.html#cb71-19" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>)</span>
<span id="cb71-20"><a href="gsea-lab.html#cb71-20" tabindex="-1"></a>  )</span>
<span id="cb71-21"><a href="gsea-lab.html#cb71-21" tabindex="-1"></a>p</span></code></pre></div>
<p><img src="Compute-Ontario---Bioinformatics-for-Pathway-Enrichment-Analysis_files/figure-html/unnamed-chunk-36-1.png" width="672" /></p>
</div>
<div id="ridge-plots" class="section level4 hasAnchor">
<h4>Ridge plots<a href="gsea-lab.html#ridge-plots" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>top pathways associated with either class</li>
</ul>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="gsea-lab.html#cb72-1" tabindex="-1"></a><span class="co"># Filter top pathways, e.g., by adjusted p-value</span></span>
<span id="cb72-2"><a href="gsea-lab.html#cb72-2" tabindex="-1"></a>topPathways <span class="ot">&lt;-</span> current_fgsea_results[padj <span class="sc">&lt;</span> <span class="fl">0.05</span>][<span class="fu">order</span>(padj)][<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb72-3"><a href="gsea-lab.html#cb72-3" tabindex="-1"></a></span>
<span id="cb72-4"><a href="gsea-lab.html#cb72-4" tabindex="-1"></a><span class="co"># Create a data.frame for plotting with the gene ranks for each of the top pathways</span></span>
<span id="cb72-5"><a href="gsea-lab.html#cb72-5" tabindex="-1"></a>ridge_data <span class="ot">&lt;-</span> <span class="fu">rbindlist</span>(<span class="fu">lapply</span>(topPathways<span class="sc">$</span>pathway, <span class="cf">function</span>(pw) {</span>
<span id="cb72-6"><a href="gsea-lab.html#cb72-6" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> all_gs<span class="sc">$</span>genesets[[pw]]</span>
<span id="cb72-7"><a href="gsea-lab.html#cb72-7" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb72-8"><a href="gsea-lab.html#cb72-8" tabindex="-1"></a>    <span class="at">gene =</span> genes,</span>
<span id="cb72-9"><a href="gsea-lab.html#cb72-9" tabindex="-1"></a>    <span class="at">stat =</span> current_ranks[genes],</span>
<span id="cb72-10"><a href="gsea-lab.html#cb72-10" tabindex="-1"></a>    <span class="at">pathway =</span> pw</span>
<span id="cb72-11"><a href="gsea-lab.html#cb72-11" tabindex="-1"></a>  )</span>
<span id="cb72-12"><a href="gsea-lab.html#cb72-12" tabindex="-1"></a>}))</span>
<span id="cb72-13"><a href="gsea-lab.html#cb72-13" tabindex="-1"></a></span>
<span id="cb72-14"><a href="gsea-lab.html#cb72-14" tabindex="-1"></a></span>
<span id="cb72-15"><a href="gsea-lab.html#cb72-15" tabindex="-1"></a><span class="co"># get NES for each of the top pathways</span></span>
<span id="cb72-16"><a href="gsea-lab.html#cb72-16" tabindex="-1"></a>nes_map <span class="ot">&lt;-</span> <span class="fu">setNames</span>(topPathways<span class="sc">$</span>NES, topPathways<span class="sc">$</span>pathway)</span>
<span id="cb72-17"><a href="gsea-lab.html#cb72-17" tabindex="-1"></a></span>
<span id="cb72-18"><a href="gsea-lab.html#cb72-18" tabindex="-1"></a><span class="co"># Add NES values to ridge_data</span></span>
<span id="cb72-19"><a href="gsea-lab.html#cb72-19" tabindex="-1"></a>ridge_data<span class="sc">$</span>NES <span class="ot">&lt;-</span> nes_map[ridge_data<span class="sc">$</span>pathway]</span>
<span id="cb72-20"><a href="gsea-lab.html#cb72-20" tabindex="-1"></a></span>
<span id="cb72-21"><a href="gsea-lab.html#cb72-21" tabindex="-1"></a><span class="co">#change the name to be just the first part of the name (because the actual names are too long)</span></span>
<span id="cb72-22"><a href="gsea-lab.html#cb72-22" tabindex="-1"></a>ridge_data<span class="sc">$</span>pathway <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(ridge_data<span class="sc">$</span>pathway, <span class="at">FUN=</span><span class="cf">function</span>(x){<span class="fu">unlist</span>(<span class="fu">strsplit</span>(x,<span class="at">split =</span> <span class="st">&quot;%&quot;</span>))[<span class="dv">1</span>]}))</span>
<span id="cb72-23"><a href="gsea-lab.html#cb72-23" tabindex="-1"></a></span>
<span id="cb72-24"><a href="gsea-lab.html#cb72-24" tabindex="-1"></a><span class="co">#get rid of na values for genes found in the gene set but not ranked in our dataset.</span></span>
<span id="cb72-25"><a href="gsea-lab.html#cb72-25" tabindex="-1"></a>ridge_data <span class="ot">&lt;-</span> ridge_data[<span class="sc">!</span><span class="fu">is.na</span>(ridge_data<span class="sc">$</span>stat),]</span>
<span id="cb72-26"><a href="gsea-lab.html#cb72-26" tabindex="-1"></a></span>
<span id="cb72-27"><a href="gsea-lab.html#cb72-27" tabindex="-1"></a><span class="fu">ggplot</span>(ridge_data, <span class="fu">aes</span>(<span class="at">x =</span> stat, <span class="at">y =</span> <span class="fu">reorder</span>(pathway, NES), <span class="at">fill =</span> NES)) <span class="sc">+</span></span>
<span id="cb72-28"><a href="gsea-lab.html#cb72-28" tabindex="-1"></a>  <span class="fu">geom_density_ridges</span>(<span class="at">scale =</span> <span class="dv">2</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb72-29"><a href="gsea-lab.html#cb72-29" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(</span>
<span id="cb72-30"><a href="gsea-lab.html#cb72-30" tabindex="-1"></a>    <span class="at">low =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">mid =</span> <span class="st">&quot;white&quot;</span>, <span class="at">high =</span> <span class="st">&quot;red&quot;</span>, <span class="at">midpoint =</span> <span class="dv">0</span>,</span>
<span id="cb72-31"><a href="gsea-lab.html#cb72-31" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;NES&quot;</span></span>
<span id="cb72-32"><a href="gsea-lab.html#cb72-32" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb72-33"><a href="gsea-lab.html#cb72-33" tabindex="-1"></a>  <span class="fu">theme_ridges</span>() <span class="sc">+</span></span>
<span id="cb72-34"><a href="gsea-lab.html#cb72-34" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>,<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>)) <span class="sc">+</span></span>
<span id="cb72-35"><a href="gsea-lab.html#cb72-35" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb72-36"><a href="gsea-lab.html#cb72-36" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;Ridgeplot of Gene Stats</span><span class="sc">\n</span><span class="st"> for Top Enriched Pathways&quot;</span>,</span>
<span id="cb72-37"><a href="gsea-lab.html#cb72-37" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;Gene-level Statistic&quot;</span>,</span>
<span id="cb72-38"><a href="gsea-lab.html#cb72-38" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Pathway&quot;</span></span>
<span id="cb72-39"><a href="gsea-lab.html#cb72-39" tabindex="-1"></a>  )</span></code></pre></div>
<p><img src="Compute-Ontario---Bioinformatics-for-Pathway-Enrichment-Analysis_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
</div>
</div>
<div id="step-8" class="section level3 hasAnchor">
<h3>Step 8<a href="gsea-lab.html#step-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Output the results so we can visualize the results with programs like cytoscape. In order to do that we need to format the fGSEA output as a file that would be created by GSEA.</p>
<p>Required columns include:
* pathway name
* pathway description
* Details column - found in the GSEA results file but it is not used by Cytoscape
* size - gene set size
* ES - enrichment score
* NES - normalized enrichment score
* Pval - empirical p-value
* Padj - adjusted p-value
* FWER
* Rank at max - the rank of the gene when the ES score is at is max/min
* leading edge genes - the set of genes that contribute to the enrichment of this set.</p>
<p>Create a function to output the properly formatted fgsea result.</p>
<p>The Function requires:
* current_fgsea_results - fgsea results returned
* current_results_dir - directory to the resulting file
* current_sample - name to give the file</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="gsea-lab.html#cb73-1" tabindex="-1"></a>write_sample_fgsea_results<span class="ot">&lt;-</span> <span class="cf">function</span>(current_fgsea_results, current_results_dir, </span>
<span id="cb73-2"><a href="gsea-lab.html#cb73-2" tabindex="-1"></a>                                       current_sample){</span>
<span id="cb73-3"><a href="gsea-lab.html#cb73-3" tabindex="-1"></a>    </span>
<span id="cb73-4"><a href="gsea-lab.html#cb73-4" tabindex="-1"></a>  current_sample <span class="ot">&lt;-</span> current_sample</span>
<span id="cb73-5"><a href="gsea-lab.html#cb73-5" tabindex="-1"></a>  </span>
<span id="cb73-6"><a href="gsea-lab.html#cb73-6" tabindex="-1"></a>    current_sample_directory_fullpath <span class="ot">&lt;-</span> <span class="fu">file.path</span>(current_results_dir, current_sample)</span>
<span id="cb73-7"><a href="gsea-lab.html#cb73-7" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(current_sample_directory_fullpath)){</span>
<span id="cb73-8"><a href="gsea-lab.html#cb73-8" tabindex="-1"></a>      <span class="fu">dir.create</span>(current_sample_directory_fullpath)</span>
<span id="cb73-9"><a href="gsea-lab.html#cb73-9" tabindex="-1"></a>    }</span>
<span id="cb73-10"><a href="gsea-lab.html#cb73-10" tabindex="-1"></a>  </span>
<span id="cb73-11"><a href="gsea-lab.html#cb73-11" tabindex="-1"></a>    <span class="co">#calculate the rank at max</span></span>
<span id="cb73-12"><a href="gsea-lab.html#cb73-12" tabindex="-1"></a>    <span class="co">#fgsea returns the leading edge.  Just need to extract the highest rank from </span></span>
<span id="cb73-13"><a href="gsea-lab.html#cb73-13" tabindex="-1"></a>    <span class="co"># set to get the rank at max</span></span>
<span id="cb73-14"><a href="gsea-lab.html#cb73-14" tabindex="-1"></a>    calculated_rank_at_max <span class="ot">&lt;-</span> <span class="fu">apply</span>(current_fgsea_results,<span class="dv">1</span>,<span class="at">FUN=</span><span class="cf">function</span>(x){ <span class="fu">max</span>(<span class="fu">which</span>(<span class="fu">names</span>(current_ranks) <span class="sc">%in%</span> <span class="fu">unlist</span>(x[<span class="dv">8</span>])))})</span>
<span id="cb73-15"><a href="gsea-lab.html#cb73-15" tabindex="-1"></a>    </span>
<span id="cb73-16"><a href="gsea-lab.html#cb73-16" tabindex="-1"></a>    </span>
<span id="cb73-17"><a href="gsea-lab.html#cb73-17" tabindex="-1"></a>    fakeenr_current_sample <span class="ot">&lt;-</span> <span class="fu">cbind</span>(current_fgsea_results<span class="sc">$</span>pathway,</span>
<span id="cb73-18"><a href="gsea-lab.html#cb73-18" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>pathway,</span>
<span id="cb73-19"><a href="gsea-lab.html#cb73-19" tabindex="-1"></a>                                     <span class="st">&quot;Details&quot;</span>,</span>
<span id="cb73-20"><a href="gsea-lab.html#cb73-20" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>size,</span>
<span id="cb73-21"><a href="gsea-lab.html#cb73-21" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>ES,</span>
<span id="cb73-22"><a href="gsea-lab.html#cb73-22" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>NES,</span>
<span id="cb73-23"><a href="gsea-lab.html#cb73-23" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>pval,</span>
<span id="cb73-24"><a href="gsea-lab.html#cb73-24" tabindex="-1"></a>                                     current_fgsea_results<span class="sc">$</span>padj,</span>
<span id="cb73-25"><a href="gsea-lab.html#cb73-25" tabindex="-1"></a>                                     <span class="dv">0</span>,</span>
<span id="cb73-26"><a href="gsea-lab.html#cb73-26" tabindex="-1"></a>                                     calculated_rank_at_max,</span>
<span id="cb73-27"><a href="gsea-lab.html#cb73-27" tabindex="-1"></a>                                     <span class="fu">apply</span>(current_fgsea_results,<span class="dv">1</span>,</span>
<span id="cb73-28"><a href="gsea-lab.html#cb73-28" tabindex="-1"></a>                                           <span class="at">FUN=</span><span class="cf">function</span>(x){<span class="fu">paste</span>(<span class="fu">unlist</span>(x[<span class="dv">8</span>]),<span class="at">collapse=</span><span class="st">&quot;,&quot;</span>)})) </span>
<span id="cb73-29"><a href="gsea-lab.html#cb73-29" tabindex="-1"></a>    </span>
<span id="cb73-30"><a href="gsea-lab.html#cb73-30" tabindex="-1"></a>    <span class="fu">colnames</span>(fakeenr_current_sample) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;name&quot;</span>,<span class="st">&quot;description&quot;</span>,<span class="st">&quot;GS details&quot;</span>,<span class="st">&quot;SIZE&quot;</span>,<span class="st">&quot;ES&quot;</span>,<span class="st">&quot;NES&quot;</span>,<span class="st">&quot;pval&quot;</span>,<span class="st">&quot;padj&quot;</span>,<span class="st">&quot;FWER&quot;</span>,<span class="st">&quot;Rank at Max&quot;</span>,<span class="st">&quot;leading edge genes&quot;</span>)</span>
<span id="cb73-31"><a href="gsea-lab.html#cb73-31" tabindex="-1"></a>    </span>
<span id="cb73-32"><a href="gsea-lab.html#cb73-32" tabindex="-1"></a>    fakeenr_filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(current_sample, <span class="st">&quot;_fgsea_enr_results.txt&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb73-33"><a href="gsea-lab.html#cb73-33" tabindex="-1"></a>    fakeenr_filename_docker <span class="ot">&lt;-</span> <span class="fu">file.path</span>(current_sample_directory_fullpath,fakeenr_filename)</span>
<span id="cb73-34"><a href="gsea-lab.html#cb73-34" tabindex="-1"></a>    </span>
<span id="cb73-35"><a href="gsea-lab.html#cb73-35" tabindex="-1"></a>    <span class="fu">write.table</span>(fakeenr_current_sample ,</span>
<span id="cb73-36"><a href="gsea-lab.html#cb73-36" tabindex="-1"></a>                fakeenr_filename_docker,</span>
<span id="cb73-37"><a href="gsea-lab.html#cb73-37" tabindex="-1"></a>                <span class="at">col.name=</span><span class="cn">TRUE</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">fileEncoding=</span><span class="st">&quot;latin1&quot;</span>)</span>
<span id="cb73-38"><a href="gsea-lab.html#cb73-38" tabindex="-1"></a>    </span>
<span id="cb73-39"><a href="gsea-lab.html#cb73-39" tabindex="-1"></a>    <span class="co"># &quot;upload&quot; the files to the host machine and replace each path with the host machine path</span></span>
<span id="cb73-40"><a href="gsea-lab.html#cb73-40" tabindex="-1"></a>    </span>
<span id="cb73-41"><a href="gsea-lab.html#cb73-41" tabindex="-1"></a>    <span class="co">#create a fake expression file</span></span>
<span id="cb73-42"><a href="gsea-lab.html#cb73-42" tabindex="-1"></a>    fakeexp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">name =</span> <span class="fu">names</span>(current_ranks), </span>
<span id="cb73-43"><a href="gsea-lab.html#cb73-43" tabindex="-1"></a>                          <span class="at">description =</span> <span class="fu">names</span>(current_ranks),current_ranks)</span>
<span id="cb73-44"><a href="gsea-lab.html#cb73-44" tabindex="-1"></a>    fakeexp_filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(current_sample,<span class="st">&quot;fakeexpression.txt&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb73-45"><a href="gsea-lab.html#cb73-45" tabindex="-1"></a>    fakeexp_name_docker <span class="ot">&lt;-</span> <span class="fu">file.path</span>( current_sample_directory_fullpath,fakeexp_filename)</span>
<span id="cb73-46"><a href="gsea-lab.html#cb73-46" tabindex="-1"></a></span>
<span id="cb73-47"><a href="gsea-lab.html#cb73-47" tabindex="-1"></a>    <span class="fu">write.table</span>(fakeexp,</span>
<span id="cb73-48"><a href="gsea-lab.html#cb73-48" tabindex="-1"></a>                fakeexp_name_docker,</span>
<span id="cb73-49"><a href="gsea-lab.html#cb73-49" tabindex="-1"></a>                <span class="at">col.name=</span><span class="cn">TRUE</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">fileEncoding=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb73-50"><a href="gsea-lab.html#cb73-50" tabindex="-1"></a>    </span>
<span id="cb73-51"><a href="gsea-lab.html#cb73-51" tabindex="-1"></a>    </span>
<span id="cb73-52"><a href="gsea-lab.html#cb73-52" tabindex="-1"></a>    <span class="co">#create a rank expression file</span></span>
<span id="cb73-53"><a href="gsea-lab.html#cb73-53" tabindex="-1"></a>    fakernk <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">name =</span> <span class="fu">names</span>(current_ranks), </span>
<span id="cb73-54"><a href="gsea-lab.html#cb73-54" tabindex="-1"></a>                          current_ranks)</span>
<span id="cb73-55"><a href="gsea-lab.html#cb73-55" tabindex="-1"></a>    fakernk_filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(current_sample,<span class="st">&quot;fakeranks.rnk&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb73-56"><a href="gsea-lab.html#cb73-56" tabindex="-1"></a>    fakernk_name_docker <span class="ot">&lt;-</span> <span class="fu">file.path</span>( current_sample_directory_fullpath,fakernk_filename)</span>
<span id="cb73-57"><a href="gsea-lab.html#cb73-57" tabindex="-1"></a></span>
<span id="cb73-58"><a href="gsea-lab.html#cb73-58" tabindex="-1"></a>    <span class="fu">write.table</span>(fakernk,</span>
<span id="cb73-59"><a href="gsea-lab.html#cb73-59" tabindex="-1"></a>                fakernk_name_docker,</span>
<span id="cb73-60"><a href="gsea-lab.html#cb73-60" tabindex="-1"></a>                <span class="at">col.name=</span><span class="cn">TRUE</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span><span class="cn">FALSE</span>,<span class="at">quote=</span><span class="cn">FALSE</span>,<span class="at">fileEncoding=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb73-61"><a href="gsea-lab.html#cb73-61" tabindex="-1"></a>    </span>
<span id="cb73-62"><a href="gsea-lab.html#cb73-62" tabindex="-1"></a>      </span>
<span id="cb73-63"><a href="gsea-lab.html#cb73-63" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="gsea-lab.html#cb74-1" tabindex="-1"></a><span class="co">#write out the fgsea results for this patient</span></span>
<span id="cb74-2"><a href="gsea-lab.html#cb74-2" tabindex="-1"></a>      <span class="fu">write_sample_fgsea_results</span>(current_fgsea_results,output_dir,analysis_name)</span></code></pre></div>
<pre><code>## Warning in write.table(fakeenr_current_sample, fakeenr_filename_docker, :
## invalid char string in output conversion
## Warning in write.table(fakeenr_current_sample, fakeenr_filename_docker, :
## invalid char string in output conversion
## Warning in write.table(fakeenr_current_sample, fakeenr_filename_docker, :
## invalid char string in output conversion
## Warning in write.table(fakeenr_current_sample, fakeenr_filename_docker, :
## invalid char string in output conversion</code></pre>
<p><a href="http://www.baderlab.org/CancerStemCellProject/VeroniqueVoisin/AdditionalResources/GSEA#GSEA_enrichment_scores_and_statistics">Link to information about GSEA results</a></p>
</div>
</div>
<div id="additional_information" class="section level2 hasAnchor">
<h2>Additional information<a href="gsea-lab.html#additional_information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="http://www.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats">More on GSEA data format</a></p>
<p><a href="https://baderlab.github.io/Cytoscape_workflows/EnrichmentMapPipeline/supplemental_protocol1_rnaseq.html">More on processing the RNAseq using EdgeR and generate the .rank file</a></p>
<p><a href="http://download.baderlab.org/EM_Genesets/">More on which .gmt file to download from the Baderlab gene-set file</a>, select current release, Human, symbol, Human_GOBP_AllPathways_no_GO_iea_….gmt</p>
<p><a href="http://www.baderlab.org/CancerStemCellProject/VeroniqueVoisin/AdditionalResources/GSEA">More on GSEA : link to the Baderlab wiki page on GSEA</a></p>
</div>
<div id="bonus---automation." class="section level2 hasAnchor">
<h2>Bonus - Automation.<a href="gsea-lab.html#bonus---automation." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Run analysis directly from R for easy integration into existing pipelines.</p>
<div class="rmd-bonus">
<p>
Instead of using the GSEA application you can run it directly from R
using the GSEA java jar that can be easily used within the workshop
docker image (workshop_base_image) that you setup duing your
prework.
</p>
<p>
Follow the step by step instructions on how to run from R here -
<a href="https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/run-gsea-from-within-r.html" class="uri">https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/run-gsea-from-within-r.html</a>
</p>
<p>
First, make sure your environment is set up correctly by following
there instructions -
<a href="https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/setup.html" class="uri">https://risserlin.github.io/CBW_pathways_workshop_R_notebooks/setup.html</a>
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gprofiler-lab-new.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="module-3-network-visualization-and-analysis-with-cytoscape.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": "https://github.com/risserlin/ComputeOntario_Pathways_2025/commit/master/2.2-Module2lab_gsea.Rmd",
"text": null
},
"view": {
"link": "https://github.com/risserlin/ComputeOntario_Pathways_2025/master/2.2-Module2lab_gsea.Rmd",
"text": null
},
"download": ["Compute Ontario - Bioinformatics for Pathway Enrichment Analysis.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
