<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Welcome  | Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )</title>
  <meta name="description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Welcome  | Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Welcome  | Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )" />
  
  <meta name="twitter:description" content="Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information." />
  

<meta name="author" content="Ruth Isserlin, Veronique Voisin" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="intro.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://training.computeontario.ca/index.php"><img src="./images/compute_ontario_logo.png" width="280"></a><a href="./">Compute Ontario Summer school</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Compute Ontario Summer School</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html"><i class="fa fa-check"></i>Welcome <a id="welcome"></a></a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#meet-your-faculty"><i class="fa fa-check"></i>Meet your Faculty</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#veronique-voisin"><i class="fa fa-check"></i>Veronique Voisin</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#ruth-isserlin"><i class="fa fa-check"></i>Ruth Isserlin</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#class-materials"><i class="fa fa-check"></i>Class Materials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#schedule"><i class="fa fa-check"></i>Workshop Schedule</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop"><i class="fa fa-check"></i>Pre-Workshop Materials and Laptop Setup Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#laptop-setup-instructions"><i class="fa fa-check"></i>Laptop Setup Instructions</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#basic-programs"><i class="fa fa-check"></i>Basic programs</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-installation"><i class="fa fa-check"></i>Cytoscape Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#gsea-installation"><i class="fa fa-check"></i>GSEA Installation</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#docker-installation"><i class="fa fa-check"></i>Docker Installation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-tutorials"><i class="fa fa-check"></i>Pre-workshop Tutorials</a>
<ul>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#cytoscape-preparation-tutorials"><i class="fa fa-check"></i>Cytoscape Preparation tutorials</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#r-tutorial"><i class="fa fa-check"></i>R Tutorial</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#pre-workshop-readings-and-lectures"><i class="fa fa-check"></i>Pre-workshop Readings and Lectures</a></li>
<li class="chapter" data-level="" data-path="welcome.html"><a href="welcome.html#additional-tutorials"><i class="fa fa-check"></i>Additional tutorials <a id="additionaltutorials"></a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i>Module 1 - Introduction to Pathway and Network Analysis (Gary Bader)</a></li>
<li class="chapter" data-level="" data-path="module-2-finding-over-represented-pathways-veronique-voisin.html"><a href="module-2-finding-over-represented-pathways-veronique-voisin.html"><i class="fa fa-check"></i>Module 2: Finding Over-represented Pathways (Veronique Voisin)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html"><i class="fa fa-check"></i>Module 2 lab - g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#goal-of-the-exercise-1"><i class="fa fa-check"></i>Goal of the exercise 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-1"><i class="fa fa-check"></i>Exercise 1 - run g:Profiler</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-1---launch-gprofiler."><i class="fa fa-check"></i>Step 1 - Launch g:Profiler.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-2---input-query"><i class="fa fa-check"></i>Step 2 - input query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-3---adjust-parameters."><i class="fa fa-check"></i>Step 3 - Adjust parameters.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-4---run-query"><i class="fa fa-check"></i>Step 4 - Run query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-5---explore-the-results."><i class="fa fa-check"></i>Step 5 - Explore the results.</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-6-expand-the-stats-tab"><i class="fa fa-check"></i>Step 6: Expand the stats tab</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-7-save-the-results"><i class="fa fa-check"></i>Step 7: Save the results</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-8-optional-but-recommended"><i class="fa fa-check"></i>Step 8 (Optional but recommended)</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#step-9-optional-by-recommended"><i class="fa fa-check"></i>Step 9 (Optional by recommended)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#exercise-2-load-and-use-a-custom-.gmt-file-and-run-the-query"><i class="fa fa-check"></i>Exercise 2: Load and use a custom .gmt file and run the query</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-steps"><i class="fa fa-check"></i>Optional steps</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#optional-1"><i class="fa fa-check"></i><strong>Optional 1</strong>:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#option-2"><i class="fa fa-check"></i><strong>Option 2</strong>:</a></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#option-3"><i class="fa fa-check"></i><strong>Option 3</strong> :</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler-lab.html"><a href="gprofiler-lab.html#bonus---automation."><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html"><i class="fa fa-check"></i>Module 2 lab - GSEA</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#goal-of-the-exercise"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#data-1"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#background"><i class="fa fa-check"></i>Background</a>
<ul>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#how-to-generate-a-rank-file."><i class="fa fa-check"></i>How to generate a rank file.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#start-the-exercise"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step1."><i class="fa fa-check"></i>Step1.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-2."><i class="fa fa-check"></i>Step 2.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step3."><i class="fa fa-check"></i>Step3.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-4."><i class="fa fa-check"></i>Step 4.</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#step-5."><i class="fa fa-check"></i>Step 5.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#additional_information"><i class="fa fa-check"></i>Additional information</a></li>
<li class="chapter" data-level="" data-path="gsea-lab.html"><a href="gsea-lab.html#bonus---automation.-1"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-network-visualization-and-analysis-with-cytoscape.html"><a href="module-3-network-visualization-and-analysis-with-cytoscape.html"><i class="fa fa-check"></i>Module 3: Network Visualization and Analysis with Cytoscape</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html"><i class="fa fa-check"></i>Module 3 Lab Primer: Cytoscape Primer</a>
<ul>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#goal-of-the-exercise-2"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#data-2"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#start-the-exercise-1"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1a---create-network-from-table"><i class="fa fa-check"></i>Exercise 1a - Create Network from table</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1b---load-node-attributes"><i class="fa fa-check"></i>Exercise 1b - Load node attributes</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-1c---map-node-attributes-to-visual-style"><i class="fa fa-check"></i>Exercise 1c - Map node attributes to Visual Style</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-2---work-with-larger-networks"><i class="fa fa-check"></i>Exercise 2 - Work with larger networks</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-3---perform-basic-enrichment-analysis-using-enrichmenttable"><i class="fa fa-check"></i>Exercise 3 - Perform basic enrichment analysis using EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#enrichmenttabl-features"><i class="fa fa-check"></i>Exercise 3B - create Enrichment Map and Enhanced graphics nodes from EnrichmentTable</a></li>
<li class="chapter" data-level="" data-path="cytoscape_mod3.html"><a href="cytoscape_mod3.html#exercise-4---load-network-from-ndex"><i class="fa fa-check"></i>Exercise 4 - Load network from NDex</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: g:profiler Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#goal-of-the-exercise-3"><i class="fa fa-check"></i>Goal of the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#data-3"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#enrichmentmap"><i class="fa fa-check"></i>EnrichmentMap</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#description-of-this-exercise"><i class="fa fa-check"></i>Description of this exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#start-the-exercise-2"><i class="fa fa-check"></i>Start the exercise</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1a---compare-different-gprofiler-geneset-size-results"><i class="fa fa-check"></i>Exercise 1a - compare different gprofiler geneset size results</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step-2"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#step3-explore-the-results"><i class="fa fa-check"></i>Step3: Explore the results:</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#explore-detailed-results"><i class="fa fa-check"></i>Explore Detailed results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1b---is-specifying-the-gmt-file-important"><i class="fa fa-check"></i>Exercise 1b - Is specifying the gmt file important?</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1c---create-em-from-results-using-baderlab-genesets"><i class="fa fa-check"></i>Exercise 1c - create EM from results using Baderlab genesets</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#exercise-1d-optional---investigate-individual-pathways-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 1d (optional) - investigate individual pathways in GeneMANIA or String</a>
<ul>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#genemania"><i class="fa fa-check"></i>GeneMANIA</a></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#string"><i class="fa fa-check"></i>String</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gprofiler_mod3.html"><a href="gprofiler_mod3.html#bonus---automation.-2"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html"><i class="fa fa-check"></i>Module 3 Lab: GSEA Visualization</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#goal-of-the-exercise-4"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#data-4"><i class="fa fa-check"></i>Data</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#enrichmentmap-1"><i class="fa fa-check"></i>EnrichmentMap</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-1---gsea-output-and-enrichmentmap"><i class="fa fa-check"></i>Exercise 1 - GSEA output and EnrichmentMap</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-1-1"><i class="fa fa-check"></i>Step 1</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-2-1"><i class="fa fa-check"></i>Step 2</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-3"><i class="fa fa-check"></i>Step 3</a></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-4"><i class="fa fa-check"></i>Step 4</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-2---post-analysis-add-drug-target-gene-sets-to-the-network"><i class="fa fa-check"></i>Exercise 2 - Post analysis (add drug target gene-sets to the network)</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-5"><i class="fa fa-check"></i>Step 5</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-3---autoannotate-the-network"><i class="fa fa-check"></i>Exercise 3 - Autoannotate the Network</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-6"><i class="fa fa-check"></i>Step 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#exercise-4-optional---explore-results-in-genemania-or-string"><i class="fa fa-check"></i>Exercise 4 (Optional) - Explore results in GeneMANIA or STRING</a>
<ul>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#step-7"><i class="fa fa-check"></i>Step 7</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gsea_mod3.html"><a href="gsea_mod3.html#bonus---automation.-3"><i class="fa fa-check"></i>Bonus - Automation.</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html"><i class="fa fa-check"></i>Module 3 Lab: (Bonus) Automation</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#goal-of-the-exercise-5"><i class="fa fa-check"></i>Goal of the exercise:</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-1---install-rrstudio"><i class="fa fa-check"></i>Set Up - Option 1 - Install R/Rstudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#set-up---option-2---docker-image-with-rrstudio"><i class="fa fa-check"></i>Set Up - Option 2 - Docker image with R/Rstudio</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#what-is-docker"><i class="fa fa-check"></i>What is docker?</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker---basic-term-definition"><i class="fa fa-check"></i>Docker - Basic term definition</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#container"><i class="fa fa-check"></i>Container</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#image"><i class="fa fa-check"></i>Image</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#docker-volumes"><i class="fa fa-check"></i>Docker Volumes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#r_docker"><i class="fa fa-check"></i>Install Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#windows"><i class="fa fa-check"></i>Windows</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#macos-linux"><i class="fa fa-check"></i>MacOS / Linux</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#create-your-first-notebook-using-docker"><i class="fa fa-check"></i>Create your first notebook using Docker</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-coding"><i class="fa fa-check"></i>Start coding!</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#start-using-automation"><i class="fa fa-check"></i>Start using automation</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#running-example-notebooks-in-local-rstudio"><i class="fa fa-check"></i>Running example notebooks in local RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-1---launch-rstudio"><i class="fa fa-check"></i>Step 1 - launch RStudio</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-2---create-a-new-project"><i class="fa fa-check"></i>Step 2 - create a new project</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-3---open-example-rnotebook"><i class="fa fa-check"></i>Step 3 - Open example RNotebook</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#step-4---step-through-notebook-to-run-the-analysis"><i class="fa fa-check"></i>Step 4 - Step through notebook to run the analysis</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#exercises"><i class="fa fa-check"></i>Exercises</a></li>
<li class="chapter" data-level="" data-path="automation.html"><a href="automation.html#additional-resources"><i class="fa fa-check"></i>Additional resources</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Compute Ontario - Bioinformatics for Pathway Enrichment Analysis ( June 2025 )</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="welcome" class="section level1 hasAnchor">
<h1>Welcome <a id="welcome"></a><a href="welcome.html#welcome" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Welcome to Bioinformatics for Pathway Enrichment Analysis - June 2025</p>
<div id="meet-your-faculty" class="section level2 hasAnchor">
<h2>Meet your Faculty<a href="welcome.html#meet-your-faculty" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="veronique-voisin" class="section level3 hasAnchor">
<h3>Veronique Voisin<a href="welcome.html#veronique-voisin" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Research Associate,<br> Donnelly Centre for Cellular and Biomolecular Research, <br>University of Toronto</p>
<p><img src="images/faculty/veronique_voisin.jpeg" style="float:right;width:125px;padding-left:10px;" />
Veronique is currently a bioinformatician applying pathway and networks analysis to high throughput genomics data for OICR cancer stem cell program. Previously, she worked on characterizing the gene signatures of different types of leukemias using a murine model</p>
<p> 
 </p>
</div>
<div id="ruth-isserlin" class="section level3 hasAnchor">
<h3>Ruth Isserlin<a href="welcome.html#ruth-isserlin" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Research data analyst,<br>Donnelly Centre for Cellular and Bimolecular Research, <br>University of Toronto</p>
<p><img src="images/faculty/ruth_isserlin.jpeg" style="float:left;width:125px;padding-right:10px;" />
Bioinformatician and data analyst in the Bader lab applying pathway and data analysis to varied data types. Developed Enrichment Map App for Cytoscape, an app to visually translate functional enrichment results from popular enrichment tools like GSEA to networks. Further developed the Enrichment Map Pipeline including development of additional Apps to help summarize and analyze resulting Enrichment Maps, including PostAnalysis, WordCloud, and AutoAnnotate App.</p>
<hr />
<p>Thank you for attending the Bioinformatics for Pathway Enrichment Analysis
workshop! Help us make this workshop better by filling out <a href="https://forms.gle/D8w8qyJ1r71rFnZe9">our survey</a>.</p>
<hr />
</div>
</div>
<div id="class-materials" class="section level2 hasAnchor">
<h2>Class Materials<a href="welcome.html#class-materials" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can download the printed course manual <a href="https://drive.google.com/a/bioinformatics.ca/file/d/1HcPuiYUJe69w3_0aNpAfhk7DipcacA6r/view?usp=sharing">here</a>.</p>
</div>
<div id="schedule" class="section level2 hasAnchor">
<h2>Workshop Schedule<a href="welcome.html#schedule" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="images/time_table_pic.png" /></p>
</div>
<div id="pre-workshop" class="section level2 hasAnchor">
<h2>Pre-Workshop Materials and Laptop Setup Instructions<a href="welcome.html#pre-workshop" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="laptop-setup-instructions" class="section level3 hasAnchor">
<h3>Laptop Setup Instructions<a href="welcome.html#laptop-setup-instructions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A Check list to setup your laptop can be found <a href="https://docs.google.com/forms/d/e/1FAIpQLSdknqfaPi-XJDeFwji5xga7rg-jdGiYsZWxW6zTCjjqbHcHsw/viewform?usp=sharing">here</a></p>
<p>Install these tools on your laptop before coming to the workshop:</p>
</div>
<div id="basic-programs" class="section level3 hasAnchor">
<h3>Basic programs<a href="welcome.html#basic-programs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>A robust text editor:
<ul>
<li>For Windows/PC - <a href="http://notepad-plus-plus.org/">notepad++</a><br />
</li>
<li>For Linux - <a href="http://projects.gnome.org/gedit/">gEdit</a><br />
</li>
<li>For Mac – <a href="http://www.barebones.com/products/textwrangler/download.html">TextWrangler</a></li>
</ul></li>
<li>A file decompression tool.
<ul>
<li>For Windows/PC – <a href="http://www.7-zip.org/">7zip</a>.<br />
</li>
<li>For Linux – <a href="http://www.gzip.org">gzip</a>.<br />
</li>
<li>For Mac – already there.</li>
</ul></li>
<li>A robust internet browser such as:
<ul>
<li>Firefox</li>
<li>Safari</li>
<li>Chrome</li>
<li>Microsoft Edge</li>
</ul></li>
<li>A PDF Viewer
<ul>
<li>Adobe Acrobat or equivalent</li>
</ul></li>
</ol>
</div>
<div id="cytoscape-installation" class="section level3 hasAnchor">
<h3>Cytoscape Installation<a href="welcome.html#cytoscape-installation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Please install the latest version of <a href="https://github.com/cytoscape/cytoscape/releases/3.10.2/">Cytoscape 3.10.2</a> or <a href="https://cytoscape.org/download.html">Cytoscape Download</a> as well as a group of Cytoscape Apps that we will be using for different parts of the course.</p>
<ol style="list-style-type: decimal">
<li>Install Cytoscape 3.10.2:
<ul>
<li>Go to: <a href="https://github.com/cytoscape/cytoscape/releases/3.10.2/" class="uri">https://github.com/cytoscape/cytoscape/releases/3.10.2/</a> OR <a href="https://cytoscape.org/download.html" class="uri">https://cytoscape.org/download.html</a></li>
<li>Choose the version corresponding to your operating system (OS, Windows or UNIX)</li>
<li>Follow instructions to install cytoscape</li>
<li>Verify that Cytoscape has been installed correctly by launching the newly installed application</li>
</ul></li>
<li>Install the following Cytoscape Apps - Apps are installed from within Cytoscape.
<ul>
<li>In order to install Apps launch Cytoscape<br />
</li>
<li>From the menu bar, select ‘Apps’, then ‘App Store’, then ‘Show App Store’. <img src="images/cytoscape_app_menu.png" /></li>
<li>App Store will appear in left hand Panel <img src="images/Cytoscape_app_manager.png" /></li>
<li>Within search bar at the top of the panel, search for the app listed below. Once you click on search icon a web browser will be launched with the apps that match your search.</li>
<li>Select the correct app (there might be a few that match your search term).</li>
<li>Click on “Install” <img src="images/app_store_download.png" /></li>
<li>install the following:
<ul>
<li>EnrichmentMap 3.4.0</li>
<li>EnrichmentMap Pipeline Collection 1.1.0 (it will install ClusterMaker2 v2.3.4, WordCloud v3.1.4 and AutoAnnotate v1.5.0)</li>
<li>GeneMANIA 3.5.3</li>
<li>IRegulon 1.3</li>
<li>ReactomeFIPlugin 8.0.6 - <a href="http://apps.cytoscape.org/apps/reactomefiplugin" class="uri">http://apps.cytoscape.org/apps/reactomefiplugin</a></li>
<li>stringApp 2.0.3</li>
<li>scNetViz 1.7.1</li>
<li>yFiles Layout Algorithms 1.1.4</li>
</ul></li>
</ul></li>
<li>Install the data set within GeneMANIA app. <strong>This requires time and a good network connection to download completely (~15mins)</strong>
<ul>
<li>From the menu bar, select ‘Apps’, hover over ‘GeneMANIA’, then select ‘Choose Another Data Set’.</li>
<li>From the list of available data sets, select the most recent and under ‘Include only these networks:’ select ‘all’. Click on ‘Download’.</li>
<li>An ‘Install Data’ window will pop-up. Select H.Sapiens Human (2589 MB). Click on ‘Install’.</li>
</ul></li>
</ol>
</div>
<div id="gsea-installation" class="section level3 hasAnchor">
<h3>GSEA Installation<a href="welcome.html#gsea-installation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Please install the latest version of GSEA (4.3.3)</p>
<ol style="list-style-type: decimal">
<li>Download GSEA
<ul>
<li>Go to the <a href="http://www.broadinstitute.org/gsea/index.jsp">GSEA page</a><br />
</li>
<li>Register (using an institutional email address)</li>
<li>Login</li>
<li>Locate the Download page and download the version corresponding to your system
<ul>
<li>MAC users: download GSEA_4.3.3.app.zip</li>
<li>Window users: download GSEA_Win_4.3.3-installer.exe</li>
<li>Unix users: download GSEA_Linux_4.3.3.zip</li>
<li><img src="images/gsea_download_exe.png" /></li>
</ul></li>
<li>Launch GSEA to test it.</li>
</ul></li>
<li>Download GSEA for command line : this is necessary for all platform users to run GSEA from a script (integrated workflow on day 3)
<ul>
<li>Download GSEA_4.3.3.zip (and keep it for later use during the workshop)</li>
<li><img src="images/gsea_download_command.png" /></li>
</ul></li>
</ol>
</div>
<div id="docker-installation" class="section level3 hasAnchor">
<h3>Docker Installation<a href="welcome.html#docker-installation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Docker is a virtualization software that allows you to run programs isolated from your current laptop set up. It eases the burden of installing multiple software requirements and packages.</p>
<ol style="list-style-type: decimal">
<li>Please install the latest version of Docker Desktop.
<ul>
<li><a href="https://docs.docker.com/desktop/install/windows-install/">Windows</a></li>
<li><a href="https://docs.docker.com/desktop/install/mac-install/">OSX</a> - make sure to select the version specific for your computer. Newer macs (later than 2021) will contain the Apple silicon (M1/M2/M3). Older computers might be intel based.</li>
<li><a href="https://docs.docker.com/desktop/install/linux-install/">Linux</a></li>
</ul></li>
<li>Pull the required images used in the course
<ul>
<li>Open docker desktop (If docker is already running you can find the docker icon in your task bar. Right click on the icon and select “Go to Dashboard”)</li>
<li><img src="images/docker_dashboard_open.png" /></li>
<li>Find the search bar in the docker desktop dashboard</li>
<li><img src="images/docker_dashboard_search.png" /><br />
</li>
<li>Enter “risserlin/workshop_base_image” into the search bar at the top of the docker desktop dashboard.</li>
<li><img src="images/docker_dashboard_imagefind.png" /></li>
<li>Click on “Pull” to download the image.<br />
</li>
<li><img src="images/docker_dashboard_imagefind_annot.png" /></li>
<li>Enter “risserlin/nest_docker_lymphnode” into the search bar at the top of the docker desktop dashboard.</li>
<li><img src="images/docker_dashboard_imagefind_nest.png" /></li>
<li>Click on “Pull” to download the image.<br />
</li>
<li><img src="images/docker_dashboard_imagefind_nest_annot.png" /></li>
</ul></li>
<li>You should now see both of your images listed in the docker desktop image section (in the local tab)
<ul>
<li><img src="images/docker_dashboard_image_installed.png" /></li>
</ul></li>
</ol>
</div>
</div>
<div id="pre-workshop-tutorials" class="section level2 hasAnchor">
<h2>Pre-workshop Tutorials<a href="welcome.html#pre-workshop-tutorials" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is in your best interest to complete these before the workshop.</p>
<div id="cytoscape-preparation-tutorials" class="section level3 hasAnchor">
<h3>Cytoscape Preparation tutorials<a href="welcome.html#cytoscape-preparation-tutorials" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Go to : <a href="https://github.com/cytoscape/cytoscape-tutorials/wiki" class="uri">https://github.com/cytoscape/cytoscape-tutorials/wiki</a> and follow :</p>
<ul>
<li><a href="https://cytoscape.org/cytoscape-tutorials/protocols/tour-of-cytoscape/#/">Tour of Cytoscape</a></li>
<li><a href="https://cytoscape.org/cytoscape-tutorials/protocols/basic-data-visualization/#/">Basic Data Visualization</a></li>
</ul>
</div>
<div id="r-tutorial" class="section level3 hasAnchor">
<h3>R Tutorial<a href="welcome.html#r-tutorial" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Use your newly installed docker workshop_base_image to try out R and go through the following tutorial -</p>
<ul>
<li><a href="https://genviz.org/module-02-r/0002/02/01/introductionToR/">R tutorial</a> - There will be instructions on how to install R and RStudio in the tutorial. Instead of installing use the workshop_base_image docker image that you installed above as follows:
<ul>
<li>Open docker desktop (If docker is already running you can find the docker icon in your task bar. Right click on the icon and select “Go to Dashboard”)</li>
<li><img src="images/docker_dashboard_open.png" /></li>
<li>Click on Images –&gt; Local –&gt; And find the workshop_base_image. click on the Play button</li>
<li><img src="images/docker_launch_image.png" /></li>
<li>Expand the ‘optional settings’</li>
<li><img src="images/docker_new_container.png" /></li>
<li>Change -
<ul>
<li>‘container name’ to R_tutorial,</li>
<li>‘Host Port’ to 8787,</li>
<li>Add environment variable PASSWORD and set value to password</li>
<li><img src="images/docker_container_settings.png" /></li>
</ul></li>
<li>Click on ‘Run’. Docker will display a tab with all the information about the container you just launched</li>
<li><img src="images/docker_container_success.png" /></li>
<li>Open a web browser and navigate to localhost:8787</li>
<li><img src="images/docker_localhost.png" /></li>
<li>Username - rstudio, password - password (or whatever you entered as the PASSWORD settting when you launch the container)</li>
<li>You should now have an r studio session running in your web browser</li>
<li><img src="images/docker_rstudio.png" /></li>
<li>When you are finished doing the tutorial remember to turn off your docker container and dacker as they both use up a lot of your computer’s resources.</li>
<li><img src="images/docker_stop.png" /></li>
</ul></li>
</ul>
</div>
<div id="pre-workshop-readings-and-lectures" class="section level3 hasAnchor">
<h3>Pre-workshop Readings and Lectures<a href="welcome.html#pre-workshop-readings-and-lectures" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Video Module 1 - <a href="intro.html#intro">Introduction to Pathway and Network Analysis by Gary Bader</a></li>
<li><strong><em>Pathway enrichment analysis and visualization of omics data using g:Profiler, GSEA, Cytoscape and EnrichmentMap</em></strong> Reimand J, Isserlin R, Voisin V, Kucera M, Tannus-Lopes C, Rostamianfar A, Wadi L, Meyer M, Wong J, Xu C, Merico D, Bader GD <a href="https://www.nature.com/articles/s41596-018-0103-9">Nat Protoc. 2019 Feb;14(2):482-517</a> - <a href="http://baderlab.org/Publications#EM_2019">Available here as well</a></li>
</ol>
<hr />
</div>
<div id="additional-tutorials" class="section level3 hasAnchor">
<h3>Additional tutorials <a id="additionaltutorials"></a><a href="welcome.html#additional-tutorials" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p><strong><em>How to visually interpret biological data using networks</em></strong> Merico D, Gfeller D, Bader GD <a href="https://www.nature.com/articles/nbt.1567">Nature Biotechnology 2009 Oct 27, 921-924</a> - <a href="http://baderlab.org/Publications#interpret_networks">Available here as well</a></p></li>
<li><p><strong><em>g:Profiler–a web-based toolset for functional profiling of gene lists from large-scale experiments.</em></strong> Reimand J, Kull M, Peterson H, Hansen J, Vilo J <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1933153/">Nucleic Acids Res. 2007 Jul;35</a></p></li>
<li><p><strong><em>g:Profiler: a web server for functional enrichment analysis and conversions of gene lists (2019 update)</em></strong> Raudvere U, Kolberg L, Kuzmin I, Arak T, Adler P, Peterson H, Vilo J <a href="https://academic.oup.com/nar/advance-article/doi/10.1093/nar/gkz369/5486750">Nucleic Acids Res. 2019 May 8</a></p></li>
<li><p><strong><em>Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles</em></strong> Subramanian A, Tamayo P, Mootha VK, Mukherjee S, Ebert BL, Gillette MA, Paulovich A, Pomeroy SL, Golub TR, Lander ES, Mesirov JP <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/">Proc Natl Acad Sci U S A. 2005 Oct 25;102(43)</a></p></li>
</ul>
<p>Interacting with Cytoscape using CyRest and command lines (for advanced users):
<a href="https://github.com/cytoscape/cytoscape-automation/blob/master/for-scripters/R/advanced-cancer-networks-and-data-rcy3.Rmd" class="uri">https://github.com/cytoscape/cytoscape-automation/blob/master/for-scripters/R/advanced-cancer-networks-and-data-rcy3.Rmd</a></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="intro.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": "https://github.com/risserlin/ComputeOntario_Pathways_2025/commit/master/0.0.Welcome_info.Rmd",
"text": null
},
"view": {
"link": "https://github.com/risserlin/ComputeOntario_Pathways_2025/master/0.0.Welcome_info.Rmd",
"text": null
},
"download": ["Compute Ontario - Bioinformatics for Pathway Enrichment Analysis.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
